// @guardian/google-admanager-api/dist/client/services/adjustment/adjustment.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForecastAdjustmentVolumeType = exports.ForecastAdjustmentStatus = void 0;
/**
 * The status of a forecast adjustment. Inactive adjustments are not applied during forecasting.
 */
var ForecastAdjustmentStatus;
(function (ForecastAdjustmentStatus) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ForecastAdjustmentStatus["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates the current adjustment is active.
     */
    ForecastAdjustmentStatus["ACTIVE"] = "ACTIVE";
    /**
     * Indicates the current adjustment is inactive.
     */
    ForecastAdjustmentStatus["INACTIVE"] = "INACTIVE";
})(ForecastAdjustmentStatus || (exports.ForecastAdjustmentStatus = ForecastAdjustmentStatus = {}));
/**
 * Options for how the volume settings of a
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/AdjustmentService.ForecastAdjustment ForecastAdjustment} are defined.
 */
var ForecastAdjustmentVolumeType;
(function (ForecastAdjustmentVolumeType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ForecastAdjustmentVolumeType["UNKNOWN"] = "UNKNOWN";
    /**
     * Volume is defined by a series of daily ad opportunity counts.
     */
    ForecastAdjustmentVolumeType["DAILY_VOLUME"] = "DAILY_VOLUME";
    /**
     * Volume is defined by a single total ad opportunity count.
     */
    ForecastAdjustmentVolumeType["TOTAL_VOLUME"] = "TOTAL_VOLUME";
    /**
     * Volume is defined by historical volume data.
     */
    ForecastAdjustmentVolumeType["HISTORICAL_BASIS_VOLUME"] = "HISTORICAL_BASIS_VOLUME";
})(ForecastAdjustmentVolumeType || (exports.ForecastAdjustmentVolumeType = ForecastAdjustmentVolumeType = {}));
