// @guardian/google-admanager-api/dist/client/services/adjustment/adjustment.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdjustmentService = void 0;
class AdjustmentService {
    constructor(client) {
        this._client = client;
    }
    calculateDailyAdOpportunityCounts(forecastAdjustment) {
        const res = this._client.calculateDailyAdOpportunityCounts({
            forecastAdjustment,
        });
        return res;
    }
    createForecastAdjustments(forecastAdjustments) {
        const res = this._client.createForecastAdjustments({ forecastAdjustments });
        return res;
    }
    createTrafficForecastSegments(trafficForecastSegments) {
        const res = this._client.createTrafficForecastSegments({
            trafficForecastSegments,
        });
        return res;
    }
    getForecastAdjustmentsByStatement(filterStatement) {
        const res = this._client.getForecastAdjustmentsByStatement({
            filterStatement,
        });
        return res;
    }
    getTrafficForecastSegmentsByStatement(filterStatement) {
        const res = this._client.getTrafficForecastSegmentsByStatement({
            filterStatement,
        });
        return res;
    }
    performForecastAdjustmentAction(forecastAdjustmentAction, filterStatement) {
        const res = this._client.performForecastAdjustmentAction({
            forecastAdjustmentAction: {
                attributes: {
                    "xsi:type": forecastAdjustmentAction.constructor.name,
                },
            },
            filterStatement,
        });
        return res;
    }
    updateForecastAdjustments(forecastAdjustments) {
        const res = this._client.updateForecastAdjustments({ forecastAdjustments });
        return res;
    }
    updateTrafficForecastSegments(trafficForecastSegments) {
        const res = this._client.updateTrafficForecastSegments({
            trafficForecastSegments,
        });
        return res;
    }
}
exports.AdjustmentService = AdjustmentService;
