// @guardian/google-admanager-api/dist/client/services/cdnConfiguration/cdnConfiguration.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CdnConfigurationService = void 0;
class CdnConfigurationService {
    constructor(client) {
        this._client = client;
    }
    async createCdnConfigurations(cdnConfigurations) {
        return this._client.createCdnConfigurations({ cdnConfigurations });
    }
    async getCdnConfigurationsByStatement(statement) {
        return this._client.getCdnConfigurationsByStatement({
            statement,
        });
    }
    async performCdnConfigurationAction(cdnConfigurationAction, filterStatement) {
        return this._client.performCdnConfigurationAction({
            cdnConfigurationAction: {
                attributes: {
                    "xsi:type": cdnConfigurationAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateCdnConfigurations(cdnConfigurations) {
        return this._client.updateCdnConfigurations({ cdnConfigurations });
    }
}
exports.CdnConfigurationService = CdnConfigurationService;
