// @guardian/google-admanager-api/dist/client/services/cdnConfiguration/cdnConfiguration.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OriginForwardingType = exports.SecurityPolicyType = exports.CdnConfigurationStatus = exports.CdnConfigurationType = void 0;
/**
 * Indicates the type of CDN configuration for {@link https://developers.google.com/ad-manager/api/reference/v202505/CdnConfigurationService.CdnConfiguration CdnConfiguration}.
 */
var CdnConfigurationType;
(function (CdnConfigurationType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CdnConfigurationType["UNKNOWN"] = "UNKNOWN";
    /**
     * A configuration that specifies where and how LiveStreamEvent content should be ingested and delivered.
     */
    CdnConfigurationType["LIVE_STREAM_SOURCE_CONTENT"] = "LIVE_STREAM_SOURCE_CONTENT";
})(CdnConfigurationType || (exports.CdnConfigurationType = CdnConfigurationType = {}));
/**
 * Indicates the status of the {@link https://developers.google.com/ad-manager/api/reference/v202505/CdnConfigurationService.CdnConfiguration CdnConfiguration}.
 */
var CdnConfigurationStatus;
(function (CdnConfigurationStatus) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CdnConfigurationStatus["UNKNOWN"] = "UNKNOWN";
    /**
     * The CDN configuration is in use.
     */
    CdnConfigurationStatus["ACTIVE"] = "ACTIVE";
    /**
     * The CDN configuration is no longer used.
     */
    CdnConfigurationStatus["ARCHIVED"] = "ARCHIVED";
})(CdnConfigurationStatus || (exports.CdnConfigurationStatus = CdnConfigurationStatus = {}));
/**
 * Indicates the type of security policy associated with access to a CDN. Different security policies require different parameters in a **`SecurityPolicy`**.
 */
var SecurityPolicyType;
(function (SecurityPolicyType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    SecurityPolicyType["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates that no authentication is necessary.
     */
    SecurityPolicyType["NONE"] = "NONE";
    /**
     * Security policy for accessing content on the Akamai CDN.
     */
    SecurityPolicyType["AKAMAI"] = "AKAMAI";
})(SecurityPolicyType || (exports.SecurityPolicyType = SecurityPolicyType = {}));
/**
 * Indicates the type of origin forwarding used to support Akamai authentication policies for
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent}
 */
var OriginForwardingType;
(function (OriginForwardingType) {
    /**
     * Indicates that origin forwarding is set up by passing an originpath query string parameter (necessary for Akamai dynamic packaging to work)
     */
    OriginForwardingType["ORIGIN_PATH"] = "ORIGIN_PATH";
    /**
     * Indicates that conventional origin forwarding is used.
     */
    OriginForwardingType["CONVENTIONAL"] = "CONVENTIONAL";
    /**
     * Indicates that origin forwarding is not being used.
     */
    OriginForwardingType["NONE"] = "NONE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    OriginForwardingType["UNKNOWN"] = "UNKNOWN";
})(OriginForwardingType || (exports.OriginForwardingType = OriginForwardingType = {}));
