// @guardian/google-admanager-api/dist/client/services/liveStreamEvent/liveStreamEvent.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamEventService = void 0;
class LiveStreamEventService {
    constructor(client) {
        this._client = client;
    }
    createLiveStreamEvents(liveStreamEvents) {
        return this._client.createLiveStreamEvents({ liveStreamEvents });
    }
    createSlates(slates) {
        return this._client.createSlates({ slates });
    }
    getLiveStreamEventsByStatement(filterStatement) {
        return this._client.getLiveStreamEventsByStatement({
            filterStatement,
        });
    }
    getSlatesByStatement(statement) {
        return this._client.getSlatesByStatement({
            statement,
        });
    }
    performLiveStreamEventAction(liveStreamEventAction, filterStatement) {
        return this._client.performLiveStreamEventAction({
            liveStreamEventAction: {
                attributes: {
                    "xsi:type": liveStreamEventAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    performSlateAction(slateAction, filterStatement) {
        return this._client.performSlateAction({
            slateAction: {
                attributes: {
                    "xsi:type": slateAction,
                },
            },
            filterStatement,
        });
    }
    updateLiveStreamEvents(liveStreamEvents) {
        return this._client.updateLiveStreamEvents({ liveStreamEvents });
    }
    updateSlates(slates) {
        return this._client.updateSlates({ slates });
    }
}
exports.LiveStreamEventService = LiveStreamEventService;
