// @guardian/google-admanager-api/dist/client/services/liveStreamEvent/liveStreamEvent.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TranscodeStatus = exports.SlateStatus = exports.AdServingFormat = exports.AdBreakMarkupType = exports.StreamingFormat = exports.DynamicAdInsertionType = exports.RefreshType = exports.PlaylistType = exports.AdBreakFillType = exports.LiveStreamEventStatus = void 0;
/**
 * Describes the status of a {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} object.
 */
var LiveStreamEventStatus;
(function (LiveStreamEventStatus) {
    /**
     * Indicates the LiveStreamEvent has been created and is eligible for streaming.
     */
    LiveStreamEventStatus["ACTIVE"] = "ACTIVE";
    /**
     * Indicates the LiveStreamEvent has been archived.
     */
    LiveStreamEventStatus["ARCHIVED"] = "ARCHIVED";
    /**
     * Indicates the LiveStreamEvent has been paused. This can be made ACTIVE at later time.
     */
    LiveStreamEventStatus["PAUSED"] = "PAUSED";
    /**
     * Indicates that the stream is still being served, but ad insertion should be paused temporarily.
     */
    LiveStreamEventStatus["ADS_PAUSED"] = "ADS_PAUSED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    LiveStreamEventStatus["UNKNOWN"] = "UNKNOWN";
})(LiveStreamEventStatus || (exports.LiveStreamEventStatus = LiveStreamEventStatus = {}));
/**
 * Describes what should be used to fill an empty ad break during a live stream.
 */
var AdBreakFillType;
(function (AdBreakFillType) {
    /**
     * Ad break should be filled with slate.
     */
    AdBreakFillType["SLATE"] = "SLATE";
    /**
     * Ad break should be filled with underlying content.
     */
    AdBreakFillType["UNDERLYING_CONTENT"] = "UNDERLYING_CONTENT";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AdBreakFillType["UNKNOWN"] = "UNKNOWN";
})(AdBreakFillType || (exports.AdBreakFillType = AdBreakFillType = {}));
/**
 * Describes the type of the playlist associated with this live stream. This is analagous to the EXT-X-PLAYLIST-TYPE HLS tag.
 * Use PlaylistType.EVENT for streams with the value "#EXT-X-PLAYLIST-TYPE:EVENT" and use PlaylistType.LIVE for streams without the tag.
 */
var PlaylistType;
(function (PlaylistType) {
    /**
     * The playlist is an event, which means that media segments can only be added to the end of the playlist. This allows viewers to scrub back to the beginning of the playlist.
     */
    PlaylistType["EVENT"] = "EVENT";
    /**
     * The playlist is a live stream and there are no restrictions on whether media segments can be removed from the beginning of the playlist.
     */
    PlaylistType["LIVE"] = "LIVE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    PlaylistType["UNKNOWN"] = "UNKNOWN";
})(PlaylistType || (exports.PlaylistType = PlaylistType = {}));
/**
 * Enumerates the different ways an HLS master playlist on a live stream will can be refreshed.
 */
var RefreshType;
(function (RefreshType) {
    /**
     * The master playlist will automatically be refreshed.
     */
    RefreshType["AUTOMATIC"] = "AUTOMATIC";
    /**
     * The master playlist will only be refreshed when requested.
     */
    RefreshType["MANUAL"] = "MANUAL";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    RefreshType["UNKNOWN"] = "UNKNOWN";
})(RefreshType || (exports.RefreshType = RefreshType = {}));
/**
 * Describes how the live stream will have ads dynamically inserted into playlists.
 */
var DynamicAdInsertionType;
(function (DynamicAdInsertionType) {
    /**
     * Content manifest is served by Google DAI. Content and ads are stitched together into a unique video manifest per user.
     */
    DynamicAdInsertionType["LINEAR"] = "LINEAR";
    /**
     * Content manifest is served by the partner, embedding Google DAI ad segment URLs which redirect to unique Google DAI ad segments per user.
     */
    DynamicAdInsertionType["POD_SERVING_REDIRECT"] = "POD_SERVING_REDIRECT";
    /**
     * Ads manifest is served by Google DAI, containing unique ad pod segments for the video player to switch to from the content stream, or for the partner to stitch directly into the user content manifest.
     */
    DynamicAdInsertionType["POD_SERVING_MANIFEST"] = "POD_SERVING_MANIFEST";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    DynamicAdInsertionType["UNKNOWN"] = "UNKNOWN";
})(DynamicAdInsertionType || (exports.DynamicAdInsertionType = DynamicAdInsertionType = {}));
/**
 * The {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} streaming format.
 */
var StreamingFormat;
(function (StreamingFormat) {
    /**
     * The format of the live stream media is HTTP Live Streaming.
     */
    StreamingFormat["HLS"] = "HLS";
    /**
     * The format of the live stream media is MPEG-DASH.
     */
    StreamingFormat["DASH"] = "DASH";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    StreamingFormat["UNKNOWN"] = "UNKNOWN";
})(StreamingFormat || (exports.StreamingFormat = StreamingFormat = {}));
/**
 * Describes the SCTE ad break markups for a LiveStreamEvent.
 */
var AdBreakMarkupType;
(function (AdBreakMarkupType) {
    /**
     * The CUE-OUT/CUE-IN ad break marker type. This mark up type is only applicable for HLS live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_HLS_EXT_CUE"] = "AD_BREAK_MARKUP_HLS_EXT_CUE";
    /**
     * The CUE (Adobe/Azure Prime Time) ad break marker type. This mark up type is only applicable for HLS live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_HLS_PRIMETIME_SPLICE"] = "AD_BREAK_MARKUP_HLS_PRIMETIME_SPLICE";
    /**
     * The DATERANGE (Anvato) ad break marker type. This mark up type is only applicable for HLS live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_HLS_DATERANGE_SPLICE"] = "AD_BREAK_MARKUP_HLS_DATERANGE_SPLICE";
    /**
     * The SCTE35 XML Splice In/Out ad break marker type. This markup type is only applicable for DASH live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_SCTE35_XML_SPLICE_INSERT"] = "AD_BREAK_MARKUP_SCTE35_XML_SPLICE_INSERT";
    /**
     * The SCTE35 Binary Splice Insert ad break marker type. This mark up type is only applicable for HLS and DASH live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_SCTE35_BINARY_SPLICE_INSERT"] = "AD_BREAK_MARKUP_SCTE35_BINARY_SPLICE_INSERT";
    /**
     * The SCTE35 Binary Time Signal: Provider Ad Start/End ad break marker type. This mark up type is only applicable for HLS and DASH live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_SCTE35_BINARY_PROVIDER_AD_START_END"] = "AD_BREAK_MARKUP_SCTE35_BINARY_PROVIDER_AD_START_END";
    /**
     * The SCTE35 Binary Time Signal: Provider Placement Opportunity Start/End ad break marker type. This mark up type is only applicable for HLS and DASH live streams.
     */
    AdBreakMarkupType["AD_BREAK_MARKUP_SCTE35_BINARY_PROVIDER_PLACEMENT_OP_START_END"] = "AD_BREAK_MARKUP_SCTE35_BINARY_PROVIDER_PLACEMENT_OP_START_END";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AdBreakMarkupType["UNKNOWN"] = "UNKNOWN";
})(AdBreakMarkupType || (exports.AdBreakMarkupType = AdBreakMarkupType = {}));
/**
 * Indicates how the ads of the live stream should be served.
 */
var AdServingFormat;
(function (AdServingFormat) {
    /**
     * The ads are served through Google Ad Manager DAI.
     */
    AdServingFormat["AD_MANAGER_DAI"] = "AD_MANAGER_DAI";
    /**
     * The ads are served through Google Ad Manager Ad Serving.
     */
    AdServingFormat["DIRECT"] = "DIRECT";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AdServingFormat["UNKNOWN"] = "UNKNOWN";
})(AdServingFormat || (exports.AdServingFormat = AdServingFormat = {}));
/**
 * Describes the status of a Slate object.
 */
var SlateStatus;
(function (SlateStatus) {
    /**
     * Indicates the Slate has been created and is eligible for streaming.
     */
    SlateStatus["ACTIVE"] = "ACTIVE";
    /**
     * Indicates the Slate has been archived.
     */
    SlateStatus["ARCHIVED"] = "ARCHIVED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    SlateStatus["UNKNOWN"] = "UNKNOWN";
})(SlateStatus || (exports.SlateStatus = SlateStatus = {}));
/**
 * Possible server side transcoding states.
 */
var TranscodeStatus;
(function (TranscodeStatus) {
    TranscodeStatus["UNKNOWN"] = "UNKNOWN";
    TranscodeStatus["NOT_READY"] = "NOT_READY";
    TranscodeStatus["COMPLETED"] = "COMPLETED";
    TranscodeStatus["FAILED"] = "FAILED";
    TranscodeStatus["NEEDS_TRANSCODE"] = "NEEDS_TRANSCODE";
    TranscodeStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(TranscodeStatus || (exports.TranscodeStatus = TranscodeStatus = {}));
