// @guardian/google-admanager-api/dist/client/services/liveStreamEvent/liveStreamEvent.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefreshLiveStreamEventMasterPlaylists = exports.PauseLiveStreamEvents = exports.PauseLiveStreamEventAds = exports.ArchiveLiveStreamEvents = exports.ActivateLiveStreamEvents = exports.LiveStreamEventAction = void 0;
/**
 * Represents the actions that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} objects.
 */
class LiveStreamEventAction {
}
exports.LiveStreamEventAction = LiveStreamEventAction;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} objects.
 */
class ActivateLiveStreamEvents {
}
exports.ActivateLiveStreamEvents = ActivateLiveStreamEvents;
/**
 * The action used for archiving {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} objects.
 */
class ArchiveLiveStreamEvents {
}
exports.ArchiveLiveStreamEvents = ArchiveLiveStreamEvents;
/**
 * The action used for pausing ads {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} objects.
 */
class PauseLiveStreamEventAds {
}
exports.PauseLiveStreamEventAds = PauseLiveStreamEventAds;
/**
 * The action used for pausing {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} objects.
 */
class PauseLiveStreamEvents {
}
exports.PauseLiveStreamEvents = PauseLiveStreamEvents;
/**
 * The action used for refreshing the master playlists of
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.LiveStreamEvent LiveStreamEvent} objects.
 * This action will only get applied to live streams with a refresh type of
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/LiveStreamEventService.RefreshType#MANUAL RefreshType.MANUAL}.
 */
class RefreshLiveStreamEventMasterPlaylists {
}
exports.RefreshLiveStreamEventMasterPlaylists = RefreshLiveStreamEventMasterPlaylists;
