// @guardian/google-admanager-api/dist/client/services/proposal/proposal.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RfpType = exports.AdExchangeEnvironment = exports.NegotiationStatus = exports.ProposalCompanyAssociationType = exports.ProposalStatus = void 0;
/**
 * Describes the {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} status.
 */
var ProposalStatus;
(function (ProposalStatus) {
    /**
     * Indicates that the Proposal has just been created or retracted but no approval has been requested yet.
     */
    ProposalStatus["DRAFT"] = "DRAFT";
    /**
     * Indicates that a request for approval has been made for the Proposal.
     */
    ProposalStatus["PENDING_APPROVAL"] = "PENDING_APPROVAL";
    /**
     * Indicates that the Proposal has been approved and is ready to serve.
     */
    ProposalStatus["APPROVED"] = "APPROVED";
    /**
     * Indicates that the Proposal has been rejected in the approval workflow.
     */
    ProposalStatus["REJECTED"] = "REJECTED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ProposalStatus["UNKNOWN"] = "UNKNOWN";
})(ProposalStatus || (exports.ProposalStatus = ProposalStatus = {}));
/**
 * Describes the type of a Company associated with a Proposal.
 */
var ProposalCompanyAssociationType;
(function (ProposalCompanyAssociationType) {
    /**
     * The company is a primary agency.
     */
    ProposalCompanyAssociationType["PRIMARY_AGENCY"] = "PRIMARY_AGENCY";
    /**
     * The company is a billing agency.
     */
    ProposalCompanyAssociationType["BILLING_AGENCY"] = "BILLING_AGENCY";
    /**
     * The company is a branding agency.
     */
    ProposalCompanyAssociationType["BRANDING_AGENCY"] = "BRANDING_AGENCY";
    /**
     * The company is other type of agency.
     */
    ProposalCompanyAssociationType["OTHER_AGENCY"] = "OTHER_AGENCY";
    /**
     * The company is advertiser.
     */
    ProposalCompanyAssociationType["ADVERTISER"] = "ADVERTISER";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ProposalCompanyAssociationType["UNKNOWN"] = "UNKNOWN";
})(ProposalCompanyAssociationType || (exports.ProposalCompanyAssociationType = ProposalCompanyAssociationType = {}));
/**
 * Represents the proposal's negotiation status for Marketplace.
 */
var NegotiationStatus;
(function (NegotiationStatus) {
    /**
     * Indicates that a new Proposal has been created by the seller and has not been sent to Marketplace yet.
     */
    NegotiationStatus["SELLER_INITIATED"] = "SELLER_INITIATED";
    /**
     * Indicates that a new Proposal has been created by the buyer and is awaiting seller action.
     */
    NegotiationStatus["BUYER_INITIATED"] = "BUYER_INITIATED";
    /**
     * Indicates that a Proposal has been updated by the buyer and is awaiting seller approval.
     */
    NegotiationStatus["AWAITING_SELLER_REVIEW"] = "AWAITING_SELLER_REVIEW";
    /**
     * Indicates that a Proposal has been updated by the seller and is awaiting buyer approval.
     */
    NegotiationStatus["AWAITING_BUYER_REVIEW"] = "AWAITING_BUYER_REVIEW";
    /**
     * Indicates that the seller has accepted the Proposal and is awaiting the buyer's acceptance.
     */
    NegotiationStatus["ONLY_SELLER_ACCEPTED"] = "ONLY_SELLER_ACCEPTED";
    /**
     * Indicates that the Proposal has been accepted by both the buyer and the seller.
     */
    NegotiationStatus["FINALIZED"] = "FINALIZED";
    /**
     * Indicates that negotiations for the Proposal have been cancelled.
     */
    NegotiationStatus["CANCELLED"] = "CANCELLED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    NegotiationStatus["UNKNOWN"] = "UNKNOWN";
})(NegotiationStatus || (exports.NegotiationStatus = NegotiationStatus = {}));
/**
 * Identifies the format of inventory or "channel" in which ads serve.
 */
var AdExchangeEnvironment;
(function (AdExchangeEnvironment) {
    /**
     * Ads serve in a browser.
     */
    AdExchangeEnvironment["DISPLAY"] = "DISPLAY";
    /**
     * In-stream video ads serve in a video.
     */
    AdExchangeEnvironment["VIDEO"] = "VIDEO";
    /**
     * In-stream video ads serve in a game.
     */
    AdExchangeEnvironment["GAMES"] = "GAMES";
    /**
     * Ads serve in a mobile app.
     */
    AdExchangeEnvironment["MOBILE"] = "MOBILE";
    /**
     * Out-stream video ads serve in a mobile app. Examples include mobile app interstitials and mobile app rewarded ads.
     */
    AdExchangeEnvironment["MOBILE_OUTSTREAM_VIDEO"] = "MOBILE_OUTSTREAM_VIDEO";
    /**
     * Out-stream video ads serve in a browser. Examples include in-feed and in-banner video ads.
     */
    AdExchangeEnvironment["DISPLAY_OUTSTREAM_VIDEO"] = "DISPLAY_OUTSTREAM_VIDEO";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AdExchangeEnvironment["UNKNOWN"] = "UNKNOWN";
})(AdExchangeEnvironment || (exports.AdExchangeEnvironment = AdExchangeEnvironment = {}));
/**
 * Decribes the type of BuyerRfp.
 */
var RfpType;
(function (RfpType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    RfpType["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates the BuyerRfp is a Programmatic Guaranteed RFP.
     */
    RfpType["PROGRAMMATIC_GUARANTEED"] = "PROGRAMMATIC_GUARANTEED";
    /**
     * Indicates the BuyerRfp is a Preferred Deal RFP.
     */
    RfpType["PREFERRED_DEAL"] = "PREFERRED_DEAL";
})(RfpType || (exports.RfpType = RfpType = {}));
