// @guardian/google-admanager-api/dist/client/services/proposal/proposal.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOrderWithSellerData = exports.UnarchiveProposals = exports.TerminateNegotiations = exports.ResumeProposals = exports.ReserveProposals = exports.RequestBuyerReview = exports.RequestBuyerAcceptance = exports.PauseProposals = exports.EditProposalsForNegotiation = exports.DiscardLocalVersionEdits = exports.ArchiveProposals = exports.ProposalAction = void 0;
/**
 * Represents the actions that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} objects.
 */
class ProposalAction {
    buildAttributes() {
        return {};
    }
}
exports.ProposalAction = ProposalAction;
/**
 * The action used for archiving {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} objects.
 */
class ArchiveProposals {
    buildAttributes() {
        return {};
    }
}
exports.ArchiveProposals = ArchiveProposals;
/**
 * The action reverting the local {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} modifications to reflect the latest terms and private data in Marketplace.
 */
class DiscardLocalVersionEdits {
    buildAttributes() {
        return {};
    }
}
exports.DiscardLocalVersionEdits = DiscardLocalVersionEdits;
/**
 * Opens the fields of a {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} for edit.
 * This proposal will not receive updates from Marketplace while it's open for edit.
 * If the buyer updates the proposal while it is open for local editing,
 * Google will set {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.ProposalMarketplaceInfo#isNewVersionFromBuyer ProposalMarketplaceInfo.isNewVersionFromBuyer} to true.
 * You will then need to call DiscardProposalDrafts to revert your edits to get the buyer's latest changes,
 * and then perform this action to start making your edits again.
 */
class EditProposalsForNegotiation {
    buildAttributes() {
        return {};
    }
}
exports.EditProposalsForNegotiation = EditProposalsForNegotiation;
/**
 * The action used for pausing programmatic {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} objects.
 */
class PauseProposals {
    get reason() {
        return this.reasonField;
    }
    set reason(value) {
        this.reasonField = value;
    }
    buildAttributes() {
        return this.reason ? { reason: this.reason } : {};
    }
}
exports.PauseProposals = PauseProposals;
/**
 * The action used to request acceptance from the buyer for the
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} through Marketplace.
 * This action does check forecasting unless
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.RequestBuyerAcceptance#allowOverbook allowOverbook} is set to true.
 */
class RequestBuyerAcceptance {
    constructor() {
        this.allowOverbookField = false;
    }
    get allowOverbook() {
        return this.allowOverbookField;
    }
    set allowOverbook(value) {
        this.allowOverbookField = value;
    }
    buildAttributes() {
        return { allowOverbook: this.allowOverbook };
    }
}
exports.RequestBuyerAcceptance = RequestBuyerAcceptance;
/**
 * The action used to request buyer review for the {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal}.
 */
class RequestBuyerReview {
    buildAttributes() {
        return {};
    }
}
exports.RequestBuyerReview = RequestBuyerReview;
/**
 * The action to reserve inventory for {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} objects.
 * It does not allow overbooking unless {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.ReserveProposals#allowOverbook allowOverbook} is set to true.
 */
class ReserveProposals {
    constructor() {
        this.allowOverbookField = false;
    }
    get allowOverbook() {
        return this.allowOverbookField;
    }
    set allowOverbook(value) {
        this.allowOverbookField = value;
    }
    buildAttributes() {
        return { allowOverbook: this.allowOverbook };
    }
}
exports.ReserveProposals = ReserveProposals;
/**
 * The action used for resuming programmatic {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} objects.
 */
class ResumeProposals {
    buildAttributes() {
        return {};
    }
}
exports.ResumeProposals = ResumeProposals;
/**
 * The action for marking all negotiations on the {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} as terminated in Marketplace.
 */
class TerminateNegotiations {
    buildAttributes() {
        return {};
    }
}
exports.TerminateNegotiations = TerminateNegotiations;
/**
 * The action used for unarchiving {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalService.Proposal Proposal} objects.
 */
class UnarchiveProposals {
    buildAttributes() {
        return {};
    }
}
exports.UnarchiveProposals = UnarchiveProposals;
/**
 * The action to update a finalized Marketplace {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order} with the seller's data.
 */
class UpdateOrderWithSellerData {
    buildAttributes() {
        return {};
    }
}
exports.UpdateOrderWithSellerData = UpdateOrderWithSellerData;
