// @guardian/google-admanager-api/dist/client/services/proposal/proposal.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProposalService = void 0;
class ProposalService {
    constructor(client) {
        this._client = client;
    }
    async createProposals(proposals) {
        return this._client.createProposals({ proposals });
    }
    async getMarketplaceCommentsByStatement(filterStatement) {
        return this._client.getMarketplaceCommentsByStatement({
            filterStatement,
        });
    }
    async getProposalsByStatement(filterStatement) {
        return this._client.getProposalsByStatement({
            filterStatement,
        });
    }
    async performProposalAction(proposalAction, filterStatement) {
        return this._client.performProposalAction({
            ProposalAction: Object.assign({ attributes: {
                    "xsi:type": proposalAction.constructor.name,
                } }, proposalAction.buildAttributes()),
            filterStatement,
        });
    }
    async updateProposals(proposals) {
        return this._client.updateProposals({ proposals });
    }
}
exports.ProposalService = ProposalService;
