// @guardian/google-admanager-api/dist/client/services/streamActivityMonitor/streamActivityMonitor.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreativeTranscodeIdType = exports.SamErrorType = exports.ReportingType = void 0;
var ReportingType;
(function (ReportingType) {
    ReportingType["DISABLED"] = "DISABLED";
    ReportingType["CLIENT"] = "CLIENT";
    ReportingType["SERVER"] = "SERVER";
    ReportingType["AD_MEDIA"] = "AD_MEDIA";
    ReportingType["UNKNOWN"] = "UNKNOWN";
})(ReportingType || (exports.ReportingType = ReportingType = {}));
var SamErrorType;
(function (SamErrorType) {
    SamErrorType["INTERNAL_ERROR"] = "INTERNAL_ERROR";
    SamErrorType["AD_REQUEST_ERROR"] = "AD_REQUEST_ERROR";
    SamErrorType["VAST_PARSE_ERROR"] = "VAST_PARSE_ERROR";
    SamErrorType["UNSUPPORTED_AD_SYSTEM"] = "UNSUPPORTED_AD_SYSTEM";
    SamErrorType["CANNOT_FIND_UNIQUE_TRANSCODE_ID"] = "CANNOT_FIND_UNIQUE_TRANSCODE_ID";
    SamErrorType["CANNOT_FIND_MEDIA_FILE_PATH"] = "CANNOT_FIND_MEDIA_FILE_PATH";
    SamErrorType["MISSING_INLINE_ELEMENTS"] = "MISSING_INLINE_ELEMENTS";
    SamErrorType["MAX_WRAPPER_DEPTH_REACHED"] = "MAX_WRAPPER_DEPTH_REACHED";
    SamErrorType["INVALID_AD_SEQUENCE_NUMBER"] = "INVALID_AD_SEQUENCE_NUMBER";
    SamErrorType["FAILED_PING"] = "FAILED_PING";
    SamErrorType["AD_TAG_PARSE_ERROR"] = "AD_TAG_PARSE_ERROR";
    SamErrorType["VMAP_PARSE_ERROR"] = "VMAP_PARSE_ERROR";
    SamErrorType["INVALID_VMAP_RESPONSE"] = "INVALID_VMAP_RESPONSE";
    SamErrorType["NO_AD_BREAKS_IN_VMAP"] = "NO_AD_BREAKS_IN_VMAP";
    SamErrorType["CUSTOM_AD_SOURCE_IN_VMAP"] = "CUSTOM_AD_SOURCE_IN_VMAP";
    SamErrorType["AD_BREAK_TYPE_NOT_SUPPORTED"] = "AD_BREAK_TYPE_NOT_SUPPORTED";
    SamErrorType["NEITHER_AD_SOURCE_NOR_TRACKING"] = "NEITHER_AD_SOURCE_NOR_TRACKING";
    SamErrorType["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
    SamErrorType["AD_POD_DROPPED_TO_MANY_AD_PODS"] = "AD_POD_DROPPED_TO_MANY_AD_PODS";
    SamErrorType["AD_POD_DROPPED_EMPTY_ADS"] = "AD_POD_DROPPED_EMPTY_ADS";
    SamErrorType["AD_BREAK_WITHOUT_AD_POD"] = "AD_BREAK_WITHOUT_AD_POD";
    SamErrorType["TRANSCODING_IN_PROGRESS"] = "TRANSCODING_IN_PROGRESS";
    SamErrorType["UNSUPPORTED_VAST_VERSION"] = "UNSUPPORTED_VAST_VERSION";
    SamErrorType["AD_POD_DROPPED_BUMPER_ERROR"] = "AD_POD_DROPPED_BUMPER_ERROR";
    SamErrorType["NO_VALID_MEDIAFILES_FOUND"] = "NO_VALID_MEDIAFILES_FOUND";
    SamErrorType["EXCEEDS_MAX_FILLER"] = "EXCEEDS_MAX_FILLER";
    SamErrorType["SKIPPABLE_AD_NOT_SUPPORTED"] = "SKIPPABLE_AD_NOT_SUPPORTED";
    SamErrorType["AD_REQUEST_TIMEOUT"] = "AD_REQUEST_TIMEOUT";
    SamErrorType["AD_POD_DROPPED_UNSUPPORTED_TYPE"] = "AD_POD_DROPPED_UNSUPPORTED_TYPE";
    SamErrorType["DUPLICATE_AD_TAG"] = "DUPLICATE_AD_TAG";
    SamErrorType["FOLLOW_REDIRECTS_IS_FALSE"] = "FOLLOW_REDIRECTS_IS_FALSE";
    SamErrorType["UNKNOWN"] = "UNKNOWN";
})(SamErrorType || (exports.SamErrorType = SamErrorType = {}));
var CreativeTranscodeIdType;
(function (CreativeTranscodeIdType) {
    CreativeTranscodeIdType["AD_ID"] = "AD_ID";
    CreativeTranscodeIdType["CREATIVE_ID"] = "CREATIVE_ID";
    CreativeTranscodeIdType["CREATIVE_ADID"] = "CREATIVE_ADID";
    CreativeTranscodeIdType["UNIVERSAL_AD_ID"] = "UNIVERSAL_AD_ID";
    CreativeTranscodeIdType["MEDIA_URI"] = "MEDIA_URI";
    CreativeTranscodeIdType["MEDIA_URI_PATH"] = "MEDIA_URI_PATH";
    CreativeTranscodeIdType["CREATIVE_ADID_WITH_FALLBACK"] = "CREATIVE_ADID_WITH_FALLBACK";
    CreativeTranscodeIdType["CANONICALIZED_MEDIA_URI"] = "CANONICALIZED_MEDIA_URI";
    CreativeTranscodeIdType["GV_REGISTRY_ID"] = "GV_REGISTRY_ID";
    CreativeTranscodeIdType["UNKNOWN_ID_TYPE"] = "UNKNOWN_ID_TYPE";
    CreativeTranscodeIdType["MEDIA_URI_HASH"] = "MEDIA_URI_HASH";
    CreativeTranscodeIdType["UNKNOWN"] = "UNKNOWN";
})(CreativeTranscodeIdType || (exports.CreativeTranscodeIdType = CreativeTranscodeIdType = {}));
