// @guardian/google-admanager-api/dist/client/services/lineItem/lineItem.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnarchiveLineItems = exports.ResumeLineItems = exports.ReserveLineItems = exports.ReleaseLineItems = exports.PauseLineItems = exports.DeleteLineItems = exports.ArchiveLineItems = exports.ActivateLineItems = exports.LineItemAction = void 0;
/**
 * Represents the actions that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class LineItemAction {
    buildAttributes() {
        return {};
    }
}
exports.LineItemAction = LineItemAction;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class ActivateLineItems {
    buildAttributes() {
        return {};
    }
}
exports.ActivateLineItems = ActivateLineItems;
/**
 * The action used for archiving {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class ArchiveLineItems {
    buildAttributes() {
        return {};
    }
}
exports.ArchiveLineItems = ArchiveLineItems;
/**
 * The action used for deleting {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 * A line item can be deleted if it has never been eligible to serve.
 * Note: deleted line items will still count against your network limits.
 * For more information, see the {@link https://support.google.com/admanager/answer/1628457 Help Center}.
 */
class DeleteLineItems {
    buildAttributes() {
        return {};
    }
}
exports.DeleteLineItems = DeleteLineItems;
/**
 * The action used for pausing {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class PauseLineItems {
    buildAttributes() {
        return {};
    }
}
exports.PauseLineItems = PauseLineItems;
/**
 * The action used for releasing {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class ReleaseLineItems {
    buildAttributes() {
        return {};
    }
}
exports.ReleaseLineItems = ReleaseLineItems;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class ReserveLineItems {
    constructor() {
        this.skipInventoryCheckField = false;
    }
    /**
     * Indicates whether the inventory check should be skipped when performing this action. The default value is false.
     */
    get skipInventoryCheck() {
        return this.skipInventoryCheckField;
    }
    /**
     * Indicates whether the inventory check should be skipped when performing this action. The default value is false.
     */
    set skipInventoryCheck(value) {
        this.skipInventoryCheckField = value;
    }
    buildAttributes() {
        return this.skipInventoryCheck
            ? { skipInventoryCheck: true }
            : { skipInventoryCheck: false };
    }
}
exports.ReserveLineItems = ReserveLineItems;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class ResumeLineItems {
    constructor() {
        this.skipInventoryCheckField = false;
    }
    /**
     * Indicates whether the inventory check should be skipped when performing this action. The default value is false.
     */
    get skipInventoryCheck() {
        return this.skipInventoryCheckField;
    }
    /**
     * Indicates whether the inventory check should be skipped when performing this action. The default value is false.
     */
    set skipInventoryCheck(value) {
        this.skipInventoryCheckField = value;
    }
    buildAttributes() {
        return this.skipInventoryCheck
            ? { skipInventoryCheck: true }
            : { skipInventoryCheck: false };
    }
}
exports.ResumeLineItems = ResumeLineItems;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} objects.
 */
class UnarchiveLineItems {
    buildAttributes() {
        return {};
    }
}
exports.UnarchiveLineItems = UnarchiveLineItems;
