// @guardian/google-admanager-api/dist/client/services/creative/creative.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VastRedirectType = exports.VideoDeliveryType = exports.MimeType = exports.ScalableType = exports.AdIdType = exports.ImageDensity = exports.DestinationUrlType = exports.LockedOrientation = exports.SslManualOverride = exports.SslScanResult = exports.RichMediaStudioChildAssetPropertyType = exports.RichMediaStudioCreativeBillingAttribute = exports.RichMediaStudioCreativeArtworkType = exports.RichMediaStudioCreativeFormat = exports.CreativePolicyViolation = void 0;
/**
 * Represents the different types of policy violations that may be detected on a given creative.
 *
 * For more information about the various types of policy violations, see {@link https://support.google.com/adspolicy/answer/6008942 here}.
 */
var CreativePolicyViolation;
(function (CreativePolicyViolation) {
    /**
     * Malware was found in the creative.
     *
     * For more information see here.
     */
    CreativePolicyViolation["MALWARE_IN_CREATIVE"] = "MALWARE_IN_CREATIVE";
    /**
     * Malware was found in the landing page.
     *
     * For more information see here.
     */
    CreativePolicyViolation["MALWARE_IN_LANDING_PAGE"] = "MALWARE_IN_LANDING_PAGE";
    /**
     * The redirect url contains legally objectionable content.
     */
    CreativePolicyViolation["LEGALLY_BLOCKED_REDIRECT_URL"] = "LEGALLY_BLOCKED_REDIRECT_URL";
    /**
     * The creative misrepresents the product or service being advertised.
     *
     * For more information see here.
     */
    CreativePolicyViolation["MISREPRESENTATION_OF_PRODUCT"] = "MISREPRESENTATION_OF_PRODUCT";
    /**
     * The creative has been determined to be self clicking.
     */
    CreativePolicyViolation["SELF_CLICKING_CREATIVE"] = "SELF_CLICKING_CREATIVE";
    /**
     * The creative has been determined as attempting to game the Google network.
     *
     * For more information see here.
     */
    CreativePolicyViolation["GAMING_GOOGLE_NETWORK"] = "GAMING_GOOGLE_NETWORK";
    /**
     * The landing page for the creative uses a dynamic DNS.
     *
     * For more information see here.
     */
    CreativePolicyViolation["DYNAMIC_DNS"] = "DYNAMIC_DNS";
    /**
     * The creative has been determined as attempting to circumvent Google advertising systems.
     */
    CreativePolicyViolation["CIRCUMVENTING_SYSTEMS"] = "CIRCUMVENTING_SYSTEMS";
    /**
     * Phishing found in creative or landing page.
     *
     * For more information see here.
     */
    CreativePolicyViolation["PHISHING"] = "PHISHING";
    /**
     * The creative prompts the user to download a file.
     *
     * For more information see here
     */
    CreativePolicyViolation["DOWNLOAD_PROMPT_IN_CREATIVE"] = "DOWNLOAD_PROMPT_IN_CREATIVE";
    /**
     * The creative sets an unauthorized cookie on a Google domain.
     *
     * For more information see here
     */
    CreativePolicyViolation["UNAUTHORIZED_COOKIE_DETECTED"] = "UNAUTHORIZED_COOKIE_DETECTED";
    /**
     * The creative has been temporarily paused while we investigate.
     */
    CreativePolicyViolation["TEMPORARY_PAUSE_FOR_VENDOR_INVESTIGATION"] = "TEMPORARY_PAUSE_FOR_VENDOR_INVESTIGATION";
    /**
     * The landing page contains an abusive experience.
     *
     * For more information see here.
     */
    CreativePolicyViolation["ABUSIVE_EXPERIENCE"] = "ABUSIVE_EXPERIENCE";
    /**
     * The creative is designed to mislead or trick the user into interacting with it.
     *
     * For more information see here.
     */
    CreativePolicyViolation["TRICK_TO_CLICK"] = "TRICK_TO_CLICK";
    /**
     * Non-allowlisted OMID verification script.
     *
     * For more information see here.
     */
    CreativePolicyViolation["USE_OF_NON_ALLOWLISTED_OMID_VERIFICATION_SCRIPT"] = "USE_OF_NON_ALLOWLISTED_OMID_VERIFICATION_SCRIPT";
    /**
     * OMID sdk injected by creative. < p>For more information see here.
     */
    CreativePolicyViolation["MISUSE_OF_OMID_API"] = "MISUSE_OF_OMID_API";
    /**
     * Unacceptable HTML5 ad.
     *
     * For more information see here.
     */
    CreativePolicyViolation["UNACCEPTABLE_HTML_AD"] = "UNACCEPTABLE_HTML_AD";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CreativePolicyViolation["UNKNOWN"] = "UNKNOWN";
})(CreativePolicyViolation || (exports.CreativePolicyViolation = CreativePolicyViolation = {}));
/**
 * Different creative format supported by Rich Media Studio creative.
 */
var RichMediaStudioCreativeFormat;
(function (RichMediaStudioCreativeFormat) {
    /**
     * In-page creatives are served into an ad slot on publishers page. In-page implies that they maintain a static size, e.g, 468x60 and do not break out of these dimensions.
     */
    RichMediaStudioCreativeFormat["IN_PAGE"] = "IN_PAGE";
    /**
     * Expanding creatives expand/collapse on user interaction such as mouse over. It consists of an initial, or collapsed and an expanded creative area.
     */
    RichMediaStudioCreativeFormat["EXPANDING"] = "EXPANDING";
    /**
     * Creatives that are served in an instant messenger application such as AOL Instant Messanger or Yahoo! Messenger. This can also be used in desktop applications such as weatherbug.
     */
    RichMediaStudioCreativeFormat["IM_EXPANDING"] = "IM_EXPANDING";
    /**
     * Floating creatives float on top of publishers page and can be closed with a close button.
     */
    RichMediaStudioCreativeFormat["FLOATING"] = "FLOATING";
    /**
     * Peel-down creatives show a glimpse of your ad in the corner of a web page. When the user interacts, the rest of the ad peels down to reveal the full message.
     */
    RichMediaStudioCreativeFormat["PEEL_DOWN"] = "PEEL_DOWN";
    /**
     * An In-Page with Floating creative is a dual-asset creative consisting of an in-page asset and a floating asset. This creative type lets you deliver a static primary ad to a webpage, while inviting a user to find out more through a floating asset delivered when the user interacts with the creative.
     */
    RichMediaStudioCreativeFormat["IN_PAGE_WITH_FLOATING"] = "IN_PAGE_WITH_FLOATING";
    /**
     * A Flash ad that renders in a Flash environment. The adserver will serve this using VAST, but it is not a proper VAST XML ad. It's an amalgamation of the proprietary InStream protocol, rendered inside VAST so that we can capture some standard behavior such as companions.
     */
    RichMediaStudioCreativeFormat["FLASH_IN_FLASH"] = "FLASH_IN_FLASH";
    /**
     * An expanding flash ad that renders in a Flash environment. The adserver will serve this using VAST, but it is not a proper VAST XML ad. It's an amalgamation of the proprietary InStream protocol, rendered inside VAST so that we can capture some standard behavior such as companions.
     */
    RichMediaStudioCreativeFormat["FLASH_IN_FLASH_EXPANDING"] = "FLASH_IN_FLASH_EXPANDING";
    /**
     * In-app creatives are served into an ad slot within a publisher's app. In-app implies that they maintain a static size, e.g, 468x60 and do not break out of these dimensions.
     */
    RichMediaStudioCreativeFormat["IN_APP"] = "IN_APP";
    /**
     * The creative format is unknown or not supported in the API version in use.
     */
    RichMediaStudioCreativeFormat["UNKNOWN"] = "UNKNOWN";
})(RichMediaStudioCreativeFormat || (exports.RichMediaStudioCreativeFormat = RichMediaStudioCreativeFormat = {}));
/**
 * Rich Media Studio creative artwork types.
 */
var RichMediaStudioCreativeArtworkType;
(function (RichMediaStudioCreativeArtworkType) {
    /**
     * The creative is a Flash creative.
     */
    RichMediaStudioCreativeArtworkType["FLASH"] = "FLASH";
    /**
     * The creative is HTML5.
     */
    RichMediaStudioCreativeArtworkType["HTML5"] = "HTML5";
    /**
     * The creative is Flash if available, and HTML5 otherwise.
     */
    RichMediaStudioCreativeArtworkType["MIXED"] = "MIXED";
})(RichMediaStudioCreativeArtworkType || (exports.RichMediaStudioCreativeArtworkType = RichMediaStudioCreativeArtworkType = {}));
/**
 * Rich Media Studio creative supported billing attributes.
 *
 * This is determined by Rich Media Studio based on the content of the creative and is not updateable.
 */
var RichMediaStudioCreativeBillingAttribute;
(function (RichMediaStudioCreativeBillingAttribute) {
    /**
     * Applies to any RichMediaStudioCreativeFormat.IN_PAGE, without Video.
     */
    RichMediaStudioCreativeBillingAttribute["IN_PAGE"] = "IN_PAGE";
    /**
     * Applies to any of these following RichMediaStudioCreativeFormat, without Video: RichMediaStudioCreativeFormat.EXPANDING, RichMediaStudioCreativeFormat.IM_EXPANDING, RichMediaStudioCreativeFormat.FLOATING, RichMediaStudioCreativeFormat.PEEL_DOWN, RichMediaStudioCreativeFormat.IN_PAGE_WITH_FLOATING
     */
    RichMediaStudioCreativeBillingAttribute["FLOATING_EXPANDING"] = "FLOATING_EXPANDING";
    /**
     * Applies to any creatives that includes a video.
     */
    RichMediaStudioCreativeBillingAttribute["VIDEO"] = "VIDEO";
    /**
     * Applies to any RichMediaStudioCreativeFormat.FLASH_IN_FLASH, without Video.
     */
    RichMediaStudioCreativeBillingAttribute["FLASH_IN_FLASH"] = "FLASH_IN_FLASH";
})(RichMediaStudioCreativeBillingAttribute || (exports.RichMediaStudioCreativeBillingAttribute = RichMediaStudioCreativeBillingAttribute = {}));
/**
 * Type of RichMediaStudioChildAssetProperty
 */
var RichMediaStudioChildAssetPropertyType;
(function (RichMediaStudioChildAssetPropertyType) {
    /**
     * SWF files
     */
    RichMediaStudioChildAssetPropertyType["FLASH"] = "FLASH";
    /**
     * FLVS and any other video file types
     */
    RichMediaStudioChildAssetPropertyType["VIDEO"] = "VIDEO";
    /**
     * Image files
     */
    RichMediaStudioChildAssetPropertyType["IMAGE"] = "IMAGE";
    /**
     * The rest of the supported file types .txt, .xml, etc.
     */
    RichMediaStudioChildAssetPropertyType["DATA"] = "DATA";
})(RichMediaStudioChildAssetPropertyType || (exports.RichMediaStudioChildAssetPropertyType = RichMediaStudioChildAssetPropertyType = {}));
/**
 * Enum to store the creative SSL compatibility scan result.
 */
var SslScanResult;
(function (SslScanResult) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    SslScanResult["UNKNOWN"] = "UNKNOWN";
    SslScanResult["UNSCANNED"] = "UNSCANNED";
    SslScanResult["SCANNED_SSL"] = "SCANNED_SSL";
    SslScanResult["SCANNED_NON_SSL"] = "SCANNED_NON_SSL";
})(SslScanResult || (exports.SslScanResult = SslScanResult = {}));
/**
 * Enum to store the creative SSL compatibility manual override. Its three states are similar to that of SslScanResult.
 */
var SslManualOverride;
(function (SslManualOverride) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    SslManualOverride["UNKNOWN"] = "UNKNOWN";
    SslManualOverride["NO_OVERRIDE"] = "NO_OVERRIDE";
    SslManualOverride["SSL_COMPATIBLE"] = "SSL_COMPATIBLE";
    SslManualOverride["NOT_SSL_COMPATIBLE"] = "NOT_SSL_COMPATIBLE";
})(SslManualOverride || (exports.SslManualOverride = SslManualOverride = {}));
/**
 * Describes the orientation that a creative should be served with.
 */
var LockedOrientation;
(function (LockedOrientation) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    LockedOrientation["UNKNOWN"] = "UNKNOWN";
    LockedOrientation["FREE_ORIENTATION"] = "FREE_ORIENTATION";
    LockedOrientation["PORTRAIT_ONLY"] = "PORTRAIT_ONLY";
    LockedOrientation["LANDSCAPE_ONLY"] = "LANDSCAPE_ONLY";
})(LockedOrientation || (exports.LockedOrientation = LockedOrientation = {}));
/**
 * The valid actions that a destination URL may perform if the user clicks on the ad.
 */
var DestinationUrlType;
(function (DestinationUrlType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    DestinationUrlType["UNKNOWN"] = "UNKNOWN";
    /**
     * Navigate to a web page. (a.k.a. "Click-through URL").
     */
    DestinationUrlType["CLICK_TO_WEB"] = "CLICK_TO_WEB";
    /**
     * Start an application.
     */
    DestinationUrlType["CLICK_TO_APP"] = "CLICK_TO_APP";
    /**
     * Make a phone call.
     */
    DestinationUrlType["CLICK_TO_CALL"] = "CLICK_TO_CALL";
    /**
     * Destination URL not present. Useful for video creatives where a landing page or a product isn't necessarily applicable.
     */
    DestinationUrlType["NONE"] = "NONE";
})(DestinationUrlType || (exports.DestinationUrlType = DestinationUrlType = {}));
/**
 * Image densities.
 */
var ImageDensity;
(function (ImageDensity) {
    /**
     * Indicates that there is a 1:1 ratio between the dimensions of the raw image and the dimensions that it should be displayed at in device-independent pixels.
     */
    ImageDensity["ONE_TO_ONE"] = "ONE_TO_ONE";
    /**
     * Indicates that there is a 3:2 ratio between the dimensions of the raw image and the dimensions that it should be displayed at in device-independent pixels.
     */
    ImageDensity["THREE_TO_TWO"] = "THREE_TO_TWO";
    /**
     * Indicates that there is a 2:1 ratio between the dimensions of the raw image and the dimensions that it should be displayed at in device-independent pixels.
     */
    ImageDensity["TWO_TO_ONE"] = "TWO_TO_ONE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ImageDensity["UNKNOWN"] = "UNKNOWN";
})(ImageDensity || (exports.ImageDensity = ImageDensity = {}));
/**
 * The registry that an ad ID belongs to.
 */
var AdIdType;
(function (AdIdType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AdIdType["UNKNOWN"] = "UNKNOWN";
    /**
     * The ad ID is registered with ad-id.org.
     */
    AdIdType["AD_ID"] = "AD_ID";
    /**
     * The ad ID is registered with clearcast.co.uk.
     */
    AdIdType["CLEARCAST"] = "CLEARCAST";
    /**
     * The creative does not have an ad ID outside of Ad Manager.
     */
    AdIdType["NONE"] = "NONE";
})(AdIdType || (exports.AdIdType = AdIdType = {}));
/**
 * The different ways a video/flash can scale.
 */
var ScalableType;
(function (ScalableType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ScalableType["UNKNOWN"] = "UNKNOWN";
    /**
     * The creative should not be scaled.
     */
    ScalableType["NOT_SCALABLE"] = "NOT_SCALABLE";
    /**
     * The creative can be scaled and its aspect-ratio must be maintained.
     */
    ScalableType["RATIO_SCALABLE"] = "RATIO_SCALABLE";
    /**
     * The creative can be scaled and its aspect-ratio can be distorted.
     */
    ScalableType["STRETCH_SCALABLE"] = "STRETCH_SCALABLE";
})(ScalableType || (exports.ScalableType = ScalableType = {}));
/**
 *
 */
var MimeType;
(function (MimeType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    MimeType["UNKNOWN"] = "UNKNOWN";
    /**
     * application/x-asp
     */
    MimeType["ASP"] = "ASP";
    /**
     * audio/aiff
     */
    MimeType["AUDIO_AIFF"] = "AUDIO_AIFF";
    /**
     * audio/basic
     */
    MimeType["AUDIO_BASIC"] = "AUDIO_BASIC";
    /**
     * audio/flac
     */
    MimeType["AUDIO_FLAC"] = "AUDIO_FLAC";
    /**
     * audio/mid
     */
    MimeType["AUDIO_MID"] = "AUDIO_MID";
    /**
     * audio/mpeg
     */
    MimeType["AUDIO_MP3"] = "AUDIO_MP3";
    /**
     * audio/mp4
     */
    MimeType["AUDIO_MP4"] = "AUDIO_MP4";
    /**
     * audio/x-mpegurl
     */
    MimeType["AUDIO_MPEG_URL"] = "AUDIO_MPEG_URL";
    /**
     * audio/x-ms-wma
     */
    MimeType["AUDIO_MS_WMA"] = "AUDIO_MS_WMA";
    /**
     * audio/ogg
     */
    MimeType["AUDIO_OGG"] = "AUDIO_OGG";
    /**
     * audio/x-pn-realaudio-plugin
     */
    MimeType["AUDIO_REAL_AUDIO_PLUGIN"] = "AUDIO_REAL_AUDIO_PLUGIN";
    /**
     * audio/x-wav
     */
    MimeType["AUDIO_WAV"] = "AUDIO_WAV";
    /**
     * application/binary
     */
    MimeType["BINARY"] = "BINARY";
    /**
     * application/dash+xml
     */
    MimeType["DASH"] = "DASH";
    /**
     * application/x-director
     */
    MimeType["DIRECTOR"] = "DIRECTOR";
    /**
     * application/x-shockwave-flash
     */
    MimeType["FLASH"] = "FLASH";
    /**
     * application/graphicconverter
     */
    MimeType["GRAPHIC_CONVERTER"] = "GRAPHIC_CONVERTER";
    /**
     * application/x-javascript
     */
    MimeType["JAVASCRIPT"] = "JAVASCRIPT";
    /**
     * application/json
     */
    MimeType["JSON"] = "JSON";
    /**
     * image/x-win-bitmap
     */
    MimeType["IMAGE_BITMAP"] = "IMAGE_BITMAP";
    /**
     * image/bmp
     */
    MimeType["IMAGE_BMP"] = "IMAGE_BMP";
    /**
     * image/gif
     */
    MimeType["IMAGE_GIF"] = "IMAGE_GIF";
    /**
     * image/jpeg
     */
    MimeType["IMAGE_JPEG"] = "IMAGE_JPEG";
    /**
     * image/photoshop
     */
    MimeType["IMAGE_PHOTOSHOP"] = "IMAGE_PHOTOSHOP";
    /**
     * image/png
     */
    MimeType["IMAGE_PNG"] = "IMAGE_PNG";
    /**
     * image/tiff
     */
    MimeType["IMAGE_TIFF"] = "IMAGE_TIFF";
    /**
     * image/vnd.wap.wbmp
     */
    MimeType["IMAGE_WBMP"] = "IMAGE_WBMP";
    /**
     * application/x-mpegURL
     */
    MimeType["M3U8"] = "M3U8";
    /**
     * application/mac-binhex40
     */
    MimeType["MAC_BIN_HEX_40"] = "MAC_BIN_HEX_40";
    /**
     * application/vnd.ms-excel
     */
    MimeType["MS_EXCEL"] = "MS_EXCEL";
    /**
     * application/ms-powerpoint
     */
    MimeType["MS_POWERPOINT"] = "MS_POWERPOINT";
    /**
     * application/msword
     */
    MimeType["MS_WORD"] = "MS_WORD";
    /**
     * application/octet-stream
     */
    MimeType["OCTET_STREAM"] = "OCTET_STREAM";
    /**
     * application/pdf
     */
    MimeType["PDF"] = "PDF";
    /**
     * application/postscript
     */
    MimeType["POSTSCRIPT"] = "POSTSCRIPT";
    /**
     * application/vnd.rn-realmedia
     */
    MimeType["RN_REAL_MEDIA"] = "RN_REAL_MEDIA";
    /**
     * message/rfc822
     */
    MimeType["RFC_822"] = "RFC_822";
    /**
     * application/rtf
     */
    MimeType["RTF"] = "RTF";
    /**
     * text/calendar
     */
    MimeType["TEXT_CALENDAR"] = "TEXT_CALENDAR";
    /**
     * text/css
     */
    MimeType["TEXT_CSS"] = "TEXT_CSS";
    /**
     * text/csv
     */
    MimeType["TEXT_CSV"] = "TEXT_CSV";
    /**
     * text/html
     */
    MimeType["TEXT_HTML"] = "TEXT_HTML";
    /**
     * text/java
     */
    MimeType["TEXT_JAVA"] = "TEXT_JAVA";
    /**
     * text/plain
     */
    MimeType["TEXT_PLAIN"] = "TEXT_PLAIN";
    /**
     * video/3gpp
     */
    MimeType["VIDEO_3GPP"] = "VIDEO_3GPP";
    /**
     * video/3gpp2
     */
    MimeType["VIDEO_3GPP2"] = "VIDEO_3GPP2";
    /**
     * video/avi
     */
    MimeType["VIDEO_AVI"] = "VIDEO_AVI";
    /**
     * video/x-flv
     */
    MimeType["VIDEO_FLV"] = "VIDEO_FLV";
    /**
     * video/mp4
     */
    MimeType["VIDEO_MP4"] = "VIDEO_MP4";
    /**
     * video/mp4v-es
     */
    MimeType["VIDEO_MP4V_ES"] = "VIDEO_MP4V_ES";
    /**
     * video/mpeg
     */
    MimeType["VIDEO_MPEG"] = "VIDEO_MPEG";
    /**
     * video/x-ms-asf
     */
    MimeType["VIDEO_MS_ASF"] = "VIDEO_MS_ASF";
    /**
     * video/x-ms-wm
     */
    MimeType["VIDEO_MS_WM"] = "VIDEO_MS_WM";
    /**
     * video/x-ms-wmv
     */
    MimeType["VIDEO_MS_WMV"] = "VIDEO_MS_WMV";
    /**
     * video/x-ms-wvx
     */
    MimeType["VIDEO_MS_WVX"] = "VIDEO_MS_WVX";
    /**
     * video/ogg
     */
    MimeType["VIDEO_OGG"] = "VIDEO_OGG";
    /**
     * video/x-quicktime
     */
    MimeType["VIDEO_QUICKTIME"] = "VIDEO_QUICKTIME";
    /**
     * video/webm
     */
    MimeType["VIDEO_WEBM"] = "VIDEO_WEBM";
    /**
     * application/xaml+xml
     */
    MimeType["XAML"] = "XAML";
    /**
     * application/xhtml+xml
     */
    MimeType["XHTML"] = "XHTML";
    /**
     * application/xml
     */
    MimeType["XML"] = "XML";
    /**
     * application/zip
     */
    MimeType["ZIP"] = "ZIP";
})(MimeType || (exports.MimeType = MimeType = {}));
/**
 * The video delivery type.
 */
var VideoDeliveryType;
(function (VideoDeliveryType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    VideoDeliveryType["UNKNOWN"] = "UNKNOWN";
    /**
     * Video will be served through a progressive download.
     */
    VideoDeliveryType["PROGRESSIVE"] = "PROGRESSIVE";
    /**
     * Video will be served via a streaming protocol like RTMP.
     */
    VideoDeliveryType["STREAMING"] = "STREAMING";
})(VideoDeliveryType || (exports.VideoDeliveryType = VideoDeliveryType = {}));
/**
 * The types of VAST ads that a VastRedirectCreative can point to.
 */
var VastRedirectType;
(function (VastRedirectType) {
    /**
     * The VAST XML contains only linear ads.
     */
    VastRedirectType["LINEAR"] = "LINEAR";
    /**
     * The VAST XML contains only nonlinear ads.
     */
    VastRedirectType["NON_LINEAR"] = "NON_LINEAR";
    /**
     * The VAST XML contains both linear and nonlinear ads.
     */
    VastRedirectType["LINEAR_AND_NON_LINEAR"] = "LINEAR_AND_NON_LINEAR";
})(VastRedirectType || (exports.VastRedirectType = VastRedirectType = {}));
