// @guardian/google-admanager-api/dist/client/services/creativeWrapper/creativeWrapper.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreativeWrapperService = void 0;
class CreativeWrapperService {
    constructor(client) {
        this._client = client;
    }
    async createCreativeWrappers(creativeWrappers) {
        return this._client.createCreativeWrappers({ creativeWrappers });
    }
    async getCreativeWrappersByStatement(filterStatement) {
        return this._client.getCreativeWrappersByStatement({
            filterStatement,
        });
    }
    async performCreativeWrapperAction(creativeWrapperAction, filterStatement) {
        return this._client.performCreativeWrapperAction({
            creativeWrapperAction: {
                attributes: {
                    "xsi:type": creativeWrapperAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateCreativeWrappers(creativeWrappers) {
        return this._client.updateCreativeWrappers({ creativeWrappers });
    }
}
exports.CreativeWrapperService = CreativeWrapperService;
