// @guardian/google-admanager-api/dist/client/services/creativeWrapper/creativeWrapper.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreativeWrapperStatus = exports.CreativeWrapperOrdering = exports.ConversionEvent = exports.CreativeWrapperType = void 0;
/**
 * The type of a creative wrapper which is specified on the CreativeWrapper.
 */
var CreativeWrapperType;
(function (CreativeWrapperType) {
    /**
     * HTML creative wrappers that include header/footer HTML snippets.
     */
    CreativeWrapperType["HTML"] = "HTML";
    /**
     * Video Tracking URL creative wrappers that include tracking URIs.
     */
    CreativeWrapperType["VIDEO_TRACKING_URL"] = "VIDEO_TRACKING_URL";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CreativeWrapperType["UNKNOWN"] = "UNKNOWN";
})(CreativeWrapperType || (exports.CreativeWrapperType = CreativeWrapperType = {}));
/**
 * All possible tracking event types. Not all events are supported by every kind of creative.
 */
var ConversionEvent;
(function (ConversionEvent) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ConversionEvent["UNKNOWN"] = "UNKNOWN";
    /**
     * Corresponds to the creativeView tracking event.
     */
    ConversionEvent["CREATIVE_VIEW"] = "CREATIVE_VIEW";
    /**
     * Corresponds to the start tracking event.
     */
    ConversionEvent["START"] = "START";
    /**
     * An event that is fired when a video skip button is shown, usually after 5 seconds of viewing the video. This event does not correspond to any VAST element and is implemented using an extension.
     */
    ConversionEvent["SKIP_SHOWN"] = "SKIP_SHOWN";
    /**
     * Corresponds to the firstQuartile tracking event.
     */
    ConversionEvent["FIRST_QUARTILE"] = "FIRST_QUARTILE";
    /**
     * Corresponds to the midpoint tracking event.
     */
    ConversionEvent["MIDPOINT"] = "MIDPOINT";
    /**
     * Corresponds to the thirdQuartile tracking event.
     */
    ConversionEvent["THIRD_QUARTILE"] = "THIRD_QUARTILE";
    /**
     * An event that is fired after 30 seconds of viewing the video or when the video finished (if the video duration is less than 30 seconds). This event does not correspond to any VAST element and is implemented using an extension.
     */
    ConversionEvent["ENGAGED_VIEW"] = "ENGAGED_VIEW";
    /**
     * Corresponds to the complete tracking event.
     */
    ConversionEvent["COMPLETE"] = "COMPLETE";
    /**
     * Corresponds to the mute tracking event.
     */
    ConversionEvent["MUTE"] = "MUTE";
    /**
     * Corresponds to the unmute tracking event.
     */
    ConversionEvent["UNMUTE"] = "UNMUTE";
    /**
     * Corresponds to the pause tracking event.
     */
    ConversionEvent["PAUSE"] = "PAUSE";
    /**
     * Corresponds to the rewind tracking event.
     */
    ConversionEvent["REWIND"] = "REWIND";
    /**
     * Corresponds to the resume tracking event.
     */
    ConversionEvent["RESUME"] = "RESUME";
    /**
     * An event that is fired when a video was skipped. This event does not correspond to any VAST element and is implemented using an extension.
     */
    ConversionEvent["SKIPPED"] = "SKIPPED";
    /**
     * Corresponds to the fullscreen tracking event.
     */
    ConversionEvent["FULLSCREEN"] = "FULLSCREEN";
    /**
     * Corresponds to the expand tracking event.
     */
    ConversionEvent["EXPAND"] = "EXPAND";
    /**
     * Corresponds to the collapse tracking event.
     */
    ConversionEvent["COLLAPSE"] = "COLLAPSE";
    /**
     * Corresponds to the acceptInvitation tracking event.
     */
    ConversionEvent["ACCEPT_INVITATION"] = "ACCEPT_INVITATION";
    /**
     * Corresponds to the close tracking event.
     */
    ConversionEvent["CLOSE"] = "CLOSE";
    /**
     * Corresponds to the Linear.VideoClicks.ClickTracking node.
     */
    ConversionEvent["CLICK_TRACKING"] = "CLICK_TRACKING";
    /**
     * Corresponds to the InLine.Survey node.
     */
    ConversionEvent["SURVEY"] = "SURVEY";
    /**
     * Corresponds to the Linear.VideoClicks.CustomClick node.
     */
    ConversionEvent["CUSTOM_CLICK"] = "CUSTOM_CLICK";
    /**
     * Corresponds to the measurableImpression tracking event.
     */
    ConversionEvent["MEASURABLE_IMPRESSION"] = "MEASURABLE_IMPRESSION";
    /**
     * Corresponds to the viewableImpression tracking event.
     */
    ConversionEvent["VIEWABLE_IMPRESSION"] = "VIEWABLE_IMPRESSION";
    /**
     * Corresponds to the abandon tracking event.
     */
    ConversionEvent["VIDEO_ABANDON"] = "VIDEO_ABANDON";
    /**
     * Corresponds to the fullyViewableAudibleHalfDurationImpression tracking event.
     */
    ConversionEvent["FULLY_VIEWABLE_AUDIBLE_HALF_DURATION_IMPRESSION"] = "FULLY_VIEWABLE_AUDIBLE_HALF_DURATION_IMPRESSION";
})(ConversionEvent || (exports.ConversionEvent = ConversionEvent = {}));
/**
 * Defines the order in which the header and footer HTML snippets will be wrapped around the served creative.
 * INNER snippets will be wrapped first, followed by NO_PREFERENCE and finally OUTER.
 * If the creative needs to be wrapped with more than one snippet with the same CreativeWrapperOrdering, then the order is unspecified.
 */
var CreativeWrapperOrdering;
(function (CreativeWrapperOrdering) {
    /**
     * Wrapping occurs after INNER but before OUTER
     */
    CreativeWrapperOrdering["NO_PREFERENCE"] = "NO_PREFERENCE";
    /**
     * Wrapping occurs as early as possible.
     */
    CreativeWrapperOrdering["INNER"] = "INNER";
    /**
     * Wrapping occurs after both NO_PREFERENCE and INNER
     */
    CreativeWrapperOrdering["OUTER"] = "OUTER";
})(CreativeWrapperOrdering || (exports.CreativeWrapperOrdering = CreativeWrapperOrdering = {}));
/**
 * Indicates whether the CreativeWrapper is active. HTML snippets are served to creatives only when the creative wrapper is active.
 */
var CreativeWrapperStatus;
(function (CreativeWrapperStatus) {
    CreativeWrapperStatus["ACTIVE"] = "ACTIVE";
    CreativeWrapperStatus["INACTIVE"] = "INACTIVE";
})(CreativeWrapperStatus || (exports.CreativeWrapperStatus = CreativeWrapperStatus = {}));
