// @guardian/google-admanager-api/dist/client/services/adRule/adRule.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdSpotFillType = exports.AdSpotTargetingType = exports.AdRuleSlotBumper = exports.MidrollFrequencyType = exports.AdRuleSlotBehavior = exports.FrequencyCapBehavior = exports.AdRuleStatus = void 0;
/**
 * Represents the status of ad rules and ad rule slots.
 */
var AdRuleStatus;
(function (AdRuleStatus) {
    /**
     * 	Created and ready to be served. Is user-visible.
     */
    AdRuleStatus["ACTIVE"] = "ACTIVE";
    /**
     * 	Paused, user-visible.
     */
    AdRuleStatus["INACTIVE"] = "INACTIVE";
    /**
     * 	Marked as deleted, not user-visible.
     */
    AdRuleStatus["DELETED"] = "DELETED";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    AdRuleStatus["UNKNOWN"] = "UNKNOWN";
})(AdRuleStatus || (exports.AdRuleStatus = AdRuleStatus = {}));
/**
 * Types of behavior for frequency caps within ad rules.
 */
var FrequencyCapBehavior;
(function (FrequencyCapBehavior) {
    /**
     * 	Turn on at least one of the frequency caps.
     */
    FrequencyCapBehavior["TURN_ON"] = "TURN_ON";
    /**
     * 	Turn off all frequency caps.
     */
    FrequencyCapBehavior["TURN_OFF"] = "TURN_OFF";
    /**
     * 	Defer frequency cap decisions to the next ad rule in priority order.
     */
    FrequencyCapBehavior["DEFER"] = "DEFER";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    FrequencyCapBehavior["UNKNOWN"] = "UNKNOWN";
})(FrequencyCapBehavior || (exports.FrequencyCapBehavior = FrequencyCapBehavior = {}));
/**
 * The types of behaviors for ads within a ad rule slot.
 */
var AdRuleSlotBehavior;
(function (AdRuleSlotBehavior) {
    /**
     * 	This ad rule always includes this slot's ads.
     */
    AdRuleSlotBehavior["ALWAYS_SHOW"] = "ALWAYS_SHOW";
    /**
     * 	This ad rule never includes this slot's ads.
     */
    AdRuleSlotBehavior["NEVER_SHOW"] = "NEVER_SHOW";
    /**
     * 	Defer to lower priority rules. This ad rule doesn't specify guidelines for this slot's ads.
     */
    AdRuleSlotBehavior["DEFER"] = "DEFER";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    AdRuleSlotBehavior["UNKNOWN"] = "UNKNOWN";
})(AdRuleSlotBehavior || (exports.AdRuleSlotBehavior = AdRuleSlotBehavior = {}));
/**
 * Frequency types for mid-roll ad rule slots.
 */
var MidrollFrequencyType;
(function (MidrollFrequencyType) {
    /**
     * 	The ad rule slot is not a mid-roll and hence MidrollFrequency should be ignored.
     */
    MidrollFrequencyType["NONE"] = "NONE";
    /**
     * 	MidrollFrequency is a time interval and mentioned as a single numeric value in seconds. For example, "100" would mean "play a mid-roll every 100 seconds".
     */
    MidrollFrequencyType["EVERY_N_SECONDS"] = "EVERY_N_SECONDS";
    /**
     * 	MidrollFrequency is a comma-delimited list of points in time (in seconds) when an ad should play. For example, "100,300" would mean "play an ad at 100 seconds and 300 seconds".
     */
    MidrollFrequencyType["FIXED_TIME"] = "FIXED_TIME";
    /**
     * 	MidrollFrequency is a cue point interval and is a single integer value, such as "5", which means "play a mid-roll every 5th cue point".
     */
    MidrollFrequencyType["EVERY_N_CUEPOINTS"] = "EVERY_N_CUEPOINTS";
    /**
     * 	Same as FIXED_TIME, except the values represent the ordinal cue points ("1,3,5", for example).
     */
    MidrollFrequencyType["FIXED_CUE_POINTS"] = "FIXED_CUE_POINTS";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    MidrollFrequencyType["UNKNOWN"] = "UNKNOWN";
})(MidrollFrequencyType || (exports.MidrollFrequencyType = MidrollFrequencyType = {}));
/**
 * Types of bumper ads on an ad rule slot.
 */
var AdRuleSlotBumper;
(function (AdRuleSlotBumper) {
    /**
     * 	Do not show a bumper ad.
     */
    AdRuleSlotBumper["NONE"] = "NONE";
    /**
     * 	Show a bumper ad before the slot's other ads.
     */
    AdRuleSlotBumper["BEFORE"] = "BEFORE";
    /**
     * 	Show a bumper ad after the slot's other ads.
     */
    AdRuleSlotBumper["AFTER"] = "AFTER";
    /**
     * 	Show a bumper before and after the slot's other ads.
     */
    AdRuleSlotBumper["BEFORE_AND_AFTER"] = "BEFORE_AND_AFTER";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    AdRuleSlotBumper["UNKNOWN"] = "UNKNOWN";
})(AdRuleSlotBumper || (exports.AdRuleSlotBumper = AdRuleSlotBumper = {}));
/**
 * Defines the targeting behavior of an ad spot.
 */
var AdSpotTargetingType;
(function (AdSpotTargetingType) {
    /**
     * 	Line items not targeting this ad spot explicitly may serve in it.
     */
    AdSpotTargetingType["NOT_REQUIRED"] = "NOT_REQUIRED";
    /**
     * 	Only line items targeting this ad spots explicitly may serve in it
     */
    AdSpotTargetingType["EXPLICITLY_TARGETED"] = "EXPLICITLY_TARGETED";
    /**
     * 	If house ads are an allowed reservation type, they may serve in the ad spot regardless of whether they explicitly target it. Ads of other reservation types (whose type is allowed in the ad spot), may serve in the ad spot only if explicitly targeted.
     */
    AdSpotTargetingType["EXPLICITLY_TARGETED_EXCEPT_HOUSE"] = "EXPLICITLY_TARGETED_EXCEPT_HOUSE";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    AdSpotTargetingType["UNKNOWN"] = "UNKNOWN";
})(AdSpotTargetingType || (exports.AdSpotTargetingType = AdSpotTargetingType = {}));
/**
 * The different options for how ad spots are filled. Only some allocations of ads to subpods produce a valid final pod.
 */
var AdSpotFillType;
(function (AdSpotFillType) {
    /**
     * 	If this ad spot is empty, the overall pod is invalid.
     */
    AdSpotFillType["REQUIRED"] = "REQUIRED";
    /**
     * 	The ad spot is always "satisfied", whether empty or nonempty.
     */
    AdSpotFillType["OPTIONAL"] = "OPTIONAL";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    AdSpotFillType["UNKNOWN"] = "UNKNOWN";
})(AdSpotFillType || (exports.AdSpotFillType = AdSpotFillType = {}));
