// @guardian/google-admanager-api/dist/client/services/adRule/adRule.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdRuleService = void 0;
class AdRuleService {
    constructor(client) {
        this._client = client;
    }
    createAdRules(adRules) {
        return this._client.createAdRules({ adRules });
    }
    createAdSpots(adSports) {
        return this._client.createAdSpots({ adSports });
    }
    createBreakTemplates(breakTemplate) {
        return this._client.createBreakTemplates({ breakTemplate });
    }
    getAdRulesByStatement(statement) {
        return this._client.getAdRulesByStatement({
            statement,
        });
    }
    getAdSpotsByStatement(filterStatement) {
        return this._client.getAdSpotsByStatement({
            filterStatement,
        });
    }
    getBreakTemplatesByStatement(filterStatement) {
        return this._client.getBreakTemplatesByStatement({
            filterStatement,
        });
    }
    performAdRuleAction(adRuleAction, filterStatement) {
        return this._client.performAdRuleAction({
            adRuleAction: {
                attributes: {
                    "xsi:type": adRuleAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    updateAdRules(adRules) {
        return this._client.updateAdRules({ adRules });
    }
    updateAdSpots(adSports) {
        return this._client.updateAdSpots({ adSports });
    }
    updateBreakTemplates(breakTemplate) {
        return this._client.updateBreakTemplates({ breakTemplate });
    }
}
exports.AdRuleService = AdRuleService;
