// @guardian/google-admanager-api/dist/client/services/user/user.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserService = void 0;
class UserService {
    constructor(client) {
        this._client = client;
    }
    async createUsers(users) {
        return this._client.createUsers({ users });
    }
    async getAllRoles() {
        return this._client.getAllRolesAsync();
    }
    async getCurrentUser() {
        return this._client.getCurrentUser();
    }
    async getUsersByStatement(filterStatement) {
        return this._client.getUsersByStatement({
            filterStatement,
        });
    }
    async performUserAction(userAction, filterStatement) {
        return this._client.performUserAction({
            userAction: {
                attributes: {
                    "xsi:type": userAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateUsers(users) {
        return this._client.updateUsers({ users });
    }
}
exports.UserService = UserService;
