// @guardian/google-admanager-api/dist/client/services/audienceSegment/audienceSegment.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseType = exports.AudienceSegmentApprovalStatus = exports.AudienceSegmentType = exports.AudienceSegmentStatus = void 0;
/**
 * Specifies the statuses for {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.AudienceSegment AudienceSegment} objects.
 */
var AudienceSegmentStatus;
(function (AudienceSegmentStatus) {
    /**
     * Active status means this audience segment is available for targeting.
     */
    AudienceSegmentStatus["ACTIVE"] = "ACTIVE";
    /**
     * Inactive status means this audience segment is not available for targeting.
     */
    AudienceSegmentStatus["INACTIVE"] = "INACTIVE";
})(AudienceSegmentStatus || (exports.AudienceSegmentStatus = AudienceSegmentStatus = {}));
/**
 * Specifies types for {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.AudienceSegment AudienceSegment} objects.
 */
var AudienceSegmentType;
(function (AudienceSegmentType) {
    /**
     * First party segments created and owned by the publisher.
     */
    AudienceSegmentType["FIRST_PARTY"] = "FIRST_PARTY";
    /**
     * First party segments shared by other clients.
     */
    AudienceSegmentType["SHARED"] = "SHARED";
    /**
     * Third party segments licensed by the publisher from data providers. This doesn't include Google-provided licensed segments.
     */
    AudienceSegmentType["THIRD_PARTY"] = "THIRD_PARTY";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AudienceSegmentType["UNKNOWN"] = "UNKNOWN";
})(AudienceSegmentType || (exports.AudienceSegmentType = AudienceSegmentType = {}));
/**
 * Approval status values for {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.ThirdPartyAudienceSegment ThirdPartyAudienceSegment} objects.
 */
var AudienceSegmentApprovalStatus;
(function (AudienceSegmentApprovalStatus) {
    /**
     * Specifies that this segment is waiting to be approved or rejected. It cannot be targeted.
     */
    AudienceSegmentApprovalStatus["UNAPPROVED"] = "UNAPPROVED";
    /**
     * Specifies that this segment is approved and can be targeted.
     */
    AudienceSegmentApprovalStatus["APPROVED"] = "APPROVED";
    /**
     * Specifies that this segment is rejected and cannot be targeted.
     */
    AudienceSegmentApprovalStatus["REJECTED"] = "REJECTED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AudienceSegmentApprovalStatus["UNKNOWN"] = "UNKNOWN";
})(AudienceSegmentApprovalStatus || (exports.AudienceSegmentApprovalStatus = AudienceSegmentApprovalStatus = {}));
/**
 * Specifies the license type of a {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.ThirdPartyAudienceSegment ThirdPartyAudienceSegment}.
 */
var LicenseType;
(function (LicenseType) {
    /**
     * A direct license is the result of a direct contract between the data provider and the publisher.
     */
    LicenseType["DIRECT_LICENSE"] = "DIRECT_LICENSE";
    /**
     * A global license is the result of an agreement between Google and the data provider, which agrees to license their audience segments to all the publishers and/or advertisers of the Google ecosystem.
     */
    LicenseType["GLOBAL_LICENSE"] = "GLOBAL_LICENSE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    LicenseType["UNKNOWN"] = "UNKNOWN";
})(LicenseType || (exports.LicenseType = LicenseType = {}));
