// @guardian/google-admanager-api/dist/client/services/audienceSegment/audienceSegment.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RejectAudienceSegments = exports.PopulateAudienceSegments = exports.DeactivateAudienceSegments = exports.ApproveAudienceSegments = exports.ActivateAudienceSegments = exports.AudienceSegmentAction = void 0;
/**
 * Action that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.AudienceSegment AudienceSegment} objects.
 */
class AudienceSegmentAction {
}
exports.AudienceSegmentAction = AudienceSegmentAction;
/**
 * Action that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.FirstPartyAudienceSegment FirstPartyAudienceSegment} objects to activate them.
 */
class ActivateAudienceSegments {
}
exports.ActivateAudienceSegments = ActivateAudienceSegments;
/**
 * Action that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.ThirdPartyAudienceSegment ThirdPartyAudienceSegment} objects to approve them.
 */
class ApproveAudienceSegments {
}
exports.ApproveAudienceSegments = ApproveAudienceSegments;
/**
 * Action that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.FirstPartyAudienceSegment FirstPartyAudienceSegment} objects to deactivate them.
 */
class DeactivateAudienceSegments {
}
exports.DeactivateAudienceSegments = DeactivateAudienceSegments;
/**
 * Action that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.FirstPartyAudienceSegment FirstPartyAudienceSegment} objects to populate them based on last 30 days of traffic.
 */
class PopulateAudienceSegments {
}
exports.PopulateAudienceSegments = PopulateAudienceSegments;
/**
 * Action that can be performed on {@link https://developers.google.com/ad-manager/api/reference/v202505/AudienceSegmentService.ThirdPartyAudienceSegment ThirdPartyAudienceSegment} objects to reject them.
 */
class RejectAudienceSegments {
}
exports.RejectAudienceSegments = RejectAudienceSegments;
