// @guardian/google-admanager-api/dist/common/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpUtilities = exports.StatementBuilder = exports.DateTimeUtilities = void 0;
exports.promiseFromCallback = promiseFromCallback;
var dateTimeUtilities_util_1 = require("./dateTimeUtilities.util");
Object.defineProperty(exports, "DateTimeUtilities", { enumerable: true, get: function () { return dateTimeUtilities_util_1.DateTimeUtilities; } });
var statementBuilder_util_1 = require("./statementBuilder.util");
Object.defineProperty(exports, "StatementBuilder", { enumerable: true, get: function () { return statementBuilder_util_1.StatementBuilder; } });
var httpUtilities_util_1 = require("./httpUtilities.util");
Object.defineProperty(exports, "HttpUtilities", { enumerable: true, get: function () { return httpUtilities_util_1.HttpUtilities; } });
const isClientError = (err) => err.message.startsWith("soap:Client:");
const isServerError = (err) => err.message.startsWith("soap:Server:");
function promiseFromCallback(fn) {
    return new Promise((resolve, reject) => {
        fn((err, result) => {
            var _a, _b, _c, _d;
            if (err) {
                const error = err.message.split(": ");
                if (isClientError(err)) {
                    reject({
                        code: err["response"].status,
                        name: error[1],
                        message: error[3],
                    });
                }
                if (isServerError(err)) {
                    reject({
                        code: err["response"].status,
                        name: (_c = (_b = (_a = error[2]) === null || _a === void 0 ? void 0 : _a.match(/"xsi:type":"(.*)"},/)) === null || _b === void 0 ? void 0 : _b[1]) !== null && _c !== void 0 ? _c : "",
                        message: error[1],
                        details: JSON.parse((_d = error[2]) !== null && _d !== void 0 ? _d : "{}"),
                    });
                }
                reject(err);
                return;
            }
            resolve(result);
        });
    });
}
