// @guardian/google-admanager-api/dist/client/services/inventory/adUnit.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmartSizeMode = exports.ValueSourceType = exports.AdSenseSettingsFontSize = exports.AdSenseSettingsFontFamily = exports.AdSenseSettingsBorderStyle = exports.AdSenseSettingsAdType = exports.InventoryStatus = exports.AdUnitTargetWindow = void 0;
/**
 * Corresponds to an HTML link's target attribute.
 */
var AdUnitTargetWindow;
(function (AdUnitTargetWindow) {
    /**
     * Specifies that the link should open in the full body of the page.
     */
    AdUnitTargetWindow["TOP"] = "TOP";
    /**
     * Specifies that the link should open in a new window.
     */
    AdUnitTargetWindow["BLANK"] = "BLANK";
})(AdUnitTargetWindow || (exports.AdUnitTargetWindow = AdUnitTargetWindow = {}));
/**
 * Represents the status of objects that represent inventory - ad units and placements.
 */
var InventoryStatus;
(function (InventoryStatus) {
    /**
     * The object is active.
     */
    InventoryStatus["ACTIVE"] = "ACTIVE";
    /**
     * The object is no longer active.
     */
    InventoryStatus["INACTIVE"] = "INACTIVE";
    /**
     * The object has been archived.
     */
    InventoryStatus["ARCHIVED"] = "ARCHIVED";
})(InventoryStatus || (exports.InventoryStatus = InventoryStatus = {}));
/**
 * Specifies the type of ads that can be served through this {@link https://developers.google.com/ad-manager/api/reference/v202505/InventoryService.AdUnit AdUnit}.
 */
var AdSenseSettingsAdType;
(function (AdSenseSettingsAdType) {
    /**
     * Allows text-only ads.
     */
    AdSenseSettingsAdType["TEXT"] = "TEXT";
    /**
     * Allows image-only ads.
     */
    AdSenseSettingsAdType["IMAGE"] = "IMAGE";
    /**
     * Allows both text and image ads.
     */
    AdSenseSettingsAdType["TEXT_AND_IMAGE"] = "TEXT_AND_IMAGE";
})(AdSenseSettingsAdType || (exports.AdSenseSettingsAdType = AdSenseSettingsAdType = {}));
/**
 * Describes the border of the HTML elements used to surround an ad displayed by the {@link https://developers.google.com/ad-manager/api/reference/v202505/InventoryService.AdUnit AdUnit}.
 */
var AdSenseSettingsBorderStyle;
(function (AdSenseSettingsBorderStyle) {
    /**
     * Uses the default border-style of the browser.
     */
    AdSenseSettingsBorderStyle["DEFAULT"] = "DEFAULT";
    /**
     * Uses a cornered border-style.
     */
    AdSenseSettingsBorderStyle["NOT_ROUNDED"] = "NOT_ROUNDED";
    /**
     * Uses a slightly rounded border-style.
     */
    AdSenseSettingsBorderStyle["SLIGHTLY_ROUNDED"] = "SLIGHTLY_ROUNDED";
    /**
     * Uses a rounded border-style.
     */
    AdSenseSettingsBorderStyle["VERY_ROUNDED"] = "VERY_ROUNDED";
})(AdSenseSettingsBorderStyle || (exports.AdSenseSettingsBorderStyle = AdSenseSettingsBorderStyle = {}));
/**
 * List of all possible font families.
 */
var AdSenseSettingsFontFamily;
(function (AdSenseSettingsFontFamily) {
    AdSenseSettingsFontFamily["DEFAULT"] = "DEFAULT";
    AdSenseSettingsFontFamily["ARIAL"] = "ARIAL";
    AdSenseSettingsFontFamily["TAHOMA"] = "TAHOMA";
    AdSenseSettingsFontFamily["GEORGIA"] = "GEORGIA";
    AdSenseSettingsFontFamily["TIMES"] = "TIMES";
    AdSenseSettingsFontFamily["VERDANA"] = "VERDANA";
})(AdSenseSettingsFontFamily || (exports.AdSenseSettingsFontFamily = AdSenseSettingsFontFamily = {}));
/**
 * List of all possible font sizes the user can choose.
 */
var AdSenseSettingsFontSize;
(function (AdSenseSettingsFontSize) {
    AdSenseSettingsFontSize["DEFAULT"] = "DEFAULT";
    AdSenseSettingsFontSize["SMALL"] = "SMALL";
    AdSenseSettingsFontSize["MEDIUM"] = "MEDIUM";
    AdSenseSettingsFontSize["LARGE"] = "LARGE";
})(AdSenseSettingsFontSize || (exports.AdSenseSettingsFontSize = AdSenseSettingsFontSize = {}));
/**
 * Identifies the source of a field's value.
 */
var ValueSourceType;
(function (ValueSourceType) {
    /**
     * The field's value is inherited from the parent object.
     */
    ValueSourceType["PARENT"] = "PARENT";
    /**
     * The field's value is user specified and not inherited.
     */
    ValueSourceType["DIRECTLY_SPECIFIED"] = "DIRECTLY_SPECIFIED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ValueSourceType["UNKNOWN"] = "UNKNOWN";
})(ValueSourceType || (exports.ValueSourceType = ValueSourceType = {}));
/**
 * Represents smart size modes.
 */
var SmartSizeMode;
(function (SmartSizeMode) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    SmartSizeMode["UNKNOWN"] = "UNKNOWN";
    /**
     * Fixed size mode (default).
     */
    SmartSizeMode["NONE"] = "NONE";
    /**
     * The height is fixed for the request, the width is a range.
     */
    SmartSizeMode["SMART_BANNER"] = "SMART_BANNER";
    /**
     * Height and width are ranges.
     */
    SmartSizeMode["DYNAMIC_SIZE"] = "DYNAMIC_SIZE";
})(SmartSizeMode || (exports.SmartSizeMode = SmartSizeMode = {}));
