// @guardian/google-admanager-api/dist/client/services/inventory/inventory.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InventoryService = void 0;
class InventoryService {
    constructor(client) {
        this._client = client;
    }
    async createAdUnits(adUnits) {
        return this._client.createAdUnits({ adUnits });
    }
    async getAdUnitSizesByStatement(filterStatement) {
        return this._client.getAdUnitSizesByStatement({
            filterStatement,
        });
    }
    async getAdUnitsByStatement(filterStatement) {
        return this._client.getAdUnitsByStatement({
            filterStatement,
        });
    }
    async performAdUnitAction(adUnitAction, filterStatement) {
        return this._client.performAdUnitAction({
            adUnitAction: {
                attributes: {
                    "xsi:type": adUnitAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateAdUnits(adUnits) {
        return this._client.updateAdUnits({ adUnits });
    }
}
exports.InventoryService = InventoryService;
