// @guardian/google-admanager-api/dist/common/enums/date.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MinuteOfHourEnum = exports.MinuteOfHour = exports.DayOfWeekEnum = exports.DayOfWeek = exports.TimeUnitEnum = exports.TimeUnit = exports.StartDateTimeTypeEnum = exports.StartDateTimeType = void 0;
const superstruct_1 = require("superstruct");
/**
 * Specifies the start type to use for an entity with a start date time field. For example, a LineItem or LineItemCreativeAssociation.
 */
var StartDateTimeType;
(function (StartDateTimeType) {
    /**
     * 	Use the value in startDateTime.
     */
    StartDateTimeType["USE_START_DATE_TIME"] = "USE_START_DATE_TIME";
    /**
     * 	The entity will start serving immediately. startDateTime in the request is ignored and will be set to the current time. Additionally, startDateTimeType will be set to StartDateTimeType.USE_START_DATE_TIME.
     */
    StartDateTimeType["IMMEDIATELY"] = "IMMEDIATELY";
    /**
     * 	The entity will start serving one hour from now. startDateTime in the request is ignored and will be set to one hour from the current time. Additionally, startDateTimeType will be set to StartDateTimeType.USE_START_DATE_TIME.
     */
    StartDateTimeType["ONE_HOUR_FROM_NOW"] = "ONE_HOUR_FROM_NOW";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    StartDateTimeType["UNKNOWN"] = "UNKNOWN";
})(StartDateTimeType || (exports.StartDateTimeType = StartDateTimeType = {}));
/**
 * Represents a StartDateTimeType enum in superstruct format.
 */
exports.StartDateTimeTypeEnum = (0, superstruct_1.enums)([
    "USE_START_DATE_TIME",
    "IMMEDIATELY",
    "ONE_HOUR_FROM_NOW",
    "UNKNOWN",
]);
/**
 * Represent the possible time units for frequency capping.
 */
var TimeUnit;
(function (TimeUnit) {
    TimeUnit["MINUTE"] = "MINUTE";
    TimeUnit["HOUR"] = "HOUR";
    TimeUnit["DAY"] = "DAY";
    TimeUnit["WEEK"] = "WEEK";
    TimeUnit["MONTH"] = "MONTH";
    TimeUnit["LIFETIME"] = "LIFETIME";
    /**
     * Per pod of ads in a video stream. Only valid for entities in a EnvironmentType.VIDEO_PLAYER environment.
     */
    TimeUnit["POD"] = "POD";
    /**
     * Per video stream. Only valid for entities in a EnvironmentType.VIDEO_PLAYER environment.
     */
    TimeUnit["STREAM"] = "STREAM";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    TimeUnit["UNKNOWN"] = "UNKNOWN";
})(TimeUnit || (exports.TimeUnit = TimeUnit = {}));
/**
 * Represents a TimeUnit enum in superstruct format.
 */
exports.TimeUnitEnum = (0, superstruct_1.enums)([
    "MINUTE",
    "HOUR",
    "DAY",
    "WEEK",
    "MONTH",
    "LIFETIME",
    "POD",
    "STREAM",
    "UNKNOWN",
]);
/**
 * Days of the week.
 */
var DayOfWeek;
(function (DayOfWeek) {
    /**
     * 	The day of week named Monday.
     */
    DayOfWeek["MONDAY"] = "MONDAY";
    /**
     * 	The day of week named Tuesday.
     */
    DayOfWeek["TUESDAY"] = "TUESDAY";
    /**
     * 	The day of week named Wednesday.
     */
    DayOfWeek["WEDNESDAY"] = "WEDNESDAY";
    /**
     * 	The day of week named Thursday.
     */
    DayOfWeek["THURSDAY"] = "THURSDAY";
    /**
     * 	The day of week named Friday.
     */
    DayOfWeek["FRIDAY"] = "FRIDAY";
    /**
     * 	The day of week named Saturday.
     */
    DayOfWeek["SATURDAY"] = "SATURDAY";
    /**
     * 	The day of week named Sunday.
     */
    DayOfWeek["SUNDAY"] = "SUNDAY";
})(DayOfWeek || (exports.DayOfWeek = DayOfWeek = {}));
/**
 * Represents a DayOfWeek enum in superstruct format.
 */
exports.DayOfWeekEnum = (0, superstruct_1.enums)([
    "MONDAY",
    "TUESDAY",
    "WEDNESDAY",
    "THURSDAY",
    "FRIDAY",
    "SATURDAY",
    "SUNDAY",
]);
/**
 * Minutes in an hour. Currently, only 0, 15, 30, and 45 are supported. This field is required.
 */
var MinuteOfHour;
(function (MinuteOfHour) {
    /**
     * 	Zero minutes past hour.
     */
    MinuteOfHour["ZERO"] = "ZERO";
    /**
     * 	Fifteen minutes past hour.
     */
    MinuteOfHour["FIFTEEN"] = "FIFTEEN";
    /**
     * 	Thirty minutes past hour.
     */
    MinuteOfHour["THIRTY"] = "THIRTY";
    /**
     * 	Forty-five minutes past hour.
     */
    MinuteOfHour["FORTY_FIVE"] = "FORTY_FIVE";
})(MinuteOfHour || (exports.MinuteOfHour = MinuteOfHour = {}));
/**
 * Represents a MinuteOfHour enum in superstruct format.
 */
exports.MinuteOfHourEnum = (0, superstruct_1.enums)([
    "ZERO",
    "FIFTEEN",
    "THIRTY",
    "FORTY_FIVE",
]);
