// @guardian/google-admanager-api/dist/common/handlers/exceptions.handler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdsReportsException = exports.ArgumentNullException = exports.InvalidOperationException = exports.BaseException = void 0;
class BaseException extends Error {
    constructor(message) {
        super(message);
        this.name = this.constructor.name;
        Object.setPrototypeOf(this, BaseException.prototype);
    }
}
exports.BaseException = BaseException;
class InvalidOperationException extends BaseException {
    constructor(message, innerException, beforeSealing) {
        super(message);
        this.statusCode = 500;
        this.name = this.constructor.name;
        this.innerException = innerException;
        this.beforeSealing = beforeSealing;
        Object.setPrototypeOf(this, InvalidOperationException.prototype);
    }
}
exports.InvalidOperationException = InvalidOperationException;
class ArgumentNullException extends BaseException {
    constructor(message, innerException, beforeSealing) {
        super(message);
        this.statusCode = 400;
        this.name = this.constructor.name;
        this.innerException = innerException;
        this.beforeSealing = beforeSealing;
        Object.setPrototypeOf(this, ArgumentNullException.prototype);
    }
}
exports.ArgumentNullException = ArgumentNullException;
class AdsReportsException extends BaseException {
    constructor(message, innerException, beforeSealing) {
        super(message);
        this.statusCode = 400;
        this.name = this.constructor.name;
        this.innerException = innerException;
        this.beforeSealing = beforeSealing;
        Object.setPrototypeOf(this, AdsReportsException.prototype);
    }
}
exports.AdsReportsException = AdsReportsException;
