// @guardian/google-admanager-api/dist/client/services/proposalLineItem/proposalLineItem.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProposalLineItemService = void 0;
class ProposalLineItemService {
    constructor(client) {
        this._client = client;
    }
    async createMakegoods(makegoodInfos) {
        return this._client.createMakegoods({ makegoodInfos });
    }
    async createProposalLineItems(proposalLineItems) {
        return this._client.createProposalLineItems({ proposalLineItems });
    }
    async getProposalLineItemsByStatement(filterStatement) {
        return this._client.getProposalLineItemsByStatement({
            filterStatement,
        });
    }
    async performProposalLineItemAction(proposalLineItemAction, filterStatement) {
        return this._client.performProposalLineItemAction({
            proposalLineItemAction: Object.assign({ attributes: {
                    "xsi:type": proposalLineItemAction.constructor.name,
                } }, proposalLineItemAction.buildAttributes()),
            filterStatement,
        });
    }
    async updateProposalLineItems(proposalLineItems) {
        return this._client.updateProposalLineItems({ proposalLineItems });
    }
}
exports.ProposalLineItemService = ProposalLineItemService;
