// @guardian/google-admanager-api/dist/client/services/proposalLineItem/proposalLineItem.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuyerPermissionType = exports.NegotiationRole = exports.ReservationStatus = exports.RateType = void 0;
/**
 * Describes the type of event the advertiser is paying for. The values here correspond to the values for the LineItem.costType field.
 */
var RateType;
(function (RateType) {
    /**
     * The rate applies to cost per mille (CPM) revenue.
     */
    RateType["CPM"] = "CPM";
    /**
     * The rate applies to cost per click (CPC) revenue.
     */
    RateType["CPC"] = "CPC";
    /**
     * The rate applies to cost per day (CPD) revenue.
     */
    RateType["CPD"] = "CPD";
    /**
     * The rate applies to cost per unit (CPU) revenue.
     */
    RateType["CPU"] = "CPU";
    /**
     * The rate applies to flat fee revenue.
     */
    RateType["FLAT_FEE"] = "FLAT_FEE";
    /**
     * The rate applies to Active View viewable cost per mille (vCPM) revenue.
     */
    RateType["VCPM"] = "VCPM";
    /**
     * The rate applies to cost per mille in-target (CPM In-Target).
     */
    RateType["CPM_IN_TARGET"] = "CPM_IN_TARGET";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    RateType["UNKNOWN"] = "UNKNOWN";
})(RateType || (exports.RateType = RateType = {}));
/**
 * Represents the inventory reservation status for ProposalLineItem objects.
 */
var ReservationStatus;
(function (ReservationStatus) {
    /**
     * The inventory is reserved.
     */
    ReservationStatus["RESERVED"] = "RESERVED";
    /**
     * The proposal line item's inventory is never reserved.
     */
    ReservationStatus["NOT_RESERVED"] = "NOT_RESERVED";
    /**
     * The inventory is once reserved and now released.
     */
    ReservationStatus["RELEASED"] = "RELEASED";
    /**
     * The reservation status of the corresponding LineItem should be used for this ProposalLineItem.
     */
    ReservationStatus["CHECK_LINE_ITEM_RESERVATION_STATUS"] = "CHECK_LINE_ITEM_RESERVATION_STATUS";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ReservationStatus["UNKNOWN"] = "UNKNOWN";
})(ReservationStatus || (exports.ReservationStatus = ReservationStatus = {}));
/**
 * The role (buyer or seller) that performed an action in the negotiation of a Proposal.
 */
var NegotiationRole;
(function (NegotiationRole) {
    NegotiationRole["BUYER"] = "BUYER";
    NegotiationRole["SELLER"] = "SELLER";
    NegotiationRole["UNKNOWN"] = "UNKNOWN";
})(NegotiationRole || (exports.NegotiationRole = NegotiationRole = {}));
/**
 * Describes the buyer permission type.
 */
var BuyerPermissionType;
(function (BuyerPermissionType) {
    /**
     * Only the buyer listed on the line item can transact.
     */
    BuyerPermissionType["NEGOTIATOR_ONLY"] = "NEGOTIATOR_ONLY";
    /**
     * All buyers under the listed buyer's bidder can transact on the line item.
     */
    BuyerPermissionType["BIDDER"] = "BIDDER";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    BuyerPermissionType["UNKNOWN"] = "UNKNOWN";
})(BuyerPermissionType || (exports.BuyerPermissionType = BuyerPermissionType = {}));
