// @guardian/google-admanager-api/dist/client/services/proposalLineItem/proposalLineItem.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnarchiveProposalLineItems = exports.ResumeProposalLineItems = exports.ReserveProposalLineItems = exports.ReleaseProposalLineItems = exports.PauseProposalLineItems = exports.ArchiveProposalLineItems = exports.ProposalLineItemAction = void 0;
/**
 * Represents the actions that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 */
class ProposalLineItemAction {
    buildAttributes() {
        return {};
    }
}
exports.ProposalLineItemAction = ProposalLineItemAction;
/**
 * The action used for archiving {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 */
class ArchiveProposalLineItems {
    buildAttributes() {
        return {};
    }
}
exports.ArchiveProposalLineItems = ArchiveProposalLineItems;
/**
 * The action used for pausing {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 */
class PauseProposalLineItems {
    /**
     * Reason to describe why the {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} is being paused.
     */
    get reason() {
        return this.reasonField;
    }
    /**
     * Reason to describe why the {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} is being paused.
     */
    set reason(value) {
        this.reasonField = value;
    }
    buildAttributes() {
        return this.reason ? { reason: this.reason } : {};
    }
}
exports.PauseProposalLineItems = PauseProposalLineItems;
/**
 * The action used for releasing inventory for {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 */
class ReleaseProposalLineItems {
    buildAttributes() {
        return {};
    }
}
exports.ReleaseProposalLineItems = ReleaseProposalLineItems;
/**
 * The action to reserve inventory for {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 * It does not overbook inventory unless {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ReserveProposalLineItems#allowOverbook allowOverbook} is set to `true`. This action is only applicable for programmatic proposals not using sales management.
 */
class ReserveProposalLineItems {
    constructor() {
        this.allowOverbookField = false;
    }
    get allowOverbook() {
        return this.allowOverbookField;
    }
    set allowOverbook(value) {
        this.allowOverbookField = value;
    }
    buildAttributes() {
        return { allowOverbook: this.allowOverbookField };
    }
}
exports.ReserveProposalLineItems = ReserveProposalLineItems;
/**
 * The action used for resuming {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 */
class ResumeProposalLineItems {
    buildAttributes() {
        return {};
    }
}
exports.ResumeProposalLineItems = ResumeProposalLineItems;
/**
 * The action used for unarchiving {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.ProposalLineItem ProposalLineItem} objects.
 */
class UnarchiveProposalLineItems {
    buildAttributes() {
        return {};
    }
}
exports.UnarchiveProposalLineItems = UnarchiveProposalLineItems;
