// @guardian/google-admanager-api/dist/common/types/date.type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeOfDayStruct = exports.DateTimeRangeStruct = exports.DateTimeStruct = exports.DateStruct = void 0;
const superstruct_1 = require("superstruct");
const enums_1 = require("../enums");
/**
 * Represents a date struct.
 */
exports.DateStruct = (0, superstruct_1.object)({
    year: (0, superstruct_1.number)(),
    month: (0, superstruct_1.number)(),
    day: (0, superstruct_1.number)(),
});
/**
 * Represents a DateTime struct.
 */
exports.DateTimeStruct = (0, superstruct_1.object)({
    date: exports.DateStruct,
    hour: (0, superstruct_1.number)(),
    minute: (0, superstruct_1.number)(),
    second: (0, superstruct_1.number)(),
    timeZoneId: (0, superstruct_1.string)(),
});
/**
 * Represents a DateTimeRange struct.
 */
exports.DateTimeRangeStruct = (0, superstruct_1.object)({
    startDateTime: exports.DateTimeStruct,
    endDateTime: exports.DateTimeStruct,
});
/**
 * Represents a TimeOfDay struct.
 */
exports.TimeOfDayStruct = (0, superstruct_1.object)({
    hour: (0, superstruct_1.number)(),
    minute: enums_1.MinuteOfHourEnum,
});
