// @guardian/google-admanager-api/dist/client/services/contact/contact.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactStatus = void 0;
/**
 * Describes the contact statuses.
 */
var ContactStatus;
(function (ContactStatus) {
    /**
     * The contact has not been invited to see their orders.
     */
    ContactStatus["UNINVITED"] = "UNINVITED";
    /**
     * The contact has been invited to see their orders, but has not yet accepted the invitation.
     */
    ContactStatus["INVITE_PENDNG"] = "INVITE_PENDNG";
    /**
     * The contact has been invited to see their orders, but the invitation has already expired.
     */
    ContactStatus["INVITE_EXPIRED"] = "INVITE_EXPIRED";
    /**
     * The contact was invited to see their orders, but the invitation was cancelled.
     */
    ContactStatus["INVITE_CANCELED"] = "INVITE_CANCELED";
    /**
     * The contact has access to login and view their orders.
     */
    ContactStatus["USER_ACTIVE"] = "USER_ACTIVE";
    /**
     * The contact accepted an invitation to see their orders, but their access was later revoked.
     */
    ContactStatus["USER_DISABLED"] = "USER_DISABLED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ContactStatus["UNKNOWN"] = "UNKNOWN";
})(ContactStatus || (exports.ContactStatus = ContactStatus = {}));
