// @guardian/google-admanager-api/dist/client/services/daiAuthenticationKey/daiAuthenticationKey.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaiAuthenticationKeyType = exports.DaiAuthenticationKeyStatus = void 0;
/**
 * Statuses associated with DaiAuthenticationKey objects.
 */
var DaiAuthenticationKeyStatus;
(function (DaiAuthenticationKeyStatus) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    DaiAuthenticationKeyStatus["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates that the API key is actively in use and that the IMA SDK API should accept it as a valid key in requests.
     */
    DaiAuthenticationKeyStatus["ACTIVE"] = "ACTIVE";
    /**
     * Indicates that the API key is no longer is use and that the IMA SDK API should not accept it as a valid key in requests.
     */
    DaiAuthenticationKeyStatus["INACTIVE"] = "INACTIVE";
})(DaiAuthenticationKeyStatus || (exports.DaiAuthenticationKeyStatus = DaiAuthenticationKeyStatus = {}));
/**
 * Key types associated with DaiAuthenticationKey objects.
 */
var DaiAuthenticationKeyType;
(function (DaiAuthenticationKeyType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    DaiAuthenticationKeyType["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates that the key is a standard API key and should be used with the api-key SDK parameter when authenticating stream create requests.
     */
    DaiAuthenticationKeyType["API"] = "API";
    /**
     * Indicates that the key is an HMAC key and should be used to generate a signature for the stream create request with the auth-token SDK parameter.
     */
    DaiAuthenticationKeyType["HMAC"] = "HMAC";
})(DaiAuthenticationKeyType || (exports.DaiAuthenticationKeyType = DaiAuthenticationKeyType = {}));
