// @guardian/google-admanager-api/dist/client/services/daiAuthenticationKey/daiAuthenticationKey.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaiAuthenticationKeyService = void 0;
class DaiAuthenticationKeyService {
    constructor(client) {
        this._client = client;
    }
    async createDaiAuthenticationKeys(daiAuthenticationKeys) {
        return this._client.createDaiAuthenticationKeys({ daiAuthenticationKeys });
    }
    async getDaiAuthenticationKeysByStatement(filterStatement) {
        return this._client.getDaiAuthenticationKeysByStatement({
            filterStatement,
        });
    }
    async performDaiAuthenticationKeyAction(daiAuthenticationKeyAction, filterStatement) {
        return this._client.performDaiAuthenticationKeyAction({
            daiAuthenticationKeyAction: {
                attributes: {
                    "xsi:type": daiAuthenticationKeyAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateDaiAuthenticationKeys(daiAuthenticationKeys) {
        return this._client.updateDaiAuthenticationKeys({ daiAuthenticationKeys });
    }
}
exports.DaiAuthenticationKeyService = DaiAuthenticationKeyService;
