// @guardian/google-admanager-api/dist/client/services/customField/customField.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldVisibility = exports.CustomFieldDataType = exports.CustomFieldEntityType = void 0;
/**
 * Entity types recognized by custom fields
 */
var CustomFieldEntityType;
(function (CustomFieldEntityType) {
    /**
     * Represents the LineItem type.
     */
    CustomFieldEntityType["LINE_ITEM"] = "LINE_ITEM";
    /**
     * Represents the Order type.
     */
    CustomFieldEntityType["ORDER"] = "ORDER";
    /**
     * Represents the Creative type.
     */
    CustomFieldEntityType["CREATIVE"] = "CREATIVE";
    /**
     * Represents the ProductTemplate type.
     */
    CustomFieldEntityType["PRODUCT_TEMPLATE"] = "PRODUCT_TEMPLATE";
    /**
     * Represents the Product type.
     */
    CustomFieldEntityType["PRODUCT"] = "PRODUCT";
    /**
     * Represents the Proposal type.
     */
    CustomFieldEntityType["PROPOSAL"] = "PROPOSAL";
    /**
     * Represents the ProposalLineItem type.
     */
    CustomFieldEntityType["PROPOSAL_LINE_ITEM"] = "PROPOSAL_LINE_ITEM";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CustomFieldEntityType["UNKNOWN"] = "UNKNOWN";
})(CustomFieldEntityType || (exports.CustomFieldEntityType = CustomFieldEntityType = {}));
/**
 * The data types allowed for {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomFieldService.CustomField CustomField} objects.
 */
var CustomFieldDataType;
(function (CustomFieldDataType) {
    /**
     * A string field. The max length is 255 characters.
     */
    CustomFieldDataType["STRING"] = "STRING";
    /**
     * A number field.
     */
    CustomFieldDataType["NUMBER"] = "NUMBER";
    /**
     * A boolean field. Values may be "true", "false", or empty.
     */
    CustomFieldDataType["TOGGLE"] = "TOGGLE";
    /**
     * A drop-down field. Values may only be the ids of CustomFieldOption objects.
     */
    CustomFieldDataType["DROP_DOWN"] = "DROP_DOWN";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CustomFieldDataType["UNKNOWN"] = "UNKNOWN";
})(CustomFieldDataType || (exports.CustomFieldDataType = CustomFieldDataType = {}));
/**
 * The visibility levels of a custom field.
 */
var CustomFieldVisibility;
(function (CustomFieldVisibility) {
    /**
     * Only visible through the API.
     */
    CustomFieldVisibility["API_ONLY"] = "API_ONLY";
    /**
     * Visible in the UI, but only editable through the API
     */
    CustomFieldVisibility["READ_ONLY"] = "READ_ONLY";
    /**
     * Visible and editable both in the API and the UI.
     */
    CustomFieldVisibility["FULL"] = "FULL";
})(CustomFieldVisibility || (exports.CustomFieldVisibility = CustomFieldVisibility = {}));
