// @guardian/google-admanager-api/dist/client/services/customField/customField.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldService = void 0;
class CustomFieldService {
    constructor(client) {
        this._client = client;
    }
    async createCustomFieldOptions(customFieldOptions) {
        return this._client.createCustomFieldOptions({ customFieldOptions });
    }
    async createCustomFields(customFields) {
        return this._client.createCustomFields({ customFields });
    }
    async getCustomFieldOption(customFieldOptionId) {
        return this._client.getCustomFieldsByStatement({
            customFieldOptionId,
        });
    }
    async getCustomFieldsByStatement(filterStatement) {
        return this._client.getCustomFieldsByStatement({
            filterStatement,
        });
    }
    async performCustomFieldAction(customFieldAction, filterStatement) {
        return this._client.performCustomFieldAction({
            customFieldAction: {
                attributes: {
                    "xsi:type": customFieldAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateCustomFieldOptions(customFieldOptions) {
        return this._client.updateCustomFields({ customFieldOptions });
    }
    async updateCustomFields(CustomFields) {
        return this._client.updateCustomFields({ CustomFields });
    }
}
exports.CustomFieldService = CustomFieldService;
