// @guardian/google-admanager-api/dist/client/services/daiEncodingProfile/daiEncodingProfile.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContainerType = exports.VariantType = exports.DaiEncodingProfileStatus = void 0;
/**
 * Describes the status of a {@link https://developers.google.com/ad-manager/api/reference/v202505/DaiEncodingProfileService.DaiEncodingProfile DaiEncodingProfile} object.
 */
var DaiEncodingProfileStatus;
(function (DaiEncodingProfileStatus) {
    /**
     * Indicates the DaiEncodingProfile has been created and is eligible for streaming.
     */
    DaiEncodingProfileStatus["ACTIVE"] = "ACTIVE";
    /**
     * Indicates the DaiEncodingProfile has been archived.
     */
    DaiEncodingProfileStatus["ARCHIVED"] = "ARCHIVED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    DaiEncodingProfileStatus["UNKNOWN"] = "UNKNOWN";
})(DaiEncodingProfileStatus || (exports.DaiEncodingProfileStatus = DaiEncodingProfileStatus = {}));
/**
 * Describes the variant playlist type that the profile represents.
 */
var VariantType;
(function (VariantType) {
    /**
     * Media variant playlist type. Media playlists may: contain audio only, video only, or audio and video.
     */
    VariantType["MEDIA"] = "MEDIA";
    /**
     * iFrame variant playlist type. iFrame playlists may: contain video or contain audio and video (i.e. video must be present).
     */
    VariantType["IFRAME"] = "IFRAME";
    /**
     * Subtitles variant playlist type.
     */
    VariantType["SUBTITLES"] = "SUBTITLES";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    VariantType["UNKNOWN"] = "UNKNOWN";
})(VariantType || (exports.VariantType = VariantType = {}));
/**
 * Describes the digital media container type of the underlying media.
 */
var ContainerType;
(function (ContainerType) {
    /**
     * Transport stream (TS) container.
     */
    ContainerType["TS"] = "TS";
    /**
     * Fragmented MPEG-4 (fMP4) output container.
     */
    ContainerType["FMP4"] = "FMP4";
    /**
     * HTTP live streaming (HLS) packed audio container.
     */
    ContainerType["HLS_AUDIO"] = "HLS_AUDIO";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ContainerType["UNKNOWN"] = "UNKNOWN";
})(ContainerType || (exports.ContainerType = ContainerType = {}));
