// @guardian/google-admanager-api/dist/client/services/daiEncodingProfile/daiEncodingProfile.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaiEncodingProfileService = void 0;
class DaiEncodingProfileService {
    constructor(client) {
        this._client = client;
    }
    async createDaiEncodingProfiles(daiEncodingProfiles) {
        return this._client.createDaiEncodingProfiles({ daiEncodingProfiles });
    }
    async getDaiEncodingProfilesByStatement(filterStatement) {
        return this._client.getDaiEncodingProfilesByStatement({
            filterStatement,
        });
    }
    async performDaiEncodingProfileAction(daiEncodingProfileAction, filterStatement) {
        return this._client.performDaiEncodingProfileAction({
            daiEncodingProfileAction: {
                attributes: {
                    "xsi:type": daiEncodingProfileAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateDaiEncodingProfiles(daiEncodingProfiles) {
        return this._client.updateDaiEncodingProfiles({ daiEncodingProfiles });
    }
}
exports.DaiEncodingProfileService = DaiEncodingProfileService;
