// @guardian/google-admanager-api/dist/client/services/activity/activity.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActivityType = exports.ActivityStatus = void 0;
/**
 * The activity status.
 */
var ActivityStatus;
(function (ActivityStatus) {
    ActivityStatus["ACTIVE"] = "ACTIVE\t";
    ActivityStatus["INACTIVE"] = "INACTIVE";
})(ActivityStatus || (exports.ActivityStatus = ActivityStatus = {}));
/**
 * The activity type.
 */
var ActivityType;
(function (ActivityType) {
    /**
     * Tracks conversions for each visit to a webpage. This is a counter type.
     */
    ActivityType["PAGE_VIEWS"] = "PAGE_VIEWS";
    /**
     * Tracks conversions for visits to a webpage, but only counts one conversion per user per day, even if a user visits the page multiple times. This is a counter type.
     */
    ActivityType["DAILY_VISITS"] = "DAILY_VISITS";
    /**
     * Tracks conversions for visits to a webpage, but only counts one conversion per user per user session. Session length is set by the advertiser. This is a counter type.
     */
    ActivityType["CUSTOM"] = "CUSTOM";
    /**
     * Tracks conversions where the user has made a purchase, the monetary value of each purchase, plus the number of items that were purchased and the order ID. This is a sales type.
     */
    ActivityType["ITEMS_PURCHASED"] = "ITEMS_PURCHASED";
    /**
     * Tracks conversions where the user has made a purchase, the monetary value of each purchase, plus the order ID (but not the number of items purchased). This is a sales type.
     */
    ActivityType["TRANSACTIONS"] = "TRANSACTIONS";
    /**
     * Tracks conversions where the user has installed an iOS application. This is a counter type.
     */
    ActivityType["IOS_APPLICATION_DOWNLOADS"] = "IOS_APPLICATION_DOWNLOADS";
    /**
     * Tracks conversions where the user has installed an Android application. This is a counter type.
     */
    ActivityType["ANDROID_APPLICATION_DOWNLOADS"] = "ANDROID_APPLICATION_DOWNLOADS";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ActivityType["UNKNOWN"] = "UNKNOWN";
})(ActivityType || (exports.ActivityType = ActivityType = {}));
