// @guardian/google-admanager-api/dist/client/services/forecast/forecast.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForecastService = void 0;
class ForecastService {
    constructor(client) {
        this._client = client;
    }
    getAvailabilityForecast(lineItem, forecastOptions) {
        return this._client.getAvailabilityForecast({ lineItem, forecastOptions });
    }
    getAvailabilityForecastById(lineItemId, forecastOptions) {
        return this._client.getAvailabilityForecastById({
            lineItemId,
            forecastOptions,
        });
    }
    getDeliveryForecast(lineItems, forecastOptions) {
        return this._client.getDeliveryForecast({ lineItems, forecastOptions });
    }
    getDeliveryForecastByIds(lineItemIds, forecastOptions) {
        return this._client.getDeliveryForecastByIds({
            lineItemIds,
            forecastOptions,
        });
    }
    getTrafficData(trafficDataRequest) {
        return this._client.getTrafficData({ trafficDataRequest });
    }
}
exports.ForecastService = ForecastService;
