// @guardian/google-admanager-api/dist/client/services/forecast/forecast.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrpAge = exports.GrpGender = exports.GrpUnitType = exports.TargetingDimension = void 0;
/**
 * Targeting dimension of targeting breakdowns.
 */
var TargetingDimension;
(function (TargetingDimension) {
    TargetingDimension["CUSTOM_CRITERIA"] = "CUSTOM_CRITERIA";
    TargetingDimension["GEOGRAPHY"] = "GEOGRAPHY";
    TargetingDimension["BROWSER"] = "BROWSER";
    TargetingDimension["BROWSER_LANGUAGE"] = "BROWSER_LANGUAGE";
    TargetingDimension["BANDWIDTH_GROUP"] = "BANDWIDTH_GROUP";
    TargetingDimension["OPERATING_SYSTEM"] = "OPERATING_SYSTEM";
    TargetingDimension["USER_DOMAIN"] = "USER_DOMAIN";
    TargetingDimension["CONTENT"] = "CONTENT";
    TargetingDimension["VIDEO_POSITION"] = "VIDEO_POSITION";
    TargetingDimension["AD_SIZE"] = "AD_SIZE";
    TargetingDimension["AD_UNIT"] = "AD_UNIT";
    TargetingDimension["PLACEMENT"] = "PLACEMENT";
    TargetingDimension["MOBILE_CARRIER"] = "MOBILE_CARRIER";
    TargetingDimension["DEVICE_CAPABILITY"] = "DEVICE_CAPABILITY";
    TargetingDimension["DEVICE_CATEGORY"] = "DEVICE_CATEGORY";
    TargetingDimension["DEVICE_MANUFACTURER"] = "DEVICE_MANUFACTURER";
    TargetingDimension["MOBILE_APPLICATION"] = "MOBILE_APPLICATION";
    TargetingDimension["FORECASTED_CREATIVE_RESTRICTION"] = "FORECASTED_CREATIVE_RESTRICTION";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    TargetingDimension["UNKNOWN"] = "UNKNOWN";
})(TargetingDimension || (exports.TargetingDimension = TargetingDimension = {}));
/**
 * Type of unit represented in a GRP demographic breakdown.
 */
var GrpUnitType;
(function (GrpUnitType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    GrpUnitType["UNKNOWN"] = "UNKNOWN";
    GrpUnitType["IMPRESSIONS"] = "IMPRESSIONS";
})(GrpUnitType || (exports.GrpUnitType = GrpUnitType = {}));
/**
 * The demographic gender associated with a GRP demographic forecast.
 */
var GrpGender;
(function (GrpGender) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    GrpGender["UNKNOWN"] = "UNKNOWN";
    /**
     * When gender is not available due to low impression levels, GRP privacy thresholds are activated and prevent us from specifying gender.
     */
    GrpGender["GENDER_UNKNOWN"] = "GENDER_UNKNOWN";
    GrpGender["GENDER_FEMALE"] = "GENDER_FEMALE";
    GrpGender["GENDER_MALE"] = "GENDER_MALE";
})(GrpGender || (exports.GrpGender = GrpGender = {}));
/**
 * The age range associated with a GRP demographic forecast.
 */
var GrpAge;
(function (GrpAge) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    GrpAge["UNKNOWN"] = "UNKNOWN";
    /**
     * When the age range is not available due to low impression levels, GRP privacy thresholds are activated and prevent us from specifying age.
     */
    GrpAge["AGE_UNKNOWN"] = "AGE_UNKNOWN";
    GrpAge["AGE_0_TO_17"] = "AGE_0_TO_17";
    GrpAge["AGE_18_TO_24"] = "AGE_18_TO_24";
    GrpAge["AGE_25_TO_34"] = "AGE_25_TO_34";
    GrpAge["AGE_35_TO_44"] = "AGE_35_TO_44";
    GrpAge["AGE_45_TO_54"] = "AGE_45_TO_54";
    GrpAge["AGE_55_TO_64"] = "AGE_55_TO_64";
    GrpAge["AGE_65_PLUS"] = "AGE_65_PLUS";
    GrpAge["AGE_18_TO_49"] = "AGE_18_TO_49";
    GrpAge["AGE_21_TO_34"] = "AGE_21_TO_34";
    GrpAge["AGE_21_TO_49"] = "AGE_21_TO_49";
    GrpAge["AGE_21_PLUS"] = "AGE_21_PLUS";
    GrpAge["AGE_25_TO_49"] = "AGE_25_TO_49";
    GrpAge["AGE_21_TO_44"] = "AGE_21_TO_44";
    GrpAge["AGE_21_TO_54"] = "AGE_21_TO_54";
    GrpAge["AGE_21_TO_64"] = "AGE_21_TO_64";
    GrpAge["AGE_35_TO_49"] = "AGE_35_TO_49";
})(GrpAge || (exports.GrpAge = GrpAge = {}));
