// @guardian/google-admanager-api/dist/client/services/cmsMetadata/cmsMetadata.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CmsMetadataService = void 0;
class CmsMetadataService {
    constructor(client) {
        this._client = client;
    }
    async getCmsMetadataKeysByStatement(statement) {
        return this._client.getCmsMetadataKeysByStatement({
            statement,
        });
    }
    async getCmsMetadataValuesByStatement(statement) {
        return this._client.getCmsMetadataValuesByStatement({
            statement,
        });
    }
    async performCmsMetadataKeyAction(cmsMetadataKeyAction, filterStatement) {
        return this._client.performCmsMetadataKeyAction({
            cmsMetadataKeyAction: {
                attributes: {
                    "xsi:type": cmsMetadataKeyAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async performCmsMetadataValueAction(cmsMetadataValueAction, filterStatement) {
        return this._client.performCmsMetadataValueAction({
            cmsMetadataValueAction: {
                attributes: {
                    "xsi:type": cmsMetadataValueAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
}
exports.CmsMetadataService = CmsMetadataService;
