// @guardian/google-admanager-api/dist/client/services/cmsMetadata/cmsMetadata.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeactivateCmsMetadataValues = exports.ActivateCmsMetadataValues = exports.CmsMetadataValueAction = exports.DeactivateCmsMetadataKeys = exports.ActivateCmsMetadataKeys = exports.CmsMetadataKeyAction = void 0;
/**
 * Represents the actions that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CmsMetadataService.CmsMetadataKey CmsMetadataKey} objects.
 */
class CmsMetadataKeyAction {
}
exports.CmsMetadataKeyAction = CmsMetadataKeyAction;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/CmsMetadataService.CmsMetadataKey CmsMetadataKey} objects.
 */
class ActivateCmsMetadataKeys {
}
exports.ActivateCmsMetadataKeys = ActivateCmsMetadataKeys;
/**
 * The action used for deactivating {@link https://developers.google.com/ad-manager/api/reference/v202505/CmsMetadataService.CmsMetadataKey CmsMetadataKey} objects.
 */
class DeactivateCmsMetadataKeys {
}
exports.DeactivateCmsMetadataKeys = DeactivateCmsMetadataKeys;
/**
 * Represents the actions that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CmsMetadataService.CmsMetadataValue CmsMetadataValue} objects.
 */
class CmsMetadataValueAction {
}
exports.CmsMetadataValueAction = CmsMetadataValueAction;
/**
 * The action used for activating {@link https://developers.google.com/ad-manager/api/reference/v202505/CmsMetadataService.CmsMetadataValue CmsMetadataValue} objects.
 */
class ActivateCmsMetadataValues {
}
exports.ActivateCmsMetadataValues = ActivateCmsMetadataValues;
/**
 * The action used for deactivating {@link https://developers.google.com/ad-manager/api/reference/v202505/CmsMetadataService.CmsMetadataValue CmsMetadataValue} objects.
 */
class DeactivateCmsMetadataValues {
}
exports.DeactivateCmsMetadataValues = DeactivateCmsMetadataValues;
