// @guardian/google-admanager-api/dist/client/services/cmsMetadata/cmsMetadata.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CmsMetadataValueStatus = exports.CmsMetadataKeyStatus = void 0;
/**
 * Status for **`CmsMetadataKey`** objects.
 */
var CmsMetadataKeyStatus;
(function (CmsMetadataKeyStatus) {
    CmsMetadataKeyStatus["ACTIVE"] = "ACTIVE";
    CmsMetadataKeyStatus["INACTIVE"] = "INACTIVE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CmsMetadataKeyStatus["UNKNOWN"] = "UNKNOWN";
})(CmsMetadataKeyStatus || (exports.CmsMetadataKeyStatus = CmsMetadataKeyStatus = {}));
/**
 * Status for **`CmsMetadataValue`** objects.
 */
var CmsMetadataValueStatus;
(function (CmsMetadataValueStatus) {
    CmsMetadataValueStatus["ACTIVE"] = "ACTIVE";
    CmsMetadataValueStatus["INACTIVE"] = "INACTIVE";
    CmsMetadataValueStatus["ARCHIVED"] = "ARCHIVED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CmsMetadataValueStatus["UNKNOWN"] = "UNKNOWN";
})(CmsMetadataValueStatus || (exports.CmsMetadataValueStatus = CmsMetadataValueStatus = {}));
