// @guardian/google-admanager-api/dist/client/services/lineItemCreativeAssociation/lineItemCreativeAssociation.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineItemCreativeAssociationService = void 0;
class LineItemCreativeAssociationService {
    constructor(client) {
        this._client = client;
    }
    async createLineItemCreativeAssociations(lineItemCreativeAssociations) {
        return this._client.createLineItemCreativeAssociations({
            lineItemCreativeAssociations,
        });
    }
    async getPreviewUrl(lineItemId, creativeId, siteUrl) {
        return this._client.getPreviewUrl({
            lineItemId,
            creativeId,
            siteUrl,
        });
    }
    async getPreviewUrlsForNativeStyles(lineItemId, creativeId, siteUrl) {
        return this._client.getPreviewUrlsForNativeStyles({
            lineItemId,
            creativeId,
            siteUrl,
        });
    }
    async getLineItemCreativeAssociationsByStatement(filterStatement) {
        return this._client.getLineItemCreativeAssociationsByStatement({
            filterStatement,
        });
    }
    async performLineItemCreativeAssociationAction(lineItemCreativeAssociationAction, filterStatement) {
        return this._client.performLineItemCreativeAssociationAction({
            lineItemCreativeAssociationAction: {
                attributes: {
                    "xsi:type": lineItemCreativeAssociationAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async pushCreativeToDevices(filterStatement, options) {
        return this._client.pushCreativeToDevices({
            filterStatement,
            options,
        });
    }
    async updateLineItemCreativeAssociations(lineItemCreativeAssociations) {
        return this._client.updateLineItemCreativeAssociations({
            lineItemCreativeAssociations,
        });
    }
}
exports.LineItemCreativeAssociationService = LineItemCreativeAssociationService;
