// @guardian/google-admanager-api/dist/client/services/contentBundle/contentBundles.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentBundleService = void 0;
class ContentBundleService {
    constructor(client) {
        this._client = client;
    }
    async createContentBundles(contentBundles) {
        return this._client.createContentBundles({ contentBundles });
    }
    async getContentBundlesByStatement(filterStatement) {
        return this._client.getContentBundlesByStatement({
            filterStatement,
        });
    }
    async performContentBundleAction(contentBundleAction, filterStatement) {
        return this._client.performContentBundleAction({
            contentBundleAction: {
                attributes: {
                    "xsi:type": contentBundleAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateContentBundles(contentBundles) {
        return this._client.updateContentBundles({ contentBundles });
    }
}
exports.ContentBundleService = ContentBundleService;
