// @guardian/google-admanager-api/dist/client/common/types/creativePlaceholder.type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreativePlaceholderStruct = void 0;
const enums_1 = require("../enums");
const general_type_1 = require("./general.type");
const superstruct_1 = require("superstruct");
const label_type_1 = require("./label.type");
/**
 * Represents a CreativePlaceholder struct.
 */
exports.CreativePlaceholderStruct = (0, superstruct_1.object)({
    size: general_type_1.SizeStruct,
    creativeTemplateId: (0, superstruct_1.optional)((0, superstruct_1.number)()),
    companions: (0, superstruct_1.optional)((0, superstruct_1.lazy)(() => (0, superstruct_1.array)(exports.CreativePlaceholderStruct))),
    appliedLabels: (0, superstruct_1.optional)((0, superstruct_1.array)(label_type_1.AppliedLabelStruct)),
    effectiveAppliedLabels: (0, superstruct_1.optional)((0, superstruct_1.array)(label_type_1.AppliedLabelStruct)),
    expectedCreativeCount: (0, superstruct_1.optional)((0, superstruct_1.number)()),
    creativeSizeType: enums_1.CreativeSizeTypeEnum,
    targetingName: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    isAmpOnly: (0, superstruct_1.optional)((0, superstruct_1.boolean)()),
});
