// @guardian/google-admanager-api/dist/client/common/types/customCriteria.type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomCriteriaSetStruct = exports.AudienceSegmentCriteria = exports.CmsMetadataCriteriaStruct = exports.CustomCriteriaStruct = void 0;
const superstruct_1 = require("superstruct");
const enums_1 = require("../enums");
/**
 * Represents a CustomCriteria struct.
 */
exports.CustomCriteriaStruct = (0, superstruct_1.object)({
    //Need to double check the type for attributes
    attributes: (0, superstruct_1.optional)((0, superstruct_1.object)({ "xsi:type": (0, superstruct_1.literal)("CustomCriteria") })),
    keyId: (0, superstruct_1.number)(),
    valueIds: (0, superstruct_1.array)((0, superstruct_1.number)()),
    operator: enums_1.ComparisonOperatorEnum,
});
/**
 * Represents a CmsMetadataCriteria struct.
 */
exports.CmsMetadataCriteriaStruct = (0, superstruct_1.object)({
    attributes: (0, superstruct_1.optional)((0, superstruct_1.object)({ "xsi:type": (0, superstruct_1.literal)("CmsMetadataCriteria") })),
    operator: enums_1.CmsMetadataCriteriaComparisonOperatorEnum,
    cmsMetadataValueIds: (0, superstruct_1.array)((0, superstruct_1.number)()),
});
/**
 * Represents an AudienceSegmentCriteria struct.
 */
exports.AudienceSegmentCriteria = (0, superstruct_1.object)({
    attributes: (0, superstruct_1.optional)((0, superstruct_1.object)({ "xsi:type": (0, superstruct_1.literal)("AudienceSegmentCriteria") })),
    operator: enums_1.ComparisonOperatorEnum,
    audienceSegmentIds: (0, superstruct_1.array)((0, superstruct_1.number)()),
});
/**
 * Represents a CustomCriteriaSet struct.
 */
exports.CustomCriteriaSetStruct = (0, superstruct_1.object)({
    attributes: (0, superstruct_1.optional)((0, superstruct_1.object)({ "xsi:type": (0, superstruct_1.literal)("CustomCriteriaSet") })),
    logicalOperator: enums_1.LogicalOperatorEnum,
    children: (0, superstruct_1.array)((0, superstruct_1.union)([
        exports.CustomCriteriaStruct,
        (0, superstruct_1.lazy)(() => exports.CustomCriteriaSetStruct),
        exports.CmsMetadataCriteriaStruct,
        exports.AudienceSegmentCriteria,
    ])),
});
