// @guardian/google-admanager-api/dist/client/common/types/lineItemSummary.type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineItemDealInfoDtoStruct = exports.CustomPacingCurveStruct = exports.LineItemActivityAssociationStruct = exports.CustomPacingGoalsStruct = void 0;
const types_1 = require("../../../common/types");
const enums_1 = require("../enums");
const _1 = require(".");
const superstruct_1 = require("superstruct");
/**
 * Represents a CustomPacingGoal struct
 */
exports.CustomPacingGoalsStruct = (0, superstruct_1.object)({
    startDateTime: types_1.DateTimeStruct,
    useLineItemStartDateTime: (0, superstruct_1.boolean)(),
    amount: (0, superstruct_1.number)(),
});
/**
 * Represents a LineItemActivityAssociation struct.
 */
exports.LineItemActivityAssociationStruct = (0, superstruct_1.object)({
    activityId: (0, superstruct_1.number)(),
    clickThroughConversionCost: _1.MoneyStruct,
    viewThroughConversionCost: _1.MoneyStruct,
});
/**
 * Represents a CustomPacingCurve struct
 */
exports.CustomPacingCurveStruct = (0, superstruct_1.object)({
    customPacingGoalUnit: enums_1.CustomPacingGoalUnitEnum,
    customPacingGoals: (0, superstruct_1.array)(exports.CustomPacingGoalsStruct),
});
/**
 * Represents a LineItemDealInfoDto struct.
 */
exports.LineItemDealInfoDtoStruct = (0, superstruct_1.object)({
    externalDealId: (0, superstruct_1.number)(),
});
