// @guardian/google-admanager-api/dist/client/common/types/thirdParty.type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThirdPartyMeasurementSettingsStruct = void 0;
const superstruct_1 = require("superstruct");
const enums_1 = require("../enums");
/**
 * Represents a ThirdPartyMeasurementSettings struct.
 */
exports.ThirdPartyMeasurementSettingsStruct = (0, superstruct_1.object)({
    viewabilityPartner: enums_1.ThirdPartyViewabilityIntegrationPartnerEnum,
    viewabilityClientId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    viewabilityReportingId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    publisherViewabilityPartner: enums_1.ThirdPartyViewabilityIntegrationPartnerEnum,
    publisherViewabilityClientId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    publisherViewabilityReportingId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    brandLiftPartner: enums_1.ThirdPartyBrandLiftIntegrationPartnerEnum,
    brandLiftClientId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    brandLiftReportingId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    reachPartner: enums_1.ThirdPartyReachIntegrationPartnerEnum,
    reachClientId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    reachReportingId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    publisherReachPartner: enums_1.ThirdPartyReachIntegrationPartnerEnum,
    publisherReachClientId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
    publisherReachReportingId: (0, superstruct_1.optional)((0, superstruct_1.string)()),
});
