// @guardian/google-admanager-api/dist/client/common/types/videoPosition.type.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoPositionTargetingStruct = exports.VideoPositionTargetStruct = exports.VideoPositionWithinPodStruct = exports.VideoPositionStruct = void 0;
const superstruct_1 = require("superstruct");
const enums_1 = require("../enums");
/**
 * Represents a VideoPosition struct.
 */
exports.VideoPositionStruct = (0, superstruct_1.object)({
    positionType: enums_1.VideoPositionTypeEnum,
    midrollIndex: (0, superstruct_1.number)(),
});
exports.VideoPositionWithinPodStruct = (0, superstruct_1.object)({ index: (0, superstruct_1.number)() });
/**
 * Represents a VideoPositionTarget struct.
 */
exports.VideoPositionTargetStruct = (0, superstruct_1.object)({
    videoPosition: exports.VideoPositionStruct,
    videoBumperType: (0, superstruct_1.optional)(enums_1.VideoBumperTypeEnum),
    videoPositionWithinPod: (0, superstruct_1.optional)(exports.VideoPositionWithinPodStruct),
    adSpotId: (0, superstruct_1.optional)((0, superstruct_1.array)((0, superstruct_1.number)())),
});
/**
 * Represents a VideoPositionTargeting struct.
 */
exports.VideoPositionTargetingStruct = (0, superstruct_1.object)({
    targetedPositions: (0, superstruct_1.array)(exports.VideoPositionTargetStruct),
});
