// @guardian/google-admanager-api/dist/client/services/mobileApplication/mobileApplication.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileApplicationService = void 0;
class MobileApplicationService {
    constructor(client) {
        this._client = client;
    }
    async createMobileApplications(mobileApplications) {
        return this._client.createMobileApplications({ mobileApplications });
    }
    async getMobileApplicationsByStatement(filterStatement) {
        return this._client.getMobileApplicationsByStatement({
            filterStatement,
        });
    }
    async performMobileApplicationAction(mobileApplicationAction, filterStatement) {
        return this._client.performMobileApplicationAction({
            mobileApplicationAction: {
                attributes: {
                    "xsi:type": mobileApplicationAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateMobileApplications(mobileApplications) {
        return this._client.updateMobileApplications({ mobileApplications });
    }
}
exports.MobileApplicationService = MobileApplicationService;
