// @guardian/google-admanager-api/dist/client/services/mobileApplication/mobileApplication.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileApplicationPlatform = exports.MobileApplicationStore = void 0;
/**
 * A store a {@link https://developers.google.com/ad-manager/api/reference/v202505/MobileApplicationService.MobileApplication MobileApplication} is available on.
 */
var MobileApplicationStore;
(function (MobileApplicationStore) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    MobileApplicationStore["UNKNOWN"] = "UNKNOWN";
    MobileApplicationStore["APPLE_ITUNES"] = "APPLE_ITUNES";
    MobileApplicationStore["GOOGLE_PLAY"] = "GOOGLE_PLAY";
    MobileApplicationStore["ROKU"] = "ROKU";
    MobileApplicationStore["AMAZON_FIRETV"] = "AMAZON_FIRETV";
    MobileApplicationStore["PLAYSTATION"] = "PLAYSTATION";
    MobileApplicationStore["XBOX"] = "XBOX";
    MobileApplicationStore["SAMSUNG_TV"] = "SAMSUNG_TV";
    MobileApplicationStore["AMAZON_APP_STORE"] = "AMAZON_APP_STORE";
    MobileApplicationStore["OPPO_APP_STORE"] = "OPPO_APP_STORE";
    MobileApplicationStore["SAMSUNG_APP_STORE"] = "SAMSUNG_APP_STORE";
    MobileApplicationStore["VIVO_APP_STORE"] = "VIVO_APP_STORE";
    MobileApplicationStore["XIAOMI_APP_STORE"] = "XIAOMI_APP_STORE";
})(MobileApplicationStore || (exports.MobileApplicationStore = MobileApplicationStore = {}));
/**
 * A platform a {@link https://developers.google.com/ad-manager/api/reference/v202505/MobileApplicationService.MobileApplication MobileApplication} can run on.
 */
var MobileApplicationPlatform;
(function (MobileApplicationPlatform) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    MobileApplicationPlatform["UNKNOWN"] = "UNKNOWN";
    MobileApplicationPlatform["ANDROID"] = "ANDROID";
    MobileApplicationPlatform["IOS"] = "IOS";
    MobileApplicationPlatform["ROKU"] = "ROKU";
    MobileApplicationPlatform["AMAZON_FIRETV"] = "AMAZON_FIRETV";
    MobileApplicationPlatform["PLAYSTATION"] = "PLAYSTATION";
    MobileApplicationPlatform["XBOX"] = "XBOX";
    MobileApplicationPlatform["SAMSUNG_TV"] = "SAMSUNG_TV";
})(MobileApplicationPlatform || (exports.MobileApplicationPlatform = MobileApplicationPlatform = {}));
