// @guardian/google-admanager-api/dist/client/services/creativeTemplate/creativeTemplate.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreativeTemplateType = exports.CreativeTemplateStatus = exports.AssetCreativeTemplateVariableMimeType = void 0;
/**
 * Different mime type that the asset variable supports.
 */
var AssetCreativeTemplateVariableMimeType;
(function (AssetCreativeTemplateVariableMimeType) {
    AssetCreativeTemplateVariableMimeType["JPG"] = "JPG";
    AssetCreativeTemplateVariableMimeType["PNG"] = "PNG";
    AssetCreativeTemplateVariableMimeType["GIF"] = "GIF";
    AssetCreativeTemplateVariableMimeType["SWF"] = "SWF";
})(AssetCreativeTemplateVariableMimeType || (exports.AssetCreativeTemplateVariableMimeType = AssetCreativeTemplateVariableMimeType = {}));
/**
 * Describes status of the creative template
 */
var CreativeTemplateStatus;
(function (CreativeTemplateStatus) {
    /**
     * The CreativeTemplate is active
     */
    CreativeTemplateStatus["ACTIVE"] = "ACTIVE";
    /**
     * The CreativeTemplate is inactive. Users cannot create new creatives from this template, but existing ones can be edited and continue to serve
     */
    CreativeTemplateStatus["INACTIVE"] = "INACTIVE";
    /**
     * The CreativeTemplate is deleted. Creatives created from this CreativeTemplate can no longer serve.
     */
    CreativeTemplateStatus["DELETED"] = "DELETED";
})(CreativeTemplateStatus || (exports.CreativeTemplateStatus = CreativeTemplateStatus = {}));
/**
 * Describes type of the creative template.
 */
var CreativeTemplateType;
(function (CreativeTemplateType) {
    /**
     * Creative templates that Google defines for users to use.
     */
    CreativeTemplateType["SYSTEM_DEFINED"] = "SYSTEM_DEFINED";
    /**
     * Arbitrary creative templates that users can define as they see fit. Such templates are bound to a specific network and can only be used with creatives being created under the network.
     */
    CreativeTemplateType["USER_DEFINED"] = "USER_DEFINED";
})(CreativeTemplateType || (exports.CreativeTemplateType = CreativeTemplateType = {}));
