// @guardian/google-admanager-api/dist/client/services/customTargeting/customTargeting.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomTargetingService = void 0;
class CustomTargetingService {
    constructor(client) {
        this._client = client;
    }
    async createCustomTargetingKeys(keys) {
        return this._client.createCustomTargetingKeys({ keys });
    }
    async createCustomTargetingValues(values) {
        return this._client.createCustomTargetingValues({ values });
    }
    async getCustomTargetingKeysByStatement(filterStatement) {
        return this._client.getCustomTargetingKeysByStatement({
            filterStatement,
        });
    }
    async getCustomTargetingValuesByStatement(filterStatement) {
        return this._client.getCustomTargetingValuesByStatement({
            filterStatement,
        });
    }
    async performCustomTargetingKeyAction(customTargetingKeyAction, filterStatement) {
        return this._client.performCustomTargetingKeyAction({
            customTargetingKeyAction: {
                attributes: {
                    "xsi:type": customTargetingKeyAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async performCustomTargetingValueAction(customTargetingValueAction, filterStatement) {
        return this._client.performCustomTargetingValueAction({
            customTargetingValueAction: {
                attributes: {
                    "xsi:type": customTargetingValueAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateCustomTargetingKeys(keys) {
        return this._client.updateCustomTargetingKeys({ keys });
    }
    async updateCustomTargetingValues(values) {
        return this._client.updateCustomTargetingValues({ values });
    }
}
exports.CustomTargetingService = CustomTargetingService;
