// @guardian/google-admanager-api/dist/client/services/customTargeting/customTargeting.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomTargetingValueStatusEnum = exports.CustomTargetingValueStatus = exports.CustomTargetingValueMatchTypeEnum = exports.CustomTargetingValueMatchType = exports.ReportableType = exports.CustomTargetingKeyStatus = exports.ValueDeclarationType = exports.CustomTargetingKeyType = void 0;
const superstruct_1 = require("superstruct");
/**
 * Specifies the types for CustomTargetingKey objects.
 */
var CustomTargetingKeyType;
(function (CustomTargetingKeyType) {
    /**
     * Target audiences by criteria values that are defined in advance.
     */
    CustomTargetingKeyType["PREDEFINED"] = "PREDEFINED";
    /**
     * Target audiences by adding criteria values when creating line items.
     */
    CustomTargetingKeyType["FREEFORM"] = "FREEFORM";
})(CustomTargetingKeyType || (exports.CustomTargetingKeyType = CustomTargetingKeyType = {}));
/**
 * Enum to represent the type of values that will be passed for this key.
 */
var ValueDeclarationType;
(function (ValueDeclarationType) {
    /**
     * Values are not user-related data and will be used in all ad requests.
     */
    ValueDeclarationType["NON_USER_DATA"] = "NON_USER_DATA";
    /**
     * Values are declared as user data that will not be used in sensitive ad requests' targeting.
     */
    ValueDeclarationType["USER_DATA"] = "USER_DATA";
    /**
     * Values are declared as user identifiers that will not be used in targeting and will be encrypted in DT using a publisher-provided public key.
     */
    ValueDeclarationType["USER_ID"] = "USER_ID";
})(ValueDeclarationType || (exports.ValueDeclarationType = ValueDeclarationType = {}));
/**
 * Describes the statuses for CustomTargetingKey objects.
 */
var CustomTargetingKeyStatus;
(function (CustomTargetingKeyStatus) {
    /**
     * The object is active.
     */
    CustomTargetingKeyStatus["ACTIVE"] = "ACTIVE";
    /**
     * The object is no longer active.
     */
    CustomTargetingKeyStatus["INACTIVE"] = "INACTIVE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CustomTargetingKeyStatus["UNKNOWN"] = "UNKNOWN";
})(CustomTargetingKeyStatus || (exports.CustomTargetingKeyStatus = CustomTargetingKeyStatus = {}));
/**
 * Represents the reportable state of a custom key.
 */
var ReportableType;
(function (ReportableType) {
    ReportableType["UNKNOWN"] = "UNKNOWN";
    /**
     * Available for reporting in the Ad Manager query tool.
     */
    ReportableType["ON"] = "ON";
    /**
     * Not available for reporting in the Ad Manager query tool.
     */
    ReportableType["OFF"] = "OFF";
    /**
     * Custom dimension available for reporting in the AdManager query tool.
     */
    ReportableType["CUSTOM_DIMENSION"] = "CUSTOM_DIMENSION";
})(ReportableType || (exports.ReportableType = ReportableType = {}));
/**
 * Represents the ways in which
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingValue#name CustomTargetingValue.name} strings will be matched with ad requests.
 */
var CustomTargetingValueMatchType;
(function (CustomTargetingValueMatchType) {
    /**
     * Used for exact matching. For example, the targeting value car=honda will only match to the ad request car=honda.
     */
    CustomTargetingValueMatchType["EXACT"] = "EXACT";
    /**
     * Used for lenient matching when at least one of the words in the ad request matches the targeted value.
     * The targeting value car=honda will match to ad requests containing the word honda.
     * So ad requests car=honda or car=honda civic or car=buy honda or car=how much does a honda cost will all have the line item delivered.
     *
     * This match type can not be used within an audience segment rule.
     */
    CustomTargetingValueMatchType["BROAD"] = "BROAD";
    /**
     * Used for 'starts with' matching when the first few characters in the ad request match all of the characters in the targeted value. The targeting value car=honda will match to ad requests car=honda or car=hondas for sale but not to car=I want a honda.
     */
    CustomTargetingValueMatchType["PREFIX"] = "PREFIX";
    /**
     * This is a combination of MatchType#BROAD and MatchType#PREFIX matching.
     * The targeting value car=honda will match to ad requests that contain words that start with the characters in the targeted value, for example with car=civic hondas.
     *
     * This match type can not be used within an audience segment rule.
     */
    CustomTargetingValueMatchType["BROAD_PREFIX"] = "BROAD_PREFIX";
    /**
     * Used for 'ends with' matching when the last characters in the ad request match all of the characters in the targeted value.
     * The targeting value car=honda will match with ad requests car=honda or car=I want a honda but not to car=hondas for sale.
     *
     * This match type can not be used within line item targeting.
     */
    CustomTargetingValueMatchType["SUFFIX"] = "SUFFIX";
    /**
     * Used for 'within' matching when the string in the ad request contains the string in the targeted value.
     * The targeting value car=honda will match with ad requests car=honda, car=I want a honda, and also with car=hondas for sale, but not with car=misspelled hond a.
     *
     * This match type can not be used within line item targeting.
     */
    CustomTargetingValueMatchType["CONTAINS"] = "CONTAINS";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CustomTargetingValueMatchType["UNKNOWN"] = "UNKNOWN";
})(CustomTargetingValueMatchType || (exports.CustomTargetingValueMatchType = CustomTargetingValueMatchType = {}));
/**
 * Represents a CustomTargetingValueMatchType enum in superstruct format.
 */
exports.CustomTargetingValueMatchTypeEnum = (0, superstruct_1.enums)([
    "EXACT",
    "BROAD",
    "PREFIX",
    "BROAD_PREFIX",
    "SUFFIX",
    "CONTAINS",
    "UNKNOWN",
]);
/**
 * Describes the statuses for CustomTargetingValue objects.
 */
var CustomTargetingValueStatus;
(function (CustomTargetingValueStatus) {
    /**
     * The object is active.
     */
    CustomTargetingValueStatus["ACTIVE"] = "ACTIVE";
    /**
     * The object is no longer active.
     */
    CustomTargetingValueStatus["INACTIVE"] = "INACTIVE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CustomTargetingValueStatus["UNKNOWN"] = "UNKNOWN";
})(CustomTargetingValueStatus || (exports.CustomTargetingValueStatus = CustomTargetingValueStatus = {}));
/**
 * Represents a CustomTargetingValueStatus enum in superstruct format.
 */
exports.CustomTargetingValueStatusEnum = (0, superstruct_1.enums)([
    "ACTIVE",
    "INACTIVE",
    "UNKNOWN",
]);
