// @guardian/google-admanager-api/dist/client/services/customTargeting/customTargeting.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteCustomTargetingValues = exports.ActivateCustomTargetingValues = exports.CustomTargetingValueAction = exports.DeleteCustomTargetingKeys = exports.ActivateCustomTargetingKeys = exports.CustomTargetingKeyAction = void 0;
/**
 * Represents the actions that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingKey CustomTargetingKey} objects.
 */
class CustomTargetingKeyAction {
}
exports.CustomTargetingKeyAction = CustomTargetingKeyAction;
/**
 * The action used for activating inactive (i.e. deleted)
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingKey CustomTargetingKey} objects.
 */
class ActivateCustomTargetingKeys {
}
exports.ActivateCustomTargetingKeys = ActivateCustomTargetingKeys;
/**
 * Represents the delete action that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingKey CustomTargetingKey} objects.
 */
class DeleteCustomTargetingKeys {
}
exports.DeleteCustomTargetingKeys = DeleteCustomTargetingKeys;
/**
 * Represents the actions that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingValue CustomTargetingValue} objects.
 */
class CustomTargetingValueAction {
}
exports.CustomTargetingValueAction = CustomTargetingValueAction;
/**
 * The action used for activating inactive (i.e. deleted)
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingValue CustomTargetingValue} objects.
 */
class ActivateCustomTargetingValues {
}
exports.ActivateCustomTargetingValues = ActivateCustomTargetingValues;
/**
 * Represents the delete action that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/CustomTargetingService.CustomTargetingValue CustomTargetingValue} objects.
 */
class DeleteCustomTargetingValues {
}
exports.DeleteCustomTargetingValues = DeleteCustomTargetingValues;
