// @guardian/google-admanager-api/dist/client/services/company/company.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanyService = void 0;
class CompanyService {
    constructor(client) {
        this._client = client;
    }
    async createCompanies(companies) {
        return this._client.createCompanies({ companies });
    }
    async getCompaniesByStatement(filterStatement) {
        return this._client.getCompaniesByStatement({
            filterStatement,
        });
    }
    async performCompanyAction(companyAction, filterStatement) {
        console.log(companyAction.constructor.name);
        return this._client.performCompanyAction({
            companyAction: Object.assign({ attributes: {
                    "xsi:type": companyAction.constructor.name,
                } }, companyAction.buildAttributes()),
            statement: filterStatement,
        });
    }
    async updateCompanies(companies) {
        return this._client.updateCompanies({ companies });
    }
}
exports.CompanyService = CompanyService;
