// @guardian/google-admanager-api/dist/client/services/company/company.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResendInvitationAction = exports.ReInviteAction = exports.EndAgreementAction = exports.CompanyAction = void 0;
/**
 * Represents the actions that can be performed on `Company` objects.
 */
class CompanyAction {
    buildAttributes() {
        return {};
    }
}
exports.CompanyAction = CompanyAction;
/**
 * The action used by the parent network to withdraw from being the MCM parent for a child.
 */
class EndAgreementAction {
    buildAttributes() {
        return {};
    }
}
exports.EndAgreementAction = EndAgreementAction;
/**
 * The action used by the parent network to send a new invitation with a potentially updated proposal to a rejected or withdrawn child publisher.
 */
class ReInviteAction {
    get proposedDelegationType() {
        return this.proposedDelegationTypeField;
    }
    set proposedDelegationType(value) {
        this.proposedDelegationTypeField = value;
        this.proposedDelegationTypeSpecified = true;
    }
    get proposedDelegationTypeSpecified() {
        return this.proposedDelegationTypeFieldSpecified;
    }
    set proposedDelegationTypeSpecified(value) {
        this.proposedDelegationTypeFieldSpecified = value;
    }
    get proposedRevenueShareMillipercent() {
        return this.proposedRevenueShareMillipercentField;
    }
    set proposedRevenueShareMillipercent(value) {
        this.proposedRevenueShareMillipercentField = value;
        this.proposedRevenueShareMillipercentSpecified = true;
    }
    get proposedRevenueShareMillipercentSpecified() {
        return this.proposedRevenueShareMillipercentFieldSpecified;
    }
    set proposedRevenueShareMillipercentSpecified(value) {
        this.proposedRevenueShareMillipercentFieldSpecified = value;
    }
    get proposedEmail() {
        return this.proposedEmailField;
    }
    set proposedEmail(value) {
        this.proposedEmailField = value;
    }
    buildAttributes() {
        const attributes = {};
        if (this.proposedDelegationTypeSpecified) {
            attributes["proposedDelegationType"] = this.proposedDelegationType;
        }
        if (this.proposedRevenueShareMillipercentSpecified) {
            attributes["proposedRevenueShareMillipercent"] =
                this.proposedRevenueShareMillipercent;
        }
        if (this.proposedEmail) {
            attributes["proposedEmail"] = this.proposedEmail;
        }
        return attributes;
    }
}
exports.ReInviteAction = ReInviteAction;
/**
 * The action used by the parent network to resend an invitation email with the same proposal to an expired child publisher.
 */
class ResendInvitationAction {
    buildAttributes() {
        return {};
    }
}
exports.ResendInvitationAction = ResendInvitationAction;
