// @guardian/google-admanager-api/dist/client/services/team/team.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamAccessType = exports.TeamStatus = void 0;
/**
 * Represents the status of a team, whether it is active or inactive.
 */
var TeamStatus;
(function (TeamStatus) {
    /**
     * The status of an active team. (i.e. visible in the UI)
     */
    TeamStatus["ACTIVE"] = "ACTIVE";
    /**
     * The status of an inactive team. (i.e. hidden in the UI)
     */
    TeamStatus["INACTIVE"] = "INACTIVE";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    TeamStatus["UNKNOWN"] = "UNKNOWN";
})(TeamStatus || (exports.TeamStatus = TeamStatus = {}));
/**
 * Represents the types of team access supported for orders.
 */
var TeamAccessType;
(function (TeamAccessType) {
    /**
     * The level of access in which team members cannot view or edit a team's orders.
     */
    TeamAccessType["NONE"] = "NONE";
    /**
     * The level of access in which team members can only view a team's orders.
     */
    TeamAccessType["READ_ONLY"] = "READ_ONLY";
    /**
     * The level of access in which team members can view and edit a team's orders.
     */
    TeamAccessType["READ_WRITE"] = "READ_WRITE";
})(TeamAccessType || (exports.TeamAccessType = TeamAccessType = {}));
