// @guardian/google-admanager-api/dist/client/services/order/order.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderStatus = void 0;
/**
 * Describes the order statuses.
 */
var OrderStatus;
(function (OrderStatus) {
    /**
     * Indicates that the {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order}
     * has just been created but no approval has been requested yet.
     */
    OrderStatus["DRAFT"] = "DRAFT";
    /**
     * Indicates that a request for approval for the Order has been made.
     */
    OrderStatus["PENDING_APPROVAL"] = "PENDING_APPROVAL";
    /**
     * Indicates that the {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order}
     * has been approved and is ready to serve.
     */
    OrderStatus["APPROVED"] = "APPROVED";
    /**
     * Indicates that the {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order}
     * has been disapproved and is not eligible to serve.
     */
    OrderStatus["DISAPPROVED"] = "DISAPPROVED";
    /**
     * This is a legacy state. Paused status should be checked on **LineItems**`s within the order.
     */
    OrderStatus["PAUSED"] = "PAUSED";
    /**
     * Indicates that the {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order}
     * has been canceled and cannot serve.
     */
    OrderStatus["CANCELED"] = "CANCELED";
    /**
     * Indicates that the {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order}
     * has been deleted by DSM.
     */
    OrderStatus["DELETED"] = "DELETED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    OrderStatus["UNKNOWN"] = "UNKNOWN";
})(OrderStatus || (exports.OrderStatus = OrderStatus = {}));
