// @guardian/google-admanager-api/dist/client/services/order/order.action.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnarchiveOrders = exports.SubmitOrdersForApprovalWithoutReservationChanges = exports.SubmitOrdersForApproval = exports.RetractOrdersWithoutReservationChanges = exports.RetractOrders = exports.ResumeOrders = exports.PauseOrders = exports.DisapproveOrdersWithoutReservationChanges = exports.DisapproveOrders = exports.DeleteOrders = exports.ArchiveOrders = exports.ApproveOrdersWithoutReservationChanges = exports.ApproveOrders = exports.OrderAction = void 0;
/**
 * Represents the actions that can be performed on
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order} objects.
 */
class OrderAction {
    buildAttributes() {
        return {};
    }
}
exports.OrderAction = OrderAction;
/**
 * The action used for approving
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order} objects.
 * All {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem LineItem}
 * objects within the order will be approved as well.
 * For more information on what happens to an order and its line items when it is approved,
 * see the {@link https://support.google.com/dfp_premium/answer/177334 Ad Manager Help Center}.
 */
class ApproveOrders {
    constructor() {
        this.skipInventoryCheckField = false;
    }
    get skipInventoryCheck() {
        return this.skipInventoryCheckField;
    }
    set skipInventoryCheck(value) {
        this.skipInventoryCheckField = value;
    }
    buildAttributes() {
        return { skipInventoryCheck: this.skipInventoryCheck };
    }
}
exports.ApproveOrders = ApproveOrders;
/**
 * The action used for approving
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/OrderService.Order Order} objects.
 * All {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem LineItem}
 * objects within the order will be approved as well.
 * This action does not make any changes to the
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem#reservationStatus LineItem.reservationStatus}
 * of the line items within the order.
 * If there are reservable line items that have not been reserved the operation will not succeed.
 */
class ApproveOrdersWithoutReservationChanges {
    buildAttributes() {
        return {};
    }
}
exports.ApproveOrdersWithoutReservationChanges = ApproveOrdersWithoutReservationChanges;
/**
 * The action used for archiving Order objects.
 */
class ArchiveOrders {
    buildAttributes() {
        return {};
    }
}
exports.ArchiveOrders = ArchiveOrders;
/**
 * The action used for deleting Order objects.
 * All line items within that order are also deleted.
 * Orders can only be deleted if none of its line items have been eligible to serve.
 * This action can be used to delete proposed orders and line items if they are no longer valid.
 */
class DeleteOrders {
    buildAttributes() {
        return {};
    }
}
exports.DeleteOrders = DeleteOrders;
/**
 * The action used for disapproving Order objects.
 * All {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem LineItem} objects within the order will be disapproved as well.
 */
class DisapproveOrders {
    buildAttributes() {
        return {};
    }
}
exports.DisapproveOrders = DisapproveOrders;
/**
 * The action used for disapproving Order objects.
 * All {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem LineItem} objects within the order will be disapproved as well.
 * This action does not make any changes to the
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem#reservationStatus LineItem.reservationStatus}
 * of the line items within the order.
 */
class DisapproveOrdersWithoutReservationChanges {
    buildAttributes() {
        return {};
    }
}
exports.DisapproveOrdersWithoutReservationChanges = DisapproveOrdersWithoutReservationChanges;
/**
 * The action used for pausing all {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem LineItem} objects within an order.
 */
class PauseOrders {
    buildAttributes() {
        return {};
    }
}
exports.PauseOrders = PauseOrders;
/**
 * The action used for resuming Order objects.
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem LineItem} objects within the order that are legible to resume will resume as well.
 */
class ResumeOrders {
    buildAttributes() {
        return {};
    }
}
exports.ResumeOrders = ResumeOrders;
/**
 * The action used for retracting Order objects.
 */
class RetractOrders {
    buildAttributes() {
        return {};
    }
}
exports.RetractOrders = RetractOrders;
/**
 * The action used for retracting Order objects.
 * This action does not make any changes to the
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem#reservationStatus LineItem.reservationStatus} of the line items within the order.
 */
class RetractOrdersWithoutReservationChanges {
    buildAttributes() {
        return {};
    }
}
exports.RetractOrdersWithoutReservationChanges = RetractOrdersWithoutReservationChanges;
/**
 * The action used for submitting Order objects for approval.
 */
class SubmitOrdersForApproval {
    buildAttributes() {
        return {};
    }
}
exports.SubmitOrdersForApproval = SubmitOrdersForApproval;
/**
 * The action used for submitting Order objects for approval.
 * This action does not make any changes to the {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.LineItem#reservationStatus LineItem.reservationStatus}
 * of the line items within the order.
 */
class SubmitOrdersForApprovalWithoutReservationChanges {
    buildAttributes() {
        return {};
    }
}
exports.SubmitOrdersForApprovalWithoutReservationChanges = SubmitOrdersForApprovalWithoutReservationChanges;
/**
 * The action used for unarchiving Order objects.
 */
class UnarchiveOrders {
    buildAttributes() {
        return {};
    }
}
exports.UnarchiveOrders = UnarchiveOrders;
