// @guardian/google-admanager-api/dist/client/services/userTeamAssociation/userTeamAssociation.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserTeamAssociationService = void 0;
class UserTeamAssociationService {
    constructor(client) {
        this._client = client;
    }
    async createUserTeamAssociations(userTeamAssociations) {
        return this._client.createUserTeamAssociations({ userTeamAssociations });
    }
    async getUserTeamAssociationsByStatement(filterStatement) {
        return this._client.getUserTeamAssociationsByStatement({
            filterStatement,
        });
    }
    async performUserTeamAssociationAction(userTeamAssociationAction, filterStatement) {
        return this._client.performUserTeamAssociationAction({
            userTeamAssociationAction: {
                attributes: {
                    "xsi:type": userTeamAssociationAction.constructor.name,
                },
            },
            filterStatement,
        });
    }
    async updateUserTeamAssociations(userTeamAssociations) {
        return this._client.updateUserTeamAssociations({ userTeamAssociations });
    }
}
exports.UserTeamAssociationService = UserTeamAssociationService;
