// @guardian/google-admanager-api/dist/client/common/enums/creativePlaceholder.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreativeSizeTypeEnum = exports.CreativeSizeType = void 0;
const superstruct_1 = require("superstruct");
/**
 * Descriptions of the types of sizes a creative can be. Not all creatives can be described by a height-width pair, this provides additional context.
 */
var CreativeSizeType;
(function (CreativeSizeType) {
    /**
     * Dimension based size, an actual height and width.
     */
    CreativeSizeType["PIXEL"] = "PIXEL";
    /**
     * Mobile size, that is expressed as a ratio of say 4 by 1, that could be met by a 100 x 25 sized image.
     */
    CreativeSizeType["ASPECT_RATIO"] = "ASPECT_RATIO";
    /**
     * Out-of-page size, that is not related to the slot it is served. But rather is a function of the snippet, and the values set. This must be used with 1x1 size.
     */
    CreativeSizeType["INTERSTITIAL"] = "INTERSTITIAL";
    /**
     * Size has no meaning
     *
     * 1. For Click Tracking entities, where size doesn't matter 2. For entities that allow all requested sizes, where the size represents all sizes.
     */
    CreativeSizeType["IGNORED"] = "IGNORED";
    /**
     * Native size, which is a function of the how the client renders the creative. This must be used with 1x1 size.
     */
    CreativeSizeType["NATIVE"] = "NATIVE";
    /**
     * Audio size. Used with audio ads. This must be used with 1x1 size.
     */
    CreativeSizeType["AUDIO"] = "AUDIO";
})(CreativeSizeType || (exports.CreativeSizeType = CreativeSizeType = {}));
/**
 * Represents a CreativeSizeType enum in superstruct format.
 */
exports.CreativeSizeTypeEnum = (0, superstruct_1.enums)([
    "PIXEL",
    "ASPECT_RATIO",
    "INTERSTITIAL",
    "IGNORED",
    "NATIVE",
    "AUDIO",
]);
