// @guardian/google-admanager-api/dist/client/common/enums/customCriteria.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CmsMetadataCriteriaComparisonOperatorEnum = exports.CmsMetadataCriteriaComparisonOperator = exports.LogicalOperatorEnum = exports.LogicalOperator = exports.ComparisonOperatorEnum = exports.ComparisonOperator = void 0;
const superstruct_1 = require("superstruct");
/**
 * Specifies the available comparison operators.
 */
var ComparisonOperator;
(function (ComparisonOperator) {
    ComparisonOperator["IS"] = "IS";
    ComparisonOperator["IS_NOT"] = "IS_NOT";
})(ComparisonOperator || (exports.ComparisonOperator = ComparisonOperator = {}));
/**
 * Represents a ComparisonOperator enum in superstruct format.
 */
exports.ComparisonOperatorEnum = (0, superstruct_1.enums)([
    "IS",
    "IS_NOT",
]);
/**
 * Specifies the available logical operators.
 */
var LogicalOperator;
(function (LogicalOperator) {
    LogicalOperator["AND"] = "AND";
    LogicalOperator["OR"] = "OR";
})(LogicalOperator || (exports.LogicalOperator = LogicalOperator = {}));
/**
 * Represents a LogicalOperator enum in superstruct format.
 */
exports.LogicalOperatorEnum = (0, superstruct_1.enums)([
    "AND",
    "OR",
]);
/**
 * Specifies the available comparison operators.
 */
var CmsMetadataCriteriaComparisonOperator;
(function (CmsMetadataCriteriaComparisonOperator) {
    CmsMetadataCriteriaComparisonOperator["EQUALS"] = "EQUALS";
    CmsMetadataCriteriaComparisonOperator["NOT_EQUALS"] = "NOT_EQUALS";
})(CmsMetadataCriteriaComparisonOperator || (exports.CmsMetadataCriteriaComparisonOperator = CmsMetadataCriteriaComparisonOperator = {}));
/**
 * Represents a CmsMetadataCriteriaComparisonOperator enum in superstruct format.
 */
exports.CmsMetadataCriteriaComparisonOperatorEnum = (0, superstruct_1.enums)([
    "EQUALS",
    "NOT_EQUALS",
]);
