// @guardian/google-admanager-api/dist/client/common/enums/delivery.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompanionDeliveryOptionEnum = exports.CompanionDeliveryOption = exports.DeliveryForecastSourceEnum = exports.DeliveryForecastSource = exports.DeliveryRateTypeEnum = exports.DeliveryRateType = exports.RequestPlatformEnum = exports.RequestPlatform = exports.DeliveryTimeZoneEnum = exports.DeliveryTimeZone = void 0;
const superstruct_1 = require("superstruct");
/**
 * Represents the time zone to be used for DayPartTargeting.
 */
var DeliveryTimeZone;
(function (DeliveryTimeZone) {
    /**
     * 	Use the time zone of the publisher.
     */
    DeliveryTimeZone["PUBLISHER"] = "PUBLISHER";
    /**
     * 	Use the time zone of the browser.
     */
    DeliveryTimeZone["BROWSER"] = "BROWSER";
})(DeliveryTimeZone || (exports.DeliveryTimeZone = DeliveryTimeZone = {}));
/**
 * Represents a DeliveryTimeZone enum in superstruct format.
 */
exports.DeliveryTimeZoneEnum = (0, superstruct_1.enums)([
    "PUBLISHER",
    "BROWSER",
]);
/**
 * Represents the platform which requests and renders the ad.
 */
var RequestPlatform;
(function (RequestPlatform) {
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    RequestPlatform["UNKNOWN"] = "UNKNOWN";
    /**
     * 	Represents a request made from a web browser. This includes both desktop and mobile web.
     */
    RequestPlatform["BROWSER"] = "BROWSER";
    /**
     * 	Represents a request made from a mobile application. This includes mobile app interstitial and rewarded video requests.
     */
    RequestPlatform["MOBILE_APP"] = "MOBILE_APP";
    /**
     * 	Represents a request made from a video player that is playing publisher content. This includes video players embedded in web pages and mobile applications, and connected TV screens.
     */
    RequestPlatform["VIDEO_PLAYER"] = "VIDEO_PLAYER";
})(RequestPlatform || (exports.RequestPlatform = RequestPlatform = {}));
/**
 * Represents a RequestPlatform enum in superstruct format.
 */
exports.RequestPlatformEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "BROWSER",
    "MOBILE_APP",
    "VIDEO_PLAYER",
]);
/**
 * Possible delivery rates for a LineItem, which dictate the manner in which they are served.
 */
var DeliveryRateType;
(function (DeliveryRateType) {
    /**
     * 	Line items are served as evenly as possible across the number of days specified in a line item's LineItem.duration.
     */
    DeliveryRateType["EVENLY"] = "EVENLY";
    /**
     * 	Line items are served more aggressively in the beginning of the flight date.
     */
    DeliveryRateType["FRONTLOADED"] = "FRONTLOADED";
    /**
     * 	The booked impressions for a line item may be delivered well before the LineItem.endDateTime. Other lower-priority or lower-value line items will be stopped from delivering until this line item meets the number of impressions or clicks it is booked for.
     */
    DeliveryRateType["AS_FAST_AS_POSSIBLE"] = "AS_FAST_AS_POSSIBLE";
})(DeliveryRateType || (exports.DeliveryRateType = DeliveryRateType = {}));
/**
 * Represents a DeliveryRateType enum in superstruct format.
 */
exports.DeliveryRateTypeEnum = (0, superstruct_1.enums)([
    "EVENLY",
    "FRONTLOADED",
    "AS_FAST_AS_POSSIBLE",
]);
/**
 * Strategies for choosing forecasted traffic shapes to pace line items.
 */
var DeliveryForecastSource;
(function (DeliveryForecastSource) {
    /**
     * 	The line item's historical traffic shape will be used to pace line item delivery.
     */
    DeliveryForecastSource["HISTORICAL"] = "HISTORICAL";
    /**
     * 	The line item's projected future traffic will be used to pace line item delivery.
     */
    DeliveryForecastSource["FORECASTING"] = "FORECASTING";
    /**
     * 	A user specified custom pacing curve will be used to pace line item delivery.
     */
    DeliveryForecastSource["CUSTOM_PACING_CURVE"] = "CUSTOM_PACING_CURVE";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    DeliveryForecastSource["UNKNOWN"] = "UNKNOWN";
})(DeliveryForecastSource || (exports.DeliveryForecastSource = DeliveryForecastSource = {}));
/**
 * Represents a DeliveryForecastSource enum in superstruct format.
 */
exports.DeliveryForecastSourceEnum = (0, superstruct_1.enums)([
    "HISTORICAL",
    "FORECASTING",
    "CUSTOM_PACING_CURVE",
    "UNKNOWN",
]);
/**
 * The delivery option for companions.
 * Used for line items whose environmentType is {@link https://developers.google.com/ad-manager/api/reference/v202505/ProposalLineItemService.EnvironmentType#VIDEO_PLAYER EnvironmentType.VIDEO_PLAYER}.
 */
var CompanionDeliveryOption;
(function (CompanionDeliveryOption) {
    /**
     * Companions are not required to serve a creative set. The creative set can serve to inventory that has zero or more matching companions.
     */
    CompanionDeliveryOption["OPTIONAL"] = "OPTIONAL";
    /**
     * At least one companion must be served in order for the creative set to be used.
     */
    CompanionDeliveryOption["AT_LEAST_ONE"] = "AT_LEAST_ONE";
    /**
     * All companions in the set must be served in order for the creative set to be used. This can still serve to inventory that has more companions than can be filled.
     */
    CompanionDeliveryOption["ALL"] = "ALL";
    /**
     * The delivery type is unknown.
     */
    CompanionDeliveryOption["UNKNOWN"] = "UNKNOWN";
})(CompanionDeliveryOption || (exports.CompanionDeliveryOption = CompanionDeliveryOption = {}));
/**
 * Represents a CompanionDeliveryOption enum in superstruct format.
 */
exports.CompanionDeliveryOptionEnum = (0, superstruct_1.enums)([
    "OPTIONAL",
    "AT_LEAST_ONE",
    "ALL",
    "UNKNOWN",
]);
