// @guardian/google-admanager-api/dist/client/common/enums/general.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnitTypeEnum = exports.UnitType = exports.GoalTypeEnum = exports.GoalType = exports.ProgrammaticCreativeSourceEnum = exports.ProgrammaticCreativeSource = exports.StatusEnum = exports.ComputedStatus = exports.AllowedFormatsEnum = exports.AllowedFormats = exports.LineItemTypeEnum = exports.LineItemType = exports.SkippableAdTypeEnum = exports.SkippableAdType = exports.RoadblockingTypeEnum = exports.RoadblockingType = exports.EnvironmentTypeEnum = exports.EnvironmentType = exports.OnboardingTask = exports.AccountStatus = exports.DeclarationType = void 0;
const superstruct_1 = require("superstruct");
/**
 * The declaration about third party data usage on the associated entity.
 */
var DeclarationType;
(function (DeclarationType) {
    /**
     * There are no companies associated. Functionally the same as DECLARED, combined with an empty company list.
     */
    DeclarationType["NONE"] = "NONE";
    /**
     * There is a set of RichMediaAdsCompanys associated with this entity.
     */
    DeclarationType["DECLARED"] = "DECLARED";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    DeclarationType["UNKNOWN"] = "UNKNOWN";
})(DeclarationType || (exports.DeclarationType = DeclarationType = {}));
/**
 * Status of the MCM child publisher's Ad Manager account with respect to delegated serving.
 * In order for the child network to be served ads for MCM, it must have accepted the invite from the parent network, and must have passed Google's policy compliance verifications.
 */
var AccountStatus;
(function (AccountStatus) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AccountStatus["UNKNOWN"] = "UNKNOWN";
    /**
     * The child publisher has not acted on the invite from the parent.
     */
    AccountStatus["INVITED"] = "INVITED";
    /**
     * The child publisher has declined the invite.
     */
    AccountStatus["DECLINED"] = "DECLINED";
    /**
     * The child publisher has accepted the invite, and is awaiting Google's policy compliance verifications.
     */
    AccountStatus["PENDING_GOOGLE_APPROVAL"] = "PENDING_GOOGLE_APPROVAL";
    /**
     * The child publisher accepted the invite, and Google found it to be compliant with its policies, i.e. no policy violations were found, and the child publisher can be served ads.
     */
    AccountStatus["APPROVED"] = "APPROVED";
    /**
     * The child publisher accepted the invite, but was disapproved by Google for violating its policies.
     */
    AccountStatus["CLOSED_POLICY_VIOLATION"] = "CLOSED_POLICY_VIOLATION";
    /**
     * The child publisher accepted the invite, but was disapproved by Google for invalid activity.
     */
    AccountStatus["CLOSED_INVALID_ACTIVITY"] = "CLOSED_INVALID_ACTIVITY";
    /**
     * The child publisher has closed their own account.
     */
    AccountStatus["CLOSED_BY_PUBLISHER"] = "CLOSED_BY_PUBLISHER";
    /**
     * The child publisher accepted the invite, but was disapproved as ineligible by Google.
     */
    AccountStatus["DISAPPROVED_INELIGIBLE"] = "DISAPPROVED_INELIGIBLE";
    /**
     * The child publisher accepted the invite, but was disapproved by Google for being a duplicate of another account.
     */
    AccountStatus["DISAPPROVED_DUPLICATE_ACCOUNT"] = "DISAPPROVED_DUPLICATE_ACCOUNT";
    /**
     * The invite was sent to the child publisher more than 90 days ago, due to which it has been deactivated.
     */
    AccountStatus["EXPIRED"] = "EXPIRED";
    /**
     * Either the child publisher disconnected from the parent network, or the parent network withdrew the invite.
     */
    AccountStatus["INACTIVE"] = "INACTIVE";
    /**
     * The association between the parent and child publishers was deactivated by Google Ad Manager.
     */
    AccountStatus["DEACTIVATED_BY_AD_MANAGER"] = "DEACTIVATED_BY_AD_MANAGER";
})(AccountStatus || (exports.AccountStatus = AccountStatus = {}));
/**
 * Pending onboarding tasks for the child publishers that must completed before Google's policy compliance is verified.
 */
var OnboardingTask;
(function (OnboardingTask) {
    OnboardingTask["UNKNOWN"] = "UNKNOWN";
    /**
     * Creation of the child publisher's payments billing profile.
     */
    OnboardingTask["BILLING_PROFILE_CREATION"] = "BILLING_PROFILE_CREATION";
    /**
     * Verification of the child publisher's phone number.
     */
    OnboardingTask["PHONE_PIN_VERIFICATION"] = "PHONE_PIN_VERIFICATION";
    /**
     * Setup of the child publisher's Ad Manager account.
     */
    OnboardingTask["AD_MANAGER_ACCOUNT_SETUP"] = "AD_MANAGER_ACCOUNT_SETUP";
})(OnboardingTask || (exports.OnboardingTask = OnboardingTask = {}));
/**
 * Enum for the valid environments in which ads can be shown.
 */
var EnvironmentType;
(function (EnvironmentType) {
    /**
     * A regular web browser.
     */
    EnvironmentType["BROWSER"] = "BROWSER";
    /**
     * Video players.
     */
    EnvironmentType["VIDEO_PLAYER"] = "VIDEO_PLAYER";
})(EnvironmentType || (exports.EnvironmentType = EnvironmentType = {}));
/**
 * Represents an EnvironmentType enum in superstruct format.
 */
exports.EnvironmentTypeEnum = (0, superstruct_1.enums)([
    "BROWSER",
    "VIDEO_PLAYER",
]);
/**
 * Describes the roadblocking types.
 */
var RoadblockingType;
(function (RoadblockingType) {
    /**
     * Only one creative from a line item can serve at a time.
     */
    RoadblockingType["ONLY_ONE"] = "ONLY_ONE";
    /**
     * Any number of creatives from a line item can serve together at a time.
     */
    RoadblockingType["ONE_OR_MORE"] = "ONE_OR_MORE";
    /**
     * As many creatives from a line item as can fit on a page will serve. This could mean anywhere from one to all of a line item's creatives given the size constraints of ad slots on a page.
     */
    RoadblockingType["AS_MANY_AS_POSSIBLE"] = "AS_MANY_AS_POSSIBLE";
    /**
     * All or none of the creatives from a line item will serve. This option will only work if served to a GPT tag using SRA (single request architecture mode).
     */
    RoadblockingType["ALL_ROADBLOCK"] = "ALL_ROADBLOCK";
    /**
     * A master/companion CreativeSet roadblocking type. A LineItem.creativePlaceholders must be set accordingly.
     */
    RoadblockingType["CREATIVE_SET"] = "CREATIVE_SET";
})(RoadblockingType || (exports.RoadblockingType = RoadblockingType = {}));
/**
 * Represents a RoadblockingType enum in superstruct format.
 */
exports.RoadblockingTypeEnum = (0, superstruct_1.enums)([
    "ONLY_ONE",
    "ONE_OR_MORE",
    "AS_MANY_AS_POSSIBLE",
    "ALL_ROADBLOCK",
    "CREATIVE_SET",
]);
/**
 * The different types of skippable ads.
 */
var SkippableAdType;
(function (SkippableAdType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    SkippableAdType["UNKNOWN"] = "UNKNOWN";
    /**
     * Skippable ad type is disabled.
     */
    SkippableAdType["DISABLED"] = "DISABLED";
    /**
     * Skippable ad type is enabled.
     */
    SkippableAdType["ENABLED"] = "ENABLED";
    /**
     * Skippable in-stream ad type.
     */
    SkippableAdType["INSTREAM_SELECT"] = "INSTREAM_SELECT";
    /**
     * Any skippable or not skippable. This is only for programmatic case when the creative skippability is decided by the buyside.
     */
    SkippableAdType["ANY"] = "ANY";
})(SkippableAdType || (exports.SkippableAdType = SkippableAdType = {}));
/**
 * Represents a SkippableAdType enum in superstruct format.
 */
exports.SkippableAdTypeEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "DISABLED",
    "ENABLED",
    "INSTREAM_SELECT",
    "ANY",
]);
/**
 * LineItemType indicates the priority of a LineItem, determined by the way in which impressions are reserved to be served for it.
 */
var LineItemType;
(function (LineItemType) {
    /**
     * 	The type of LineItem for which a percentage of all the impressions that are being sold are reserved.
     */
    LineItemType["SPONSORSHIP"] = "SPONSORSHIP";
    /**
     * 	The type of LineItem for which a fixed quantity of impressions or clicks are reserved.
     */
    LineItemType["STANDARD"] = "STANDARD";
    /**
     * 	The type of LineItem most commonly used to fill a site's unsold inventory if not contractually obligated to deliver a requested number of impressions. Users specify the daily percentage of unsold impressions or clicks when creating this line item.
     */
    LineItemType["NETWORK"] = "NETWORK";
    /**
     * 	The type of LineItem for which a fixed quantity of impressions or clicks will be delivered at a priority lower than the LineItemType.STANDARD type.
     */
    LineItemType["BULK"] = "BULK";
    /**
     * 	The type of LineItem most commonly used to fill a site's unsold inventory if not contractually obligated to deliver a requested number of impressions. Users specify the fixed quantity of unsold impressions or clicks when creating this line item.
     */
    LineItemType["PRICE_PRIORITY"] = "PRICE_PRIORITY";
    /**
     * 	The type of LineItem typically used for ads that promote products and services chosen by the publisher. These usually do not generate revenue and have the lowest delivery priority.
     */
    LineItemType["HOUSE"] = "HOUSE";
    /**
     * 	Represents a legacy LineItem that has been migrated from the DFP system. Such line items cannot be created any more. Also, these line items cannot be activated or resumed.
     */
    LineItemType["LEGACY_DFP"] = "LEGACY_DFP";
    /**
     * 	The type of LineItem used for ads that track ads being served externally of Ad Manager, for example an email newsletter. The click through would reference this ad, and the click would be tracked via this ad.
     */
    LineItemType["CLICK_TRACKING"] = "CLICK_TRACKING";
    /**
     * 	A LineItem using dynamic allocation backed by AdSense.
     */
    LineItemType["ADSENSE"] = "ADSENSE";
    /**
     * 	A LineItem using dynamic allocation backed by the Google Ad Exchange.
     */
    LineItemType["AD_EXCHANGE"] = "AD_EXCHANGE";
    /**
     * 	Represents a non-monetizable video LineItem that targets one or more bumper positions, which are short house video messages used by publishers to separate content from ad breaks.
     */
    LineItemType["BUMPER"] = "BUMPER";
    /**
     * 	A LineItem using dynamic allocation backed by AdMob.
     */
    LineItemType["ADMOB"] = "ADMOB";
    /**
     * 	The type of LineItem for which there are no impressions reserved, and will serve for a second price bid. All LineItems of type LineItemType.PREFERRED_DEAL should be created via a ProposalLineItem with a matching type. When creating a LineItem of type LineItemType.PREFERRED_DEAL, the ProposalLineItem.estimatedMinimumImpressions field is required.
     */
    LineItemType["PREFERRED_DEAL"] = "PREFERRED_DEAL";
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    LineItemType["UNKNOWN"] = "UNKNOWN";
})(LineItemType || (exports.LineItemType = LineItemType = {}));
/**
 * Represents a LineItemType enum in superstruct format.
 */
exports.LineItemTypeEnum = (0, superstruct_1.enums)([
    "STANDARD",
    "SPONSORSHIP",
    "NETWORK",
    "BULK",
    "PRICE_PRIORITY",
    "HOUSE",
    "LEGACY_DFP",
    "CLICK_TRACKING",
    "ADSENSE",
    "AD_EXCHANGE",
    "BUMPER",
    "ADMOB",
    "PREFERRED_DEAL",
    "UNKNOWN",
]);
/**
 * The formats that a publisher allows on their programmatic
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} or
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/ForecastService.ProposalLineItem ProposalLineItem}.
 */
var AllowedFormats;
(function (AllowedFormats) {
    /**
     * Audio format. This is only relevant for programmatic video line items.
     */
    AllowedFormats["AUDIO"] = "AUDIO";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    AllowedFormats["UNKNOWN"] = "UNKNOWN";
})(AllowedFormats || (exports.AllowedFormats = AllowedFormats = {}));
/**
 * Represents an AllowedFormats enum in superstruct format.
 */
exports.AllowedFormatsEnum = (0, superstruct_1.enums)([
    "AUDIO",
    "UNKNOWN",
]);
/**
 * Describes the computed LineItem status that is derived from the current state of the line item.
 */
var ComputedStatus;
(function (ComputedStatus) {
    /**
     * The LineItem has past its LineItem.endDateTime with an auto extension, but hasn't met its goal.
     */
    ComputedStatus["DELIVERY_EXTENDED"] = "DELIVERY_EXTENDED";
    /**
     * The LineItem has begun serving.
     */
    ComputedStatus["DELIVERING"] = "DELIVERING";
    /**
     * The LineItem has been activated and is ready to serve.
     */
    ComputedStatus["READY"] = "READY";
    /**
     * The LineItem has been paused from serving.
     */
    ComputedStatus["PAUSED"] = "PAUSED";
    /**
     * The LineItem is inactive. It is either caused by missing creatives or the network disabling auto-activation.
     */
    ComputedStatus["INACTIVE"] = "INACTIVE";
    /**
     * The LineItem has been paused and its reserved inventory has been released. The LineItem will not serve.
     */
    ComputedStatus["PAUSED_INVENTORY_RELEASED"] = "PAUSED_INVENTORY_RELEASED";
    /**
     * The LineItem has been submitted for approval.
     */
    ComputedStatus["PENDING_APPROVAL"] = "PENDING_APPROVAL";
    /**
     * The LineItem has completed its run.
     */
    ComputedStatus["COMPLETED"] = "COMPLETED";
    /**
     * The LineItem has been disapproved and is not eligible to serve.
     */
    ComputedStatus["DISAPPROVED"] = "DISAPPROVED";
    /**
     * The LineItem is still being drafted.
     */
    ComputedStatus["DRAFT"] = "DRAFT";
    /**
     * The LineItem has been canceled and is no longer eligible to serve. This is a legacy status imported from Google Ad Manager orders.
     */
    ComputedStatus["CANCELED"] = "CANCELED";
})(ComputedStatus || (exports.ComputedStatus = ComputedStatus = {}));
/**
 * Represents a ComputedStatus enum in superstruct format.
 */
exports.StatusEnum = (0, superstruct_1.enums)([
    "DELIVERY_EXTENDED",
    "DELIVERING",
    "READY",
    "PAUSED",
    "INACTIVE",
    "PAUSED_INVENTORY_RELEASED",
    "PENDING_APPROVAL",
    "COMPLETED",
    "DISAPPROVED",
    "DRAFT",
    "CANCELED",
]);
/**
 * Types of programmatic creative sources.
 */
var ProgrammaticCreativeSource;
(function (ProgrammaticCreativeSource) {
    /**
     * Indicates that the programmatic line item is associated with creatives provided by the publisher.
     */
    ProgrammaticCreativeSource["PUBLISHER"] = "PUBLISHER";
    /**
     * Indicates that the programmatic line item is associated with creatives provided by the advertiser.
     */
    ProgrammaticCreativeSource["ADVERTISER"] = "ADVERTISER";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ProgrammaticCreativeSource["UNKNOWN"] = "UNKNOWN";
})(ProgrammaticCreativeSource || (exports.ProgrammaticCreativeSource = ProgrammaticCreativeSource = {}));
/**
 * Represents a ProgrammaticCreativeSource enum in superstruct format.
 */
exports.ProgrammaticCreativeSourceEnum = (0, superstruct_1.enums)([
    "PUBLISHER",
    "ADVERTISER",
    "UNKNOWN",
]);
/**
 * Specifies the type of the goal for a {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem}.
 */
var GoalType;
(function (GoalType) {
    /**
     * No goal is specified for the number of ads delivered. The LineItem.lineItemType must be one of:
     *
     * - LineItemType.PRICE_PRIORITY
     * - LineItemType.AD_EXCHANGE
     * - LineItemType.CLICK_TRACKING
     */
    GoalType["NONE"] = "NONE";
    /**
     * There is a goal on the number of ads delivered for this line item during its entire lifetime. The LineItem.lineItemType must be one of:
     *
     * - LineItemType.STANDARD
     * - LineItemType.BULK
     * - LineItemType.PRICE_PRIORITY
     * - LineItemType.ADSENSE
     * - LineItemType.AD_EXCHANGE
     * - LineItemType.ADMOB
     * - LineItemType.CLICK_TRACKING
     */
    GoalType["LIFETIME"] = "LIFETIME";
    /**
     * There is a daily goal on the number of ads delivered for this line item. The LineItem.lineItemType must be one of:
     *
     * - LineItemType.SPONSORSHIP
     * - LineItemType.NETWORK
     * - LineItemType.PRICE_PRIORITY
     * - LineItemType.HOUSE
     * - LineItemType.ADSENSE
     * - LineItemType.AD_EXCHANGE
     * - LineItemType.ADMOB
     * - LineItemType.BUMPER
     */
    GoalType["DAILY"] = "DAILY";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    GoalType["UNKNOWN"] = "UNKNOWN";
})(GoalType || (exports.GoalType = GoalType = {}));
/**
 * Represents a GoalType enum in superstruct format.
 */
exports.GoalTypeEnum = (0, superstruct_1.enums)([
    "NONE",
    "LIFETIME",
    "DAILY",
    "UNKNOWN",
]);
/**
 * Indicates the type of unit used for defining a reservation.
 * The {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.CostType CostType} can differ from the
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.UnitType UnitType} - an ad can have an impression goal, but be billed by its click.
 * Usually {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.CostType CostType} and
 * {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.UnitType UnitType} will refer to the same unit.
 */
var UnitType;
(function (UnitType) {
    /**
     * The number of impressions served by creatives associated with the line item. Line items of all LineItemType support this UnitType.
     */
    UnitType["IMPRESSIONS"] = "IMPRESSIONS";
    /**
     * The number of clicks reported by creatives associated with the line item. The LineItem.lineItemType must be LineItemType.STANDARD, LineItemType.BULK or LineItemType.PRICE_PRIORITY.
     */
    UnitType["CLICKS"] = "CLICKS";
    /**
     * The number of click-through Cost-Per-Action (CPA) conversions from creatives associated with the line item. This is only supported as secondary goal and the LineItem.costType must be CostType.CPA.
     */
    UnitType["CLICK_THROUGH_CPA_CONVERSIONS"] = "CLICK_THROUGH_CPA_CONVERSIONS";
    /**
     * The number of view-through Cost-Per-Action (CPA) conversions from creatives associated with the line item. This is only supported as secondary goal and the LineItem.costType must be CostType.CPA.
     */
    UnitType["VIEW_THROUGH_CPA_CONVERSIONS"] = "VIEW_THROUGH_CPA_CONVERSIONS";
    /**
     * The number of total Cost-Per-Action (CPA) conversions from creatives associated with the line item. This is only supported as secondary goal and the LineItem.costType must be CostType.CPA.
     */
    UnitType["TOTAL_CPA_CONVERSIONS"] = "TOTAL_CPA_CONVERSIONS";
    /**
     * The number of viewable impressions reported by creatives associated with the line item. The LineItem.lineItemType must be LineItemType.STANDARD.
     */
    UnitType["VIEWABLE_IMPRESSIONS"] = "VIEWABLE_IMPRESSIONS";
    /**
     * The number of in-target impressions reported by third party measurements. The LineItem.lineItemType must be LineItemType.STANDARD.
     */
    UnitType["IN_TARGET_IMPRESSIONS"] = "IN_TARGET_IMPRESSIONS";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    UnitType["UNKNOWN"] = "UNKNOWN";
})(UnitType || (exports.UnitType = UnitType = {}));
/**
 * Represents a UnitType enum in superstruct format.
 */
exports.UnitTypeEnum = (0, superstruct_1.enums)([
    "IMPRESSIONS",
    "CLICKS",
    "CLICK_THROUGH_CPA_CONVERSIONS",
    "VIEW_THROUGH_CPA_CONVERSIONS",
    "TOTAL_CPA_CONVERSIONS",
    "VIEWABLE_IMPRESSIONS",
    "IN_TARGET_IMPRESSIONS",
    "UNKNOWN",
]);
