// @guardian/google-admanager-api/dist/client/common/enums/grpSettings.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PacingDeviceCategorizationTypeEnum = exports.PacingDeviceCategorizationType = exports.NielsenCtvPacingTypeEnum = exports.NielsenCtvPacingType = exports.GrpProviderEnum = exports.GrpProvider = exports.GrpTargetGenderEnum = exports.GrpTargetGender = void 0;
const superstruct_1 = require("superstruct");
/**
 * Represents the target gender for a GRP demographic targeted line item.
 */
var GrpTargetGender;
(function (GrpTargetGender) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    GrpTargetGender["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates that the GRP target gender is Male.
     */
    GrpTargetGender["MALE"] = "MALE";
    /**
     * Indicates that the GRP target gender is Female.
     */
    GrpTargetGender["FEMALE"] = "FEMALE";
    /**
     * Indicates that the GRP target gender is both male and female.
     */
    GrpTargetGender["BOTH"] = "BOTH";
})(GrpTargetGender || (exports.GrpTargetGender = GrpTargetGender = {}));
/**
 * Represents a GrpTargetGender enum in superstruct format.
 */
exports.GrpTargetGenderEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "MALE",
    "FEMALE",
    "BOTH",
]);
/**
 * Represents available GRP providers that a line item will have its target demographic measured by.
 */
var GrpProvider;
(function (GrpProvider) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    GrpProvider["UNKNOWN"] = "UNKNOWN";
    GrpProvider["NIELSEN"] = "NIELSEN";
    /**
     * Renamed to GOOGLE beginning in V201608.
     */
    GrpProvider["GOOGLE"] = "GOOGLE";
})(GrpProvider || (exports.GrpProvider = GrpProvider = {}));
/**
 * Represents a GrpProvider enum in superstruct format.
 */
exports.GrpProviderEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "NIELSEN",
    "GOOGLE",
]);
/**
 * Represents the pacing computation method for impressions on connected devices for a Nielsen measured line item.
 * This only applies when Nielsen measurement is enabled for connected devices.
 */
var NielsenCtvPacingType;
(function (NielsenCtvPacingType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    NielsenCtvPacingType["UNKNOWN"] = "UNKNOWN";
    /**
     * The value returned if Nielsen measurement is disabled for connected devices.
     */
    NielsenCtvPacingType["NONE"] = "NONE";
    /**
     * Indicates that Nielsen impressions on connected devices are included, and we apply coviewing in pacing.
     */
    NielsenCtvPacingType["COVIEW"] = "COVIEW";
    /**
     * Indicates that Nielsen impressions on connected devices are included, and we apply strict coviewing in pacing.
     */
    NielsenCtvPacingType["STRICT_COVIEW"] = "STRICT_COVIEW";
})(NielsenCtvPacingType || (exports.NielsenCtvPacingType = NielsenCtvPacingType = {}));
/**
 * Represents a NielsenCtvPacingType enum in superstruct format.
 */
exports.NielsenCtvPacingTypeEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "NONE",
    "COVIEW",
    "STRICT_COVIEW",
]);
/**
 * Represents whose device categorization to use on Nielsen measured line item with auto-pacing enabled.
 */
var PacingDeviceCategorizationType;
(function (PacingDeviceCategorizationType) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    PacingDeviceCategorizationType["UNKNOWN"] = "UNKNOWN";
    /**
     * Use Google's device categorization in auto-pacing.
     */
    PacingDeviceCategorizationType["GOOGLE"] = "GOOGLE";
    /**
     * Use Nielsen device categorization in auto-pacing
     */
    PacingDeviceCategorizationType["NIELSEN"] = "NIELSEN";
})(PacingDeviceCategorizationType || (exports.PacingDeviceCategorizationType = PacingDeviceCategorizationType = {}));
/**
 * Represents a PacingDeviceCategorizationType enum in superstruct format.
 */
exports.PacingDeviceCategorizationTypeEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "GOOGLE",
    "NIELSEN",
]);
