// @guardian/google-admanager-api/dist/client/common/enums/invitationStatus.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvitationStatus = void 0;
var InvitationStatus;
(function (InvitationStatus) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    InvitationStatus["UNKNOWN"] = "UNKNOWN";
    /**
     * The association request from the parent network has been accepted by the child network.
     */
    InvitationStatus["ACCEPTED"] = "ACCEPTED";
    /**
     * The invitation has expired. An invitation is valid for 90 days. Expired invitations can be resent.
     */
    InvitationStatus["EXPIRED"] = "EXPIRED";
    /**
     * The child publisher has not acted on the invite from the parent.
     */
    InvitationStatus["PENDING"] = "PENDING";
    /**
     * The child publisher has declined the invite.
     */
    InvitationStatus["REJECTED"] = "REJECTED";
    /**
     * The parent network withdrew the invite.
     */
    InvitationStatus["WITHDRAWN"] = "WITHDRAWN";
    /**
     * The invitation was disapproved by Google.
     */
    InvitationStatus["DEACTIVATED_BY_AD_MANAGER"] = "DEACTIVATED_BY_AD_MANAGER";
})(InvitationStatus || (exports.InvitationStatus = InvitationStatus = {}));
