// @guardian/google-admanager-api/dist/client/common/enums/lineItemSummary.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChildContentEligibilityEnum = exports.ChildContentEligibility = exports.CompetitiveConstraintScopeEnum = exports.CompetitiveConstraintScope = exports.LineItemSummaryReservationStatusEnum = exports.LineItemSummaryReservationStatus = exports.LineItemDiscountTypeEnum = exports.LineItemDiscountType = exports.CostTypeEnum = exports.CostType = exports.CustomPacingGoalUnitEnum = exports.CustomPacingGoalUnit = exports.CreativeRotationTypeEnum = exports.CreativeRotationType = void 0;
const superstruct_1 = require("superstruct");
/**
 * The strategy to use for displaying multiple {@link https://developers.google.com/ad-manager/api/reference/v202505/CreativeService.Creative Creative} \
 * objects that are associated with a {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem}.
 */
var CreativeRotationType;
(function (CreativeRotationType) {
    /**
     * Creatives are displayed roughly the same number of times over the duration of the line item.
     */
    CreativeRotationType["EVEN"] = "EVEN";
    /**
     * Creatives are served roughly proportionally to their performance.
     */
    CreativeRotationType["OPTIMIZED"] = "OPTIMIZED";
    /**
     * Creatives are served roughly proportionally to their weights, set on the LineItemCreativeAssociation.
     */
    CreativeRotationType["MANUAL"] = "MANUAL";
    /**
     * Creatives are served exactly in sequential order, aka Storyboarding. Set on the LineItemCreativeAssociation.
     */
    CreativeRotationType["SEQUENTIAL"] = "SEQUENTIAL";
})(CreativeRotationType || (exports.CreativeRotationType = CreativeRotationType = {}));
/**
 * Represents a CreativeRotationType enum in superstruct format.
 */
exports.CreativeRotationTypeEnum = (0, superstruct_1.enums)([
    "EVEN",
    "OPTIMIZED",
    "MANUAL",
    "SEQUENTIAL",
]);
/**
 * Options for the unit of the custom pacing goal amounts.
 */
var CustomPacingGoalUnit;
(function (CustomPacingGoalUnit) {
    /**
     * The custom pacing goal amounts represent absolute numbers corresponding to the line item's Goal.unitType.
     */
    CustomPacingGoalUnit["ABSOLUTE"] = "ABSOLUTE";
    /**
     * The custom pacing goal amounts represent a millipercent. For example, 15000 millipercent equals 15%.
     */
    CustomPacingGoalUnit["MILLI_PERCENT"] = "MILLI_PERCENT";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CustomPacingGoalUnit["UNKNOWN"] = "UNKNOWN";
})(CustomPacingGoalUnit || (exports.CustomPacingGoalUnit = CustomPacingGoalUnit = {}));
/**
 * Represents a CustomPacingGoalUnit enum in superstruct format.
 */
exports.CustomPacingGoalUnitEnum = (0, superstruct_1.enums)([
    "ABSOLUTE",
    "MILLI_PERCENT",
    "UNKNOWN",
]);
/**
 * Describes the {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem} actions that are billable.
 */
var CostType;
(function (CostType) {
    /**
     * Cost per action. The LineItem.lineItemType must be one of:
     *
     * - LineItemType.SPONSORSHIP
     * - LineItemType.STANDARD
     * - LineItemType.BULK
     * - LineItemType.NETWORK
     */
    CostType["CPA"] = "CPA";
    /**
     * Cost per click. The LineItem.lineItemType must be one of:
     *
     * - LineItemType.SPONSORSHIP
     * - LineItemType.STANDARD
     * - LineItemType.BULK
     * - LineItemType.NETWORK
     * - LineItemType.PRICE_PRIORITY
     * - LineItemType.HOUSE
     */
    CostType["CPC"] = "CPC";
    /**
     * Cost per day. The LineItem.lineItemType must be one of:
     *
     * - LineItemType.SPONSORSHIP
     * - LineItemType.NETWORK
     */
    CostType["CPD"] = "CPD";
    /**
     * Cost per mille (cost per thousand impressions). The LineItem.lineItemType must be one of:
     *
     * - LineItemType.SPONSORSHIP
     * - LineItemType.STANDARD
     * - LineItemType.BULK
     * - LineItemType.NETWORK
     * - LineItemType.PRICE_PRIORITY
     * - LineItemType.HOUSE
     */
    CostType["CPM"] = "CPM";
    /**
     * Cost per thousand Active View viewable impressions. The LineItem.lineItemType must be LineItemType.STANDARD.
     *
     */
    CostType["VCPM"] = "VCPM";
    /**
     * Cost per thousand in-target impressions. The LineItem.lineItemType must be LineItemType.STANDARD.
     */
    CostType["CPM_IN_TARGET"] = "CPM_IN_TARGET";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CostType["UNKNOWN"] = "UNKNOWN";
})(CostType || (exports.CostType = CostType = {}));
/**
 * Represents a CostType enum in superstruct format.
 */
exports.CostTypeEnum = (0, superstruct_1.enums)([
    "CPA",
    "CPC",
    "CPD",
    "CPM",
    "VCPM",
    "CPM_IN_TARGET",
    "UNKNOWN",
]);
/**
 * Describes the possible discount types on the cost of booking a {@link https://developers.google.com/ad-manager/api/reference/v202505/LineItemService.LineItem LineItem}.
 */
var LineItemDiscountType;
(function (LineItemDiscountType) {
    /**
     * An absolute value will be discounted from the line item's cost.
     */
    LineItemDiscountType["ABSOLUTE_VALUE"] = "ABSOLUTE_VALUE";
    /**
     * A percentage of the cost will be applied as discount for booking the line item.
     */
    LineItemDiscountType["PERCENTAGE"] = "PERCENTAGE";
})(LineItemDiscountType || (exports.LineItemDiscountType = LineItemDiscountType = {}));
/**
 * Represents a LineItemDiscountType enum in superstruct format.
 */
exports.LineItemDiscountTypeEnum = (0, superstruct_1.enums)([
    "ABSOLUTE_VALUE",
    "PERCENTAGE",
]);
/**
 * Specifies the reservation status of the LineItem.
 */
var LineItemSummaryReservationStatus;
(function (LineItemSummaryReservationStatus) {
    /**
     * Indicates that inventory has been reserved for the line item.
     */
    LineItemSummaryReservationStatus["RESERVED"] = "RESERVED";
    /**
     * Indicates that inventory has not been reserved for the line item.
     */
    LineItemSummaryReservationStatus["UNRESERVED"] = "UNRESERVED";
})(LineItemSummaryReservationStatus || (exports.LineItemSummaryReservationStatus = LineItemSummaryReservationStatus = {}));
/**
 * Represents a LineItemSummaryReservationStatus enum in superstruct format.
 */
exports.LineItemSummaryReservationStatusEnum = (0, superstruct_1.enums)([
    "RESERVED",
    "UNRESERVED",
]);
/**
 * The scope to which the assignment of any competitive exclusion labels for a video line item is limited.
 */
var CompetitiveConstraintScope;
(function (CompetitiveConstraintScope) {
    /**
     * The competitive exclusion label applies to all line items within a single pod (or group).
     */
    CompetitiveConstraintScope["POD"] = "POD";
    /**
     * The competitive exclusion label applies to all line items within the entire stream of content.
     */
    CompetitiveConstraintScope["STREAM"] = "STREAM";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    CompetitiveConstraintScope["UNKNOWN"] = "UNKNOWN";
})(CompetitiveConstraintScope || (exports.CompetitiveConstraintScope = CompetitiveConstraintScope = {}));
/**
 * Represents a CompetitiveConstraintScope enum in superstruct format.
 */
exports.CompetitiveConstraintScopeEnum = (0, superstruct_1.enums)([
    "POD",
    "STREAM",
    "UNKNOWN",
]);
/**
 * Child content eligibility designation.
 *
 * This field is optional and defaults to ChildContentEligibility.DISALLOWED. This field has no effect on serving enforcement unless you opt to "Child content enforcement" in the network's Child Content settings.
 */
var ChildContentEligibility;
(function (ChildContentEligibility) {
    ChildContentEligibility["UNKNOWN"] = "UNKNOWN";
    /**
     * This line item is not eligible to serve on any requests that are child-directed.
     */
    ChildContentEligibility["DISALLOWED"] = "DISALLOWED";
    /**
     * This line item is eligible to serve on requests that are child-directed.
     */
    ChildContentEligibility["ALLOWED"] = "ALLOWED";
})(ChildContentEligibility || (exports.ChildContentEligibility = ChildContentEligibility = {}));
/**
 * Represents a ChildContentEligibility enum in superstruct format.
 */
exports.ChildContentEligibilityEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "DISALLOWED",
    "ALLOWED",
]);
