// @guardian/google-admanager-api/dist/client/common/enums/thirdParty.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThirdPartyReachIntegrationPartnerEnum = exports.ThirdPartyReachIntegrationPartner = exports.ThirdPartyBrandLiftIntegrationPartnerEnum = exports.ThirdPartyBrandLiftIntegrationPartner = exports.ThirdPartyViewabilityIntegrationPartnerEnum = exports.ThirdPartyViewabilityIntegrationPartner = void 0;
const superstruct_1 = require("superstruct");
/**
 * Possible options for third-party viewabitility integration.
 */
var ThirdPartyViewabilityIntegrationPartner;
(function (ThirdPartyViewabilityIntegrationPartner) {
    /**
     * Indicates there's no third-party viewability integration partner.
     */
    ThirdPartyViewabilityIntegrationPartner["NONE"] = "NONE";
    /**
     * Indicates third-party viewability integration partner Oracle Moat.
     */
    ThirdPartyViewabilityIntegrationPartner["MOAT"] = "MOAT";
    /**
     * Indicates third-party viewability integration partner Double Verify.
     */
    ThirdPartyViewabilityIntegrationPartner["DOUBLE_VERIFY"] = "DOUBLE_VERIFY";
    /**
     * Indicates third-party viewability integration partner Integral Ad Science.
     */
    ThirdPartyViewabilityIntegrationPartner["INTEGRAL_AD_SCIENCE"] = "INTEGRAL_AD_SCIENCE";
    /**
     * Indicates third-party viewability integration partner Comscore.
     */
    ThirdPartyViewabilityIntegrationPartner["COMSCORE"] = "COMSCORE";
    /**
     * Indicates third-party viewability integration partner Telemetry.
     */
    ThirdPartyViewabilityIntegrationPartner["TELEMETRY"] = "TELEMETRY";
    /**
     * Indicates third-party viewability integration partner Meetrics.
     */
    ThirdPartyViewabilityIntegrationPartner["MEETRICS"] = "MEETRICS";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ThirdPartyViewabilityIntegrationPartner["UNKNOWN"] = "UNKNOWN";
})(ThirdPartyViewabilityIntegrationPartner || (exports.ThirdPartyViewabilityIntegrationPartner = ThirdPartyViewabilityIntegrationPartner = {}));
/**
 * Represents a ThirdPartyViewabilityIntegrationPartner enum in superstruct format.
 */
exports.ThirdPartyViewabilityIntegrationPartnerEnum = (0, superstruct_1.enums)([
    "NONE",
    "MOAT",
    "DOUBLE_VERIFY",
    "INTEGRAL_AD_SCIENCE",
    "COMSCORE",
    "TELEMETRY",
    "MEETRICS",
    "UNKNOWN",
]);
/**
 * Possible options for third-party brand lift integration.
 */
var ThirdPartyBrandLiftIntegrationPartner;
(function (ThirdPartyBrandLiftIntegrationPartner) {
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ThirdPartyBrandLiftIntegrationPartner["UNKNOWN"] = "UNKNOWN";
    /**
     * Indicates there's no third-party brand lift integration partner.
     */
    ThirdPartyBrandLiftIntegrationPartner["NONE"] = "NONE";
    /**
     * Indicates third-party brand lift integration partner Kantar.
     */
    ThirdPartyBrandLiftIntegrationPartner["KANTAR_MILLWARD_BROWN"] = "KANTAR_MILLWARD_BROWN";
    /**
     * Indicates third-party brand lift integration partner Dynata.
     */
    ThirdPartyBrandLiftIntegrationPartner["DYNATA"] = "DYNATA";
})(ThirdPartyBrandLiftIntegrationPartner || (exports.ThirdPartyBrandLiftIntegrationPartner = ThirdPartyBrandLiftIntegrationPartner = {}));
/**
 * Represents a ThirdPartyBrandLiftIntegrationPartner enum in superstruct format.
 */
exports.ThirdPartyBrandLiftIntegrationPartnerEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "NONE",
    "KANTAR_MILLWARD_BROWN",
    "DYNATA",
]);
/**
 * Possible options for third-party reach integration.
 */
var ThirdPartyReachIntegrationPartner;
(function (ThirdPartyReachIntegrationPartner) {
    /**
     * Indicates there's no third-party reach integration partner.
     */
    ThirdPartyReachIntegrationPartner["NONE"] = "NONE";
    /**
     * Indicates third-party reach integration partner Comscore.
     */
    ThirdPartyReachIntegrationPartner["COMSCORE"] = "COMSCORE";
    /**
     * Indicates third-party reach integration partner Nielsen.
     */
    ThirdPartyReachIntegrationPartner["NIELSEN"] = "NIELSEN";
    /**
     * Indicates third-party reach integration partner Kantar.
     */
    ThirdPartyReachIntegrationPartner["KANTAR_MILLWARD_BROWN"] = "KANTAR_MILLWARD_BROWN";
    /**
     * The value returned if the actual value is not exposed by the requested API version.
     */
    ThirdPartyReachIntegrationPartner["UNKNOWN"] = "UNKNOWN";
})(ThirdPartyReachIntegrationPartner || (exports.ThirdPartyReachIntegrationPartner = ThirdPartyReachIntegrationPartner = {}));
/**
 * Represents a ThirdPartyReachIntegrationPartner enum in superstruct format.
 */
exports.ThirdPartyReachIntegrationPartnerEnum = (0, superstruct_1.enums)([
    "NONE",
    "COMSCORE",
    "NIELSEN",
    "KANTAR_MILLWARD_BROWN",
    "UNKNOWN",
]);
