// @guardian/google-admanager-api/dist/client/common/enums/videoPosition.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VideoBumperTypeEnum = exports.VideoBumperType = exports.VideoPositionTypeEnum = exports.VideoPositionType = void 0;
const superstruct_1 = require("superstruct");
/**
 * Represents a targetable position within a video.
 */
var VideoPositionType;
(function (VideoPositionType) {
    /**
     * 	The value returned if the actual value is not exposed by the requested API version.
     */
    VideoPositionType["UNKNOWN"] = "UNKNOWN";
    /**
     * 	This position targets all of the above video positions.
     */
    VideoPositionType["ALL"] = "ALL";
    /**
     * 	The position defined as showing before the video starts playing.
     */
    VideoPositionType["PREROLL"] = "PREROLL";
    /**
     * 	The position defined as showing within the middle of the playing video.
     */
    VideoPositionType["MIDROLL"] = "MIDROLL";
    /**
     * 	The position defined as showing after the video is completed.
     */
    VideoPositionType["POSTROLL"] = "POSTROLL";
})(VideoPositionType || (exports.VideoPositionType = VideoPositionType = {}));
/**
 * Represents a VideoPositionType enum in superstruct format.
 */
exports.VideoPositionTypeEnum = (0, superstruct_1.enums)([
    "UNKNOWN",
    "ALL",
    "PREROLL",
    "MIDROLL",
    "POSTROLL",
]);
/**
 * Represents the options for targetable bumper positions, surrounding an ad pod, within a video stream. This includes before and after the supported ad pod positions, VideoPositionType.PREROLL, VideoPositionType.MIDROLL, and VideoPositionType.POSTROLL.
 */
var VideoBumperType;
(function (VideoBumperType) {
    /**
     * 	Represents the bumper position before the ad pod.
     */
    VideoBumperType["BEFORE"] = "BEFORE";
    /**
     * 	Represents the bumper position after the ad pod.
     */
    VideoBumperType["AFTER"] = "AFTER";
})(VideoBumperType || (exports.VideoBumperType = VideoBumperType = {}));
/**
 * Represents a VideoBumperType enum in superstruct format.
 */
exports.VideoBumperTypeEnum = (0, superstruct_1.enums)([
    "BEFORE",
    "AFTER",
]);
